// @superviz/yjs/dist/index.cjs.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from2, except, desc) => {
  if (from2 && typeof from2 === "object" || typeof from2 === "function") {
    for (let key of __getOwnPropNames(from2))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from2[key], enumerable: !(desc = __getOwnPropDesc(from2, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// ../../node_modules/.pnpm/ms@2.1.3/node_modules/ms/index.js
var require_ms = __commonJS({
  "../../node_modules/.pnpm/ms@2.1.3/node_modules/ms/index.js"(exports2, module2) {
    var s = 1e3;
    var m = s * 60;
    var h = m * 60;
    var d = h * 24;
    var w = d * 7;
    var y = d * 365.25;
    module2.exports = function(val, options) {
      options = options || {};
      var type = typeof val;
      if (type === "string" && val.length > 0) {
        return parse(val);
      } else if (type === "number" && isFinite(val)) {
        return options.long ? fmtLong(val) : fmtShort(val);
      }
      throw new Error(
        "val is not a non-empty string or a valid number. val=" + JSON.stringify(val)
      );
    };
    function parse(str) {
      str = String(str);
      if (str.length > 100) {
        return;
      }
      var match = /^(-?(?:\d+)?\.?\d+) *(milliseconds?|msecs?|ms|seconds?|secs?|s|minutes?|mins?|m|hours?|hrs?|h|days?|d|weeks?|w|years?|yrs?|y)?$/i.exec(
        str
      );
      if (!match) {
        return;
      }
      var n = parseFloat(match[1]);
      var type = (match[2] || "ms").toLowerCase();
      switch (type) {
        case "years":
        case "year":
        case "yrs":
        case "yr":
        case "y":
          return n * y;
        case "weeks":
        case "week":
        case "w":
          return n * w;
        case "days":
        case "day":
        case "d":
          return n * d;
        case "hours":
        case "hour":
        case "hrs":
        case "hr":
        case "h":
          return n * h;
        case "minutes":
        case "minute":
        case "mins":
        case "min":
        case "m":
          return n * m;
        case "seconds":
        case "second":
        case "secs":
        case "sec":
        case "s":
          return n * s;
        case "milliseconds":
        case "millisecond":
        case "msecs":
        case "msec":
        case "ms":
          return n;
        default:
          return void 0;
      }
    }
    function fmtShort(ms) {
      var msAbs = Math.abs(ms);
      if (msAbs >= d) {
        return Math.round(ms / d) + "d";
      }
      if (msAbs >= h) {
        return Math.round(ms / h) + "h";
      }
      if (msAbs >= m) {
        return Math.round(ms / m) + "m";
      }
      if (msAbs >= s) {
        return Math.round(ms / s) + "s";
      }
      return ms + "ms";
    }
    function fmtLong(ms) {
      var msAbs = Math.abs(ms);
      if (msAbs >= d) {
        return plural(ms, msAbs, d, "day");
      }
      if (msAbs >= h) {
        return plural(ms, msAbs, h, "hour");
      }
      if (msAbs >= m) {
        return plural(ms, msAbs, m, "minute");
      }
      if (msAbs >= s) {
        return plural(ms, msAbs, s, "second");
      }
      return ms + " ms";
    }
    function plural(ms, msAbs, n, name) {
      var isPlural = msAbs >= n * 1.5;
      return Math.round(ms / n) + " " + name + (isPlural ? "s" : "");
    }
  }
});

// ../../node_modules/.pnpm/debug@4.3.7/node_modules/debug/src/common.js
var require_common = __commonJS({
  "../../node_modules/.pnpm/debug@4.3.7/node_modules/debug/src/common.js"(exports2, module2) {
    function setup(env) {
      createDebug.debug = createDebug;
      createDebug.default = createDebug;
      createDebug.coerce = coerce;
      createDebug.disable = disable;
      createDebug.enable = enable;
      createDebug.enabled = enabled;
      createDebug.humanize = require_ms();
      createDebug.destroy = destroy;
      Object.keys(env).forEach((key) => {
        createDebug[key] = env[key];
      });
      createDebug.names = [];
      createDebug.skips = [];
      createDebug.formatters = {};
      function selectColor(namespace) {
        let hash = 0;
        for (let i = 0; i < namespace.length; i++) {
          hash = (hash << 5) - hash + namespace.charCodeAt(i);
          hash |= 0;
        }
        return createDebug.colors[Math.abs(hash) % createDebug.colors.length];
      }
      createDebug.selectColor = selectColor;
      function createDebug(namespace) {
        let prevTime;
        let enableOverride = null;
        let namespacesCache;
        let enabledCache;
        function debug2(...args) {
          if (!debug2.enabled) {
            return;
          }
          const self = debug2;
          const curr = Number(/* @__PURE__ */ new Date());
          const ms = curr - (prevTime || curr);
          self.diff = ms;
          self.prev = prevTime;
          self.curr = curr;
          prevTime = curr;
          args[0] = createDebug.coerce(args[0]);
          if (typeof args[0] !== "string") {
            args.unshift("%O");
          }
          let index = 0;
          args[0] = args[0].replace(/%([a-zA-Z%])/g, (match, format) => {
            if (match === "%%") {
              return "%";
            }
            index++;
            const formatter = createDebug.formatters[format];
            if (typeof formatter === "function") {
              const val = args[index];
              match = formatter.call(self, val);
              args.splice(index, 1);
              index--;
            }
            return match;
          });
          createDebug.formatArgs.call(self, args);
          const logFn = self.log || createDebug.log;
          logFn.apply(self, args);
        }
        debug2.namespace = namespace;
        debug2.useColors = createDebug.useColors();
        debug2.color = createDebug.selectColor(namespace);
        debug2.extend = extend;
        debug2.destroy = createDebug.destroy;
        Object.defineProperty(debug2, "enabled", {
          enumerable: true,
          configurable: false,
          get: () => {
            if (enableOverride !== null) {
              return enableOverride;
            }
            if (namespacesCache !== createDebug.namespaces) {
              namespacesCache = createDebug.namespaces;
              enabledCache = createDebug.enabled(namespace);
            }
            return enabledCache;
          },
          set: (v) => {
            enableOverride = v;
          }
        });
        if (typeof createDebug.init === "function") {
          createDebug.init(debug2);
        }
        return debug2;
      }
      function extend(namespace, delimiter) {
        const newDebug = createDebug(this.namespace + (typeof delimiter === "undefined" ? ":" : delimiter) + namespace);
        newDebug.log = this.log;
        return newDebug;
      }
      function enable(namespaces) {
        createDebug.save(namespaces);
        createDebug.namespaces = namespaces;
        createDebug.names = [];
        createDebug.skips = [];
        let i;
        const split = (typeof namespaces === "string" ? namespaces : "").split(/[\s,]+/);
        const len = split.length;
        for (i = 0; i < len; i++) {
          if (!split[i]) {
            continue;
          }
          namespaces = split[i].replace(/\*/g, ".*?");
          if (namespaces[0] === "-") {
            createDebug.skips.push(new RegExp("^" + namespaces.slice(1) + "$"));
          } else {
            createDebug.names.push(new RegExp("^" + namespaces + "$"));
          }
        }
      }
      function disable() {
        const namespaces = [
          ...createDebug.names.map(toNamespace),
          ...createDebug.skips.map(toNamespace).map((namespace) => "-" + namespace)
        ].join(",");
        createDebug.enable("");
        return namespaces;
      }
      function enabled(name) {
        if (name[name.length - 1] === "*") {
          return true;
        }
        let i;
        let len;
        for (i = 0, len = createDebug.skips.length; i < len; i++) {
          if (createDebug.skips[i].test(name)) {
            return false;
          }
        }
        for (i = 0, len = createDebug.names.length; i < len; i++) {
          if (createDebug.names[i].test(name)) {
            return true;
          }
        }
        return false;
      }
      function toNamespace(regexp) {
        return regexp.toString().substring(2, regexp.toString().length - 2).replace(/\.\*\?$/, "*");
      }
      function coerce(val) {
        if (val instanceof Error) {
          return val.stack || val.message;
        }
        return val;
      }
      function destroy() {
        console.warn("Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`.");
      }
      createDebug.enable(createDebug.load());
      return createDebug;
    }
    module2.exports = setup;
  }
});

// ../../node_modules/.pnpm/debug@4.3.7/node_modules/debug/src/browser.js
var require_browser = __commonJS({
  "../../node_modules/.pnpm/debug@4.3.7/node_modules/debug/src/browser.js"(exports2, module2) {
    exports2.formatArgs = formatArgs;
    exports2.save = save;
    exports2.load = load;
    exports2.useColors = useColors;
    exports2.storage = localstorage();
    exports2.destroy = /* @__PURE__ */ (() => {
      let warned = false;
      return () => {
        if (!warned) {
          warned = true;
          console.warn("Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`.");
        }
      };
    })();
    exports2.colors = [
      "#0000CC",
      "#0000FF",
      "#0033CC",
      "#0033FF",
      "#0066CC",
      "#0066FF",
      "#0099CC",
      "#0099FF",
      "#00CC00",
      "#00CC33",
      "#00CC66",
      "#00CC99",
      "#00CCCC",
      "#00CCFF",
      "#3300CC",
      "#3300FF",
      "#3333CC",
      "#3333FF",
      "#3366CC",
      "#3366FF",
      "#3399CC",
      "#3399FF",
      "#33CC00",
      "#33CC33",
      "#33CC66",
      "#33CC99",
      "#33CCCC",
      "#33CCFF",
      "#6600CC",
      "#6600FF",
      "#6633CC",
      "#6633FF",
      "#66CC00",
      "#66CC33",
      "#9900CC",
      "#9900FF",
      "#9933CC",
      "#9933FF",
      "#99CC00",
      "#99CC33",
      "#CC0000",
      "#CC0033",
      "#CC0066",
      "#CC0099",
      "#CC00CC",
      "#CC00FF",
      "#CC3300",
      "#CC3333",
      "#CC3366",
      "#CC3399",
      "#CC33CC",
      "#CC33FF",
      "#CC6600",
      "#CC6633",
      "#CC9900",
      "#CC9933",
      "#CCCC00",
      "#CCCC33",
      "#FF0000",
      "#FF0033",
      "#FF0066",
      "#FF0099",
      "#FF00CC",
      "#FF00FF",
      "#FF3300",
      "#FF3333",
      "#FF3366",
      "#FF3399",
      "#FF33CC",
      "#FF33FF",
      "#FF6600",
      "#FF6633",
      "#FF9900",
      "#FF9933",
      "#FFCC00",
      "#FFCC33"
    ];
    function useColors() {
      if (typeof window !== "undefined" && window.process && (window.process.type === "renderer" || window.process.__nwjs)) {
        return true;
      }
      if (typeof navigator !== "undefined" && navigator.userAgent && navigator.userAgent.toLowerCase().match(/(edge|trident)\/(\d+)/)) {
        return false;
      }
      let m;
      return typeof document !== "undefined" && document.documentElement && document.documentElement.style && document.documentElement.style.WebkitAppearance || // Is firebug? http://stackoverflow.com/a/398120/376773
      typeof window !== "undefined" && window.console && (window.console.firebug || window.console.exception && window.console.table) || // Is firefox >= v31?
      // https://developer.mozilla.org/en-US/docs/Tools/Web_Console#Styling_messages
      typeof navigator !== "undefined" && navigator.userAgent && (m = navigator.userAgent.toLowerCase().match(/firefox\/(\d+)/)) && parseInt(m[1], 10) >= 31 || // Double check webkit in userAgent just in case we are in a worker
      typeof navigator !== "undefined" && navigator.userAgent && navigator.userAgent.toLowerCase().match(/applewebkit\/(\d+)/);
    }
    function formatArgs(args) {
      args[0] = (this.useColors ? "%c" : "") + this.namespace + (this.useColors ? " %c" : " ") + args[0] + (this.useColors ? "%c " : " ") + "+" + module2.exports.humanize(this.diff);
      if (!this.useColors) {
        return;
      }
      const c = "color: " + this.color;
      args.splice(1, 0, c, "color: inherit");
      let index = 0;
      let lastC = 0;
      args[0].replace(/%[a-zA-Z%]/g, (match) => {
        if (match === "%%") {
          return;
        }
        index++;
        if (match === "%c") {
          lastC = index;
        }
      });
      args.splice(lastC, 0, c);
    }
    exports2.log = console.debug || console.log || (() => {
    });
    function save(namespaces) {
      try {
        if (namespaces) {
          exports2.storage.setItem("debug", namespaces);
        } else {
          exports2.storage.removeItem("debug");
        }
      } catch (error) {
      }
    }
    function load() {
      let r;
      try {
        r = exports2.storage.getItem("debug");
      } catch (error) {
      }
      if (!r && typeof process !== "undefined" && "env" in process) {
        r = process.env.DEBUG;
      }
      return r;
    }
    function localstorage() {
      try {
        return localStorage;
      } catch (error) {
      }
    }
    module2.exports = require_common()(exports2);
    var { formatters } = module2.exports;
    formatters.j = function(v) {
      try {
        return JSON.stringify(v);
      } catch (error) {
        return "[UnexpectedJSONParseError]: " + error.message;
      }
    };
  }
});

// ../../node_modules/.pnpm/has-flag@4.0.0/node_modules/has-flag/index.js
var require_has_flag = __commonJS({
  "../../node_modules/.pnpm/has-flag@4.0.0/node_modules/has-flag/index.js"(exports2, module2) {
    "use strict";
    module2.exports = (flag, argv = process.argv) => {
      const prefix = flag.startsWith("-") ? "" : flag.length === 1 ? "-" : "--";
      const position = argv.indexOf(prefix + flag);
      const terminatorPosition = argv.indexOf("--");
      return position !== -1 && (terminatorPosition === -1 || position < terminatorPosition);
    };
  }
});

// ../../node_modules/.pnpm/supports-color@8.1.1/node_modules/supports-color/index.js
var require_supports_color = __commonJS({
  "../../node_modules/.pnpm/supports-color@8.1.1/node_modules/supports-color/index.js"(exports2, module2) {
    "use strict";
    var os = require("os");
    var tty = require("tty");
    var hasFlag = require_has_flag();
    var { env } = process;
    var flagForceColor;
    if (hasFlag("no-color") || hasFlag("no-colors") || hasFlag("color=false") || hasFlag("color=never")) {
      flagForceColor = 0;
    } else if (hasFlag("color") || hasFlag("colors") || hasFlag("color=true") || hasFlag("color=always")) {
      flagForceColor = 1;
    }
    function envForceColor() {
      if ("FORCE_COLOR" in env) {
        if (env.FORCE_COLOR === "true") {
          return 1;
        }
        if (env.FORCE_COLOR === "false") {
          return 0;
        }
        return env.FORCE_COLOR.length === 0 ? 1 : Math.min(Number.parseInt(env.FORCE_COLOR, 10), 3);
      }
    }
    function translateLevel(level) {
      if (level === 0) {
        return false;
      }
      return {
        level,
        hasBasic: true,
        has256: level >= 2,
        has16m: level >= 3
      };
    }
    function supportsColor(haveStream, { streamIsTTY, sniffFlags = true } = {}) {
      const noFlagForceColor = envForceColor();
      if (noFlagForceColor !== void 0) {
        flagForceColor = noFlagForceColor;
      }
      const forceColor = sniffFlags ? flagForceColor : noFlagForceColor;
      if (forceColor === 0) {
        return 0;
      }
      if (sniffFlags) {
        if (hasFlag("color=16m") || hasFlag("color=full") || hasFlag("color=truecolor")) {
          return 3;
        }
        if (hasFlag("color=256")) {
          return 2;
        }
      }
      if (haveStream && !streamIsTTY && forceColor === void 0) {
        return 0;
      }
      const min = forceColor || 0;
      if (env.TERM === "dumb") {
        return min;
      }
      if (process.platform === "win32") {
        const osRelease = os.release().split(".");
        if (Number(osRelease[0]) >= 10 && Number(osRelease[2]) >= 10586) {
          return Number(osRelease[2]) >= 14931 ? 3 : 2;
        }
        return 1;
      }
      if ("CI" in env) {
        if (["TRAVIS", "CIRCLECI", "APPVEYOR", "GITLAB_CI", "GITHUB_ACTIONS", "BUILDKITE", "DRONE"].some((sign) => sign in env) || env.CI_NAME === "codeship") {
          return 1;
        }
        return min;
      }
      if ("TEAMCITY_VERSION" in env) {
        return /^(9\.(0*[1-9]\d*)\.|\d{2,}\.)/.test(env.TEAMCITY_VERSION) ? 1 : 0;
      }
      if (env.COLORTERM === "truecolor") {
        return 3;
      }
      if ("TERM_PROGRAM" in env) {
        const version = Number.parseInt((env.TERM_PROGRAM_VERSION || "").split(".")[0], 10);
        switch (env.TERM_PROGRAM) {
          case "iTerm.app":
            return version >= 3 ? 3 : 2;
          case "Apple_Terminal":
            return 2;
        }
      }
      if (/-256(color)?$/i.test(env.TERM)) {
        return 2;
      }
      if (/^screen|^xterm|^vt100|^vt220|^rxvt|color|ansi|cygwin|linux/i.test(env.TERM)) {
        return 1;
      }
      if ("COLORTERM" in env) {
        return 1;
      }
      return min;
    }
    function getSupportLevel(stream, options = {}) {
      const level = supportsColor(stream, {
        streamIsTTY: stream && stream.isTTY,
        ...options
      });
      return translateLevel(level);
    }
    module2.exports = {
      supportsColor: getSupportLevel,
      stdout: getSupportLevel({ isTTY: tty.isatty(1) }),
      stderr: getSupportLevel({ isTTY: tty.isatty(2) })
    };
  }
});

// ../../node_modules/.pnpm/debug@4.3.7/node_modules/debug/src/node.js
var require_node = __commonJS({
  "../../node_modules/.pnpm/debug@4.3.7/node_modules/debug/src/node.js"(exports2, module2) {
    var tty = require("tty");
    var util = require("util");
    exports2.init = init;
    exports2.log = log;
    exports2.formatArgs = formatArgs;
    exports2.save = save;
    exports2.load = load;
    exports2.useColors = useColors;
    exports2.destroy = util.deprecate(
      () => {
      },
      "Instance method `debug.destroy()` is deprecated and no longer does anything. It will be removed in the next major version of `debug`."
    );
    exports2.colors = [6, 2, 3, 4, 5, 1];
    try {
      const supportsColor = require_supports_color();
      if (supportsColor && (supportsColor.stderr || supportsColor).level >= 2) {
        exports2.colors = [
          20,
          21,
          26,
          27,
          32,
          33,
          38,
          39,
          40,
          41,
          42,
          43,
          44,
          45,
          56,
          57,
          62,
          63,
          68,
          69,
          74,
          75,
          76,
          77,
          78,
          79,
          80,
          81,
          92,
          93,
          98,
          99,
          112,
          113,
          128,
          129,
          134,
          135,
          148,
          149,
          160,
          161,
          162,
          163,
          164,
          165,
          166,
          167,
          168,
          169,
          170,
          171,
          172,
          173,
          178,
          179,
          184,
          185,
          196,
          197,
          198,
          199,
          200,
          201,
          202,
          203,
          204,
          205,
          206,
          207,
          208,
          209,
          214,
          215,
          220,
          221
        ];
      }
    } catch (error) {
    }
    exports2.inspectOpts = Object.keys(process.env).filter((key) => {
      return /^debug_/i.test(key);
    }).reduce((obj, key) => {
      const prop = key.substring(6).toLowerCase().replace(/_([a-z])/g, (_, k) => {
        return k.toUpperCase();
      });
      let val = process.env[key];
      if (/^(yes|on|true|enabled)$/i.test(val)) {
        val = true;
      } else if (/^(no|off|false|disabled)$/i.test(val)) {
        val = false;
      } else if (val === "null") {
        val = null;
      } else {
        val = Number(val);
      }
      obj[prop] = val;
      return obj;
    }, {});
    function useColors() {
      return "colors" in exports2.inspectOpts ? Boolean(exports2.inspectOpts.colors) : tty.isatty(process.stderr.fd);
    }
    function formatArgs(args) {
      const { namespace: name, useColors: useColors2 } = this;
      if (useColors2) {
        const c = this.color;
        const colorCode = "\x1B[3" + (c < 8 ? c : "8;5;" + c);
        const prefix = `  ${colorCode};1m${name} \x1B[0m`;
        args[0] = prefix + args[0].split("\n").join("\n" + prefix);
        args.push(colorCode + "m+" + module2.exports.humanize(this.diff) + "\x1B[0m");
      } else {
        args[0] = getDate() + name + " " + args[0];
      }
    }
    function getDate() {
      if (exports2.inspectOpts.hideDate) {
        return "";
      }
      return (/* @__PURE__ */ new Date()).toISOString() + " ";
    }
    function log(...args) {
      return process.stderr.write(util.formatWithOptions(exports2.inspectOpts, ...args) + "\n");
    }
    function save(namespaces) {
      if (namespaces) {
        process.env.DEBUG = namespaces;
      } else {
        delete process.env.DEBUG;
      }
    }
    function load() {
      return process.env.DEBUG;
    }
    function init(debug2) {
      debug2.inspectOpts = {};
      const keys = Object.keys(exports2.inspectOpts);
      for (let i = 0; i < keys.length; i++) {
        debug2.inspectOpts[keys[i]] = exports2.inspectOpts[keys[i]];
      }
    }
    module2.exports = require_common()(exports2);
    var { formatters } = module2.exports;
    formatters.o = function(v) {
      this.inspectOpts.colors = this.useColors;
      return util.inspect(v, this.inspectOpts).split("\n").map((str) => str.trim()).join(" ");
    };
    formatters.O = function(v) {
      this.inspectOpts.colors = this.useColors;
      return util.inspect(v, this.inspectOpts);
    };
  }
});

// ../../node_modules/.pnpm/debug@4.3.7/node_modules/debug/src/index.js
var require_src = __commonJS({
  "../../node_modules/.pnpm/debug@4.3.7/node_modules/debug/src/index.js"(exports2, module2) {
    if (typeof process === "undefined" || process.type === "renderer" || process.browser === true || process.__nwjs) {
      module2.exports = require_browser();
    } else {
      module2.exports = require_node();
    }
  }
});

// src/index.ts
var src_exports = {};
__export(src_exports, {
  SuperVizYjsProvider: () => SuperVizYjsProvider
});
module.exports = __toCommonJS(src_exports);

// ../../node_modules/.pnpm/lib0@0.2.97/node_modules/lib0/map.js
var create = () => /* @__PURE__ */ new Map();
var setIfUndefined = (map, key, createT) => {
  let set = map.get(key);
  if (set === void 0) {
    map.set(key, set = createT());
  }
  return set;
};

// ../../node_modules/.pnpm/lib0@0.2.97/node_modules/lib0/set.js
var create2 = () => /* @__PURE__ */ new Set();

// ../../node_modules/.pnpm/lib0@0.2.97/node_modules/lib0/array.js
var from = Array.from;
var isArray = Array.isArray;

// ../../node_modules/.pnpm/lib0@0.2.97/node_modules/lib0/observable.js
var ObservableV2 = class {
  constructor() {
    this._observers = create();
  }
  /**
   * @template {keyof EVENTS & string} NAME
   * @param {NAME} name
   * @param {EVENTS[NAME]} f
   */
  on(name, f) {
    setIfUndefined(
      this._observers,
      /** @type {string} */
      name,
      create2
    ).add(f);
    return f;
  }
  /**
   * @template {keyof EVENTS & string} NAME
   * @param {NAME} name
   * @param {EVENTS[NAME]} f
   */
  once(name, f) {
    const _f = (...args) => {
      this.off(
        name,
        /** @type {any} */
        _f
      );
      f(...args);
    };
    this.on(
      name,
      /** @type {any} */
      _f
    );
  }
  /**
   * @template {keyof EVENTS & string} NAME
   * @param {NAME} name
   * @param {EVENTS[NAME]} f
   */
  off(name, f) {
    const observers = this._observers.get(name);
    if (observers !== void 0) {
      observers.delete(f);
      if (observers.size === 0) {
        this._observers.delete(name);
      }
    }
  }
  /**
   * Emit a named event. All registered event listeners that listen to the
   * specified name will receive the event.
   *
   * @todo This should catch exceptions
   *
   * @template {keyof EVENTS & string} NAME
   * @param {NAME} name The event name.
   * @param {Parameters<EVENTS[NAME]>} args The arguments that are applied to the event listener.
   */
  emit(name, args) {
    return from((this._observers.get(name) || create()).values()).forEach((f) => f(...args));
  }
  destroy() {
    this._observers = create();
  }
};

// src/provider/index.ts
var Y = __toESM(require("yjs"), 1);

// src/services/awareness/index.ts
var Awareness = class extends ObservableV2 {
  constructor(doc, logger) {
    super();
    this.doc = doc;
    this.logger = logger;
    this.clientID = 0;
    this.states = /* @__PURE__ */ new Map();
    // Not used, but added because some bindings expect these fields
    this._checkInterval = null;
    this.meta = /* @__PURE__ */ new Map();
    this.participantIdToClientId = /* @__PURE__ */ new Map();
    this.room = null;
    this.TIMEOUT_TIMER_MS = 3e4;
    this.YJS_STATE = "__yjs";
    this.previousState = null;
    this.participantId = "";
    this.setLocalState = (state) => {
      var _a, _b;
      this.logger.log("[SuperViz | Awareness] - Set local state", state);
      if (state === null) {
        if (!this.states.has(this.clientID)) return;
        this.states.delete(this.clientID);
        (_a = this.room) == null ? void 0 : _a.presence.update({
          ...this.states.get(this.clientID),
          [this.YJS_STATE]: null,
          origin: "set local state null"
        });
        const update2 = { added: [], updated: [], removed: [this.clientID] };
        this.emit("change", [update2, "local" /* LOCAL */]);
        return;
      }
      const update = { added: [], updated: [], removed: [] };
      let oldState = this.states.get(this.clientID);
      if (oldState) {
        update.updated.push(this.clientID);
      } else {
        oldState = { [this.YJS_STATE]: {}, clientID: this.clientID };
        update.added.push(this.clientID);
      }
      const newState = { ...oldState, [this.YJS_STATE]: state, origin: "set local state" };
      this.states.set(this.clientID, newState);
      (_b = this.room) == null ? void 0 : _b.presence.update(newState);
      this.emit("change", [update, "local" /* LOCAL */]);
    };
    // #region events callbacks
    this.onLeave = (event) => {
      const clientID = this.participantIdToClientId.get(event.id);
      this.logger.log("[SuperViz | Awareness] - Participant left", { participant: event, clientID });
      if (!clientID) return;
      if (clientID === this.clientID) {
        const ids = Array.from(this.states.keys());
        this.removeAwarenessStates(ids, "presence" /* PRESENCE */);
        return;
      }
      const update = { added: [], updated: [], removed: [clientID] };
      this.participantIdToClientId.delete(event.id);
      this.states.delete(clientID);
      this.emit("update", [update, "presence" /* PRESENCE */]);
      this.emit("change", [update, "presence" /* PRESENCE */]);
    };
    this.onUpdate = (event) => {
      if (event.id === this.participantId) return;
      this.logger.log("[SuperViz | Awareness] - Participant updated", event);
      if (event.data[this.YJS_STATE] === null) {
        this.removeAwarenessStates([event.data.clientID], "presence" /* PRESENCE */);
        return;
      }
      const added = [];
      const updated = [];
      let clientID = this.participantIdToClientId.get(event.id);
      if (clientID) {
        updated.push(event.data.clientID);
      } else {
        this.participantIdToClientId.set(event.id, event.data.clientID);
        clientID = event.data.clientID;
        added.push(event.data.clientID);
      }
      this.states.set(clientID, {
        ...this.states.get(clientID) || {},
        ...event.data
      });
      this.emit("update", [{ added, updated, removed: [] }, "presence" /* PRESENCE */]);
      this.emit("change", [{ added, updated, removed: [] }, "presence" /* PRESENCE */]);
    };
    this.onVisibilityChange = () => {
      if (document.visibilityState === "visible") {
        clearTimeout(this.visibilityTimeout);
        if (this.previousState) {
          this.setLocalState(this.previousState);
          this.previousState = null;
        }
        return;
      }
      this.visibilityTimeout = setTimeout(() => {
        this.previousState = this.getLocalState();
        this.setLocalState(null);
      }, this.TIMEOUT_TIMER_MS);
    };
    this.clientID = this.doc.clientID;
  }
  /**
   * @function connect
   * @description Start the awareness service
   * @param {RealtimeRoom} room Main room in which it will propagate presence
   * @returns {void}
   */
  connect(participantId, room) {
    this.logger.log("[SuperViz | Awareness] - Connect awareness to room");
    this.participantId = participantId;
    this.room = room;
    this.addRoomListeners();
    this.addDocumentListeners();
    this.initializePresences();
  }
  // #region public methods
  /**
   * @function destroy
   * @description Destroy the awareness service and clean up all the listeners, states, etc
   * @returns {void}
   */
  destroy() {
    this.logger.log("[SuperViz | Awareness] - Destroy awareness");
    this.setLocalState(null);
    this.removeAwarenessStates(Array.from(this.states.keys()), "local" /* LOCAL */);
    clearTimeout(this.visibilityTimeout);
    this.visibilityTimeout = void 0;
    this.removeDocumentListeners();
    this.removeRoomListeners();
    this.room = null;
    this.states.clear();
    this.participantIdToClientId.clear();
    this.onLeave({ id: this.participantId });
  }
  /**
   * @function getLocalState
   * @description Get the local state of the current participant, stores in the field yjs_state
   * @returns
   */
  getLocalState() {
    var _a;
    this.logger.log(
      "[SuperViz | Awareness] - Get local state",
      (_a = this.states.get(this.clientID)) == null ? void 0 : _a[this.YJS_STATE]
    );
    const state = this.states.get(this.clientID);
    if (!state || !state[this.YJS_STATE]) return null;
    return state[this.YJS_STATE];
  }
  getStates() {
    const states = /* @__PURE__ */ new Map();
    this.states.forEach((state, clientID) => {
      states.set(clientID, state[this.YJS_STATE]);
    });
    this.logger.log("[SuperViz | Awareness] - Get states", states);
    return states;
  }
  setLocalStateField(field, value) {
    this.logger.log("[SuperViz | Awareness] - Set local state field", { field, value });
    const state = this.getLocalState();
    this.setLocalState({
      ...state || {},
      [field]: value
    });
  }
  // #region events listeners
  addRoomListeners() {
    this.room.presence.on("presence.update", this.onUpdate);
    this.room.presence.on("presence.leave", this.onLeave);
  }
  removeRoomListeners() {
    this.room.presence.off("presence.update");
    this.room.presence.off("presence.leave");
  }
  addDocumentListeners() {
    document.addEventListener("visibilitychange", this.onVisibilityChange);
  }
  removeDocumentListeners() {
    document.removeEventListener("visibilitychange", this.onVisibilityChange);
  }
  // #region private methods
  initializePresences() {
    this.room.presence.get((presences) => {
      const added = [];
      presences.forEach((presence) => {
        const { clientID: clientID2 } = presence.data;
        if (!clientID2) return;
        added.push(clientID2);
        this.participantIdToClientId.set(presence.id, clientID2);
        this.states.set(clientID2, {
          ...presence.data
        });
      });
      const { clientID } = this.doc;
      this.room.presence.update({
        clientID,
        [this.YJS_STATE]: {},
        ...this.states.get(clientID) || {},
        origin: "on connect"
      });
      const update = { added, updated: [], removed: [] };
      this.emit("change", [update, "presence" /* PRESENCE */]);
    });
  }
  removeAwarenessStates(removed, origin) {
    this.logger.log("[SuperViz | Awareness] - Remove awareness states", { removed, origin });
    const update = { added: [], updated: [], removed: [] };
    removed.forEach((clientID) => {
      if (!this.states.get(clientID)) return;
      this.states.delete(clientID);
      update.removed.push(clientID);
    });
    this.emit("change", [update, origin]);
  }
};

// src/services/logger/index.ts
var import_debug = __toESM(require_src(), 1);
var Logger = class {
  constructor(scope, prefix) {
    this.prefix = prefix;
    this.debug = (0, import_debug.default)(scope);
  }
  log(formatter, ...args) {
    this.debug(`${this.prefix} -`, formatter, ...args);
  }
};

// src/utils/getUpdatesHistory.ts
async function getUpdatesHistory(channelName, apiKey) {
  try {
    const response = await fetch(`https://io.superviz.com/yjs/${channelName}:yjs-provider`, {
      headers: {
        "sv-api-key": apiKey
      }
    });
    if (response.status !== 200) {
      return [];
    }
    const updates = await response.json();
    return updates.events.map((data) => new Uint8Array(data.update.data));
  } catch (err) {
    return [];
  }
}

// src/provider/types.ts
var storeType = {
  GLOBAL: "global-store"
};

// src/provider/index.ts
var SuperVizYjsProvider = class extends ObservableV2 {
  constructor(doc, opts) {
    super();
    this.doc = doc;
    this.name = "yjsProvider";
    this._synced = false;
    this.state = "provider.disconnected" /* DISCONNECTED */;
    this.isAttached = false;
    this.localParticipant = null;
    this.room = null;
    this.ioc = null;
    /**
     * @function fetch
     * @description Request and apply all document updates from server
     * @returns {void}
     */
    this.fetch = async () => {
      this.logger.log("Fetching the document");
      this._synced = false;
      const updates = await getUpdatesHistory(this.roomId, this.ioc["client"]["apiKey"]);
      this._synced = true;
      if (!updates.length) return;
      Y.applyUpdateV2(this.doc, Y.mergeUpdatesV2(updates));
    };
    /**
     * @function updateDocument
     * @description Apply an update to the local document
     * @param {Uint8Array} update The update to apply
     * @returns {void}
     */
    this.updateDocument = (update) => {
      this.logger.log("Applying remote update", update);
      Y.applyUpdateV2(this.doc, update, this);
    };
    // #region events callbacks
    /**
     * @function onDocUpdate
     * @description Broadcast the local document update to all participants
     * @param {Uint8Array} update The update to broadcast
     * @returns {void}
     */
    this.onDocUpdate = (update) => {
      this.logger.log("Local document update", update);
      this.beforeSendRealtimeMessage("provider.update" /* UPDATE */, { update });
      this.room.emit("provider.update" /* UPDATE */, { update });
    };
    /**
     * @function onLocalJoinRoom
     * @description Start the awareness service, subscribe to real-time events and fetch the document
     * @param {PresenceEvent} event
     * @returns {void}
     */
    this.onLocalJoinRoom = (event) => {
      var _a;
      if (this.state === "provider.connected" /* CONNECTED */ || event.id !== this.localParticipant.id) return;
      this.logger.log("Joined the room", event);
      (_a = this.awareness) == null ? void 0 : _a.connect(this.localParticipant.id, this.room);
      this.changeState("provider.connected" /* CONNECTED */);
      this.emit("connect", []);
      const update = Y.encodeStateAsUpdateV2(this.doc);
      this.beforeSendRealtimeMessage("provider.update" /* UPDATE */, { update });
      this.room.emit("provider.update", { update });
      this.fetch();
    };
    /**
     * @function onConnectionChange
     * @description Handle disconnection from the room
     * @param {ConnectionState} msg The connection state
     * @returns {void}
     */
    this.onConnectionChange = (msg) => {
      if (msg.state === "DISCONNECTED") {
        this.logger.log("Disconnected from the room");
        this.emit("disconnect", []);
      }
    };
    /**
     * @function onRemoteDocUpdate
     * @description Apply changes to a remote document to the local document
     * @param {SocketEvent<DocUpdate>} event The event containing the update
     * @returns {void}
     */
    this.onRemoteDocUpdate = (event) => {
      var _a;
      if (((_a = event.presence) == null ? void 0 : _a.id) === this.localParticipant.id) return;
      this.logger.log("Received remote document update", event);
      const update = new Uint8Array(event.data.update);
      this.onReceiveRealtimeMessage("provider.update" /* UPDATE */, { update });
      this.updateDocument(update);
    };
    // #region Emitting messages
    /**
     * @function onReceiveRealtimeMessage
     * @description Emit a message received from the room
     * @param {string} name The message name
     * @param {any} data The message data
     * @returns {void}
     */
    this.onReceiveRealtimeMessage = (name, data) => {
      this.logger.log("Received message from room", { name, data });
      this.emit("message", [{ name, data }]);
    };
    /**
     * @function beforeSendRealtimeMessage
     * @description Emit a message sent to the room
     * @param {string} name The message name
     * @param {any} data The message data
     * @returns {void}
     */
    this.beforeSendRealtimeMessage = (name, data) => {
      this.logger.log("Sending message to room", { name, data });
      this.emit("outgoingMessage", [{ name, data }]);
    };
    this.document = doc;
    this.logger = new Logger("SuperVizYjsProvider", "[SuperViz | YjsProvider] - ");
    if ((opts == null ? void 0 : opts.awareness) === false) return;
    this.awareness = new Awareness(this.doc, this.logger);
  }
  // #region Public methods
  /**
   * @function attach
   */
  attach(params) {
    if (Object.values(params).includes(null) || Object.values(params).includes(void 0)) {
      const message = `${this.name} @ attach - params are required`;
      this.logger.log(message);
      throw new Error(message);
    }
    const { useStore, ioc, config } = params;
    const { isDomainWhitelisted, hasJoinedRoom, localParticipant } = useStore(storeType.GLOBAL);
    if (!isDomainWhitelisted.value) {
      const message = `Component ${this.name} can't be used because this website's domain is not whitelisted. Please add your domain in https://dashboard.superviz.com/developer`;
      this.logger.log(message);
      return;
    }
    if (!hasJoinedRoom.value) {
      hasJoinedRoom.subscribe(() => this.attach(params));
      return;
    }
    this.roomId = config.roomId;
    this.ioc = ioc;
    this.localParticipant = localParticipant.value;
    this.isAttached = true;
    this.connect();
    this.emit("mount", []);
  }
  detach() {
    if (!this.isAttached) {
      this.logger.log(`${this.name} @ detach - component is not attached`);
      return;
    }
    this.logger.log("detached");
    this.destroyProvider();
    super.destroy();
    this.emit("unmount" /* UNMOUNT */, []);
  }
  /**
   * @function connect
   * @description Connect to the room. With this, start to send and
   * receive awareness and document updates. This method is called
   * automatically when a SuperVizYjsProvider instance is created,
   * unless opted-out by passing "connect: false" in the constructor
   * options. Can't connect if it is already connected
   * @emits state
   * @returns {void}
   */
  connect() {
    if (this.state !== "provider.disconnected" /* DISCONNECTED */) return;
    this.logger.log("Connecting to the room");
    this.changeState("provider.connecting" /* CONNECTING */);
    this.doc.on("updateV2", this.onDocUpdate);
    this.startRealtime();
  }
  /**
   * @function destroy
   * @description Disconnect from the room and reset the instance state.
   * @emits state @returns {void}
   */
  destroyProvider() {
    var _a;
    if (this.state === "provider.disconnected" /* DISCONNECTED */) {
      this.logger.log("Can't destroy. Provider disconnected");
      return;
    }
    this.logger.log("Destroying the provider");
    this.emit("destroy", []);
    this._synced = false;
    (_a = this.awareness) == null ? void 0 : _a.destroy();
    this.doc.off("updateV2", this.onDocUpdate);
    this.removeRoomListeners();
    if (this.room) {
      this.room.disconnect();
      this.room = null;
    }
    this.changeState("provider.disconnected" /* DISCONNECTED */);
  }
  /**
   * @type {boolean}
   * @description Whether the user is connected to the room or not
   */
  get synced() {
    return this._synced;
  }
  // #region Private methods
  /**
   * @function createRoom
   * @description Create a new room and set the realtime and room instances
   * @returns {void}
   */
  createRoom() {
    this.logger.log("Creating room");
    this.room = this.ioc.createRoom("yjs-provider");
  }
  /**
   * @function addRoomListeners
   * @description Add listeners to the room instance
   * @returns {void}
   */
  addRoomListeners() {
    this.logger.log("Adding room listeners");
    this.room.on("provider.update" /* UPDATE */, this.onRemoteDocUpdate);
    this.room.presence.on("presence.joined-room", this.onLocalJoinRoom);
    this.ioc.client.connection.on(this.onConnectionChange);
  }
  /**
   * @function removeRoomListeners
   * @description Remove listeners from the room instance
   * @returns {void}
   */
  removeRoomListeners() {
    var _a;
    this.logger.log("Removing room listeners");
    if (this.room) {
      this.room.off("provider.update" /* UPDATE */, this.onRemoteDocUpdate);
      this.room.presence.off("presence.joined-room");
    }
    (_a = this.ioc) == null ? void 0 : _a.client.connection.off();
  }
  /**
   * @function startRealtime
   * @description Start the real-time connection with the room
   * @returns {void}
   */
  startRealtime() {
    this.createRoom();
    this.addRoomListeners();
  }
  /**
   * @function changeState
   * @description Change the state of the provider
   * @param {ProviderState | `${ProviderState}`} state The new state
   * @returns {void}
   */
  changeState(state) {
    this.logger.log("Changing state", state);
    this.emit("state", [state]);
    this.state = state;
  }
};

// src/index.ts
if (typeof window !== "undefined") {
  window.SuperVizYjsProvider = SuperVizYjsProvider;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SuperVizYjsProvider
});
