// @superviz/react-sdk/dist/annotation-filter-E7SCYW2G-C-0oC5Cr.js
import { y as r, e as l } from "./index-CtPfqlcy.js";
import { u as n } from "./chunk-K5TMEGOR-DaSvC8Mu.js";
import "./chunk-3MEZQQ2J-CHe_tQrN.js";
import { l as c } from "./chunk-SDMTFDEM-n5DHnLs2.js";
import { n as m } from "./chunk-6U6BYUHV-TZ4JwLT4.js";
import { D as a } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { V as p, a as d, t as f } from "./chunk-WG46WHJF-JdNs68CV.js";
r();
var i = a(d), v = [i.styles, c], e = [{ label: "All comments" }, { label: "Resolved comments" }], s = class extends i {
  constructor() {
    super(), this.selectClick = () => {
      this.caret = this.caret === "down" ? "up" : "down";
    }, this.dropdownOptionsHandler = ({ detail: t }) => {
      this.emitEvent("select", { filter: t }), this.selectClick();
    }, this.caret = "down";
  }
  firstUpdated(t) {
    super.firstUpdated(t), this.updateComplete.then(() => {
      m.call(this, ["comments"]);
    });
  }
  render() {
    let t = this.filter === "All comments" ? e[0].label : e[1].label;
    e[0].active = this.filter === "All comments", e[1].active = this.filter === "Resolved comments";
    let o = { text: !0, "text-bold": !0, "select-content": !0, "comments__filter__selected-label": !0, "sv-gray-500": this.caret === "down", "sv-gray-700": this.caret === "up" };
    return p`
      <div class="comments__filter-container">
        <div class="comments__filter">
          <superviz-dropdown
            options=${JSON.stringify(e)}
            position="bottom-left"
            right-offset="100px"
            @click=${this.selectClick}
            @selected=${this.dropdownOptionsHandler}
            @close=${this.selectClick}
            classesPrefix="comments__dropdown"
            parentComponent="comments"
          >
            <div class="comments__filter__toggle-button" slot="dropdown">
              <span class=${n(o)}>${t}</span>
              <div class="comments__filter__icon">
                <superviz-icon name=${this.caret} size="xs"></superviz-icon>
              </div>
            </div>
          </superviz-dropdown>
        </div>
      </div>
    `;
  }
};
s.styles = v, s.properties = { filter: { type: String }, caret: { type: String } }, s = l([f("superviz-comments-annotation-filter")], s);
export {
  s as CommentsAnnotationFilter
};
