// @superviz/react-sdk/dist/chunk-K5TMEGOR-DaSvC8Mu.js
import { a as o, i as h, E as d } from "./chunk-3MEZQQ2J-CHe_tQrN.js";
import { E as u } from "./chunk-WG46WHJF-JdNs68CV.js";
import { y as l } from "./index-CtPfqlcy.js";
l();
var c = o(class extends h {
  constructor(e) {
    var s;
    if (super(e), e.type !== d.ATTRIBUTE || e.name !== "class" || ((s = e.strings) == null ? void 0 : s.length) > 2) throw Error("`classMap()` can only be used in the `class` attribute and must be the only part in the attribute.");
  }
  render(e) {
    return " " + Object.keys(e).filter((s) => e[s]).join(" ") + " ";
  }
  update(e, [s]) {
    var i, n;
    if (this.st === void 0) {
      this.st = /* @__PURE__ */ new Set(), e.strings !== void 0 && (this.nt = new Set(e.strings.join(" ").split(/\s/).filter((t) => t !== "")));
      for (let t in s) s[t] && !((i = this.nt) != null && i.has(t)) && this.st.add(t);
      return this.render(s);
    }
    let r = e.element.classList;
    for (let t of this.st) t in s || (r.remove(t), this.st.delete(t));
    for (let t in s) {
      let a = !!s[t];
      a === this.st.has(t) || (n = this.nt) != null && n.has(t) || (a ? (r.add(t), this.st.add(t)) : (r.remove(t), this.st.delete(t)));
    }
    return u;
  }
});
l();
/*! Bundled license information:

lit-html/directives/class-map.js:
  (**
   * @license
   * Copyright 2018 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)
*/
export {
  c as u
};
