// @superviz/react-sdk/dist/comments-MGTDGIFZ-B7MhYZyz.js
import { y as d, e as m } from "./index-CtPfqlcy.js";
import { u as f } from "./chunk-K5TMEGOR-DaSvC8Mu.js";
import "./chunk-3MEZQQ2J-CHe_tQrN.js";
import { i as u, x as v } from "./chunk-SDMTFDEM-n5DHnLs2.js";
import { n as y } from "./chunk-6U6BYUHV-TZ4JwLT4.js";
import { D as g } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { V as n, a as b, t as w } from "./chunk-WG46WHJF-JdNs68CV.js";
d();
d();
var p;
function h(t) {
  let e = t.querySelector("#superviz-comments");
  if (e && !p) {
    let r = { childList: !0, attributes: !0, characterData: !0, subtree: !0 };
    p = new MutationObserver((s) => {
      s.forEach((i) => {
        i.removedNodes.length && i.removedNodes.forEach((o) => {
          o.id === "poweredby-footer" && z(t);
        });
      });
    }), p.observe(e, r);
  }
}
function z(t) {
  let e = document.createElement("div");
  e.id = "poweredby-footer", e.className = "footer";
  let r = document.createElement("div");
  r.className = "powered-by powered-by--horizontal";
  let s = document.createElement("a");
  s.href = "https://superviz.com/", s.target = "_blank", s.className = "link";
  let i = document.createElement("div");
  i.textContent = "Powered by";
  let o = document.createElement("img");
  o.width = 48, o.height = 8.86, o.src = "https://production.cdn.superviz.com/static/superviz-gray-logo.svg", r.appendChild(s), s.appendChild(i), i.appendChild(o), e.appendChild(r);
  let l = t.getElementById("superviz-comments");
  l && l.appendChild(e), h(t);
}
var c = g(b), O = [c.styles, u, v], a = class extends c {
  constructor() {
    super(), this.annotations = [], this.annotationFilter = "All comments", this.waterMarkState = !1, this.participantsList = [], this.side = "left";
  }
  firstUpdated(t) {
    super.firstUpdated(t), this.updateComplete.then(() => {
      y.call(this, ["comments"]);
    });
  }
  updated(t) {
    super.updated(t), this.updateComplete.then(() => {
      this.waterMarkState && h(this.shadowRoot), t.has("offset") && this.applyOffset();
    });
  }
  participantsListed(t) {
    this.participantsList = t;
  }
  updateAnnotations(t) {
    this.annotations = t;
  }
  close() {
    this.emitEvent("close-threads", {});
  }
  waterMarkStatus(t) {
    this.waterMarkState = t;
  }
  setFilter({ detail: t }) {
    let { filter: { label: e } } = t;
    this.annotationFilter = e;
  }
  getOffset(t) {
    return t == null || t < 0 ? "10px" : `${t}px`;
  }
  applyOffset() {
    let t = this.shadowRoot.querySelector(".superviz-comments");
    if (!t) return;
    let { left: e, right: r, top: s, bottom: i } = this.offset;
    t.style.setProperty("--offset-top", this.getOffset(s)), t.style.setProperty("--offset-bottom", this.getOffset(i)), t.style.setProperty("--offset-right", this.getOffset(r)), t.style.setProperty("--offset-left", this.getOffset(e));
  }
  get poweredBy() {
    return this.waterMarkState ? n` <div id="poweredby-footer" class="footer">
      <div class="powered-by powered-by--horizontal">
        <a href="https://superviz.com/" target="_blank" class="link">
          <div class="">
            Powered by
            <img
              width="48px"
              height="8.86px"
              src="https://production.cdn.superviz.com/static/superviz-gray-logo.svg"
            />
          </div>
        </a>
      </div>
    </div>` : n``;
  }
  render() {
    let t = { "superviz-comments": !0, "threads-on-left-side": this.side === "left", "threads-on-right-side": this.side === "right", "hide-at-right": this.side === "right" && !this.open, "hide-at-left": this.side === "left" && !this.open };
    return n`
      <div id="superviz-comments" class=${f(t)}>
        <div class="header">
          <superviz-comments-topbar
            @close-threads=${this.close}
            side=${this.side.split(":")[0]}
          ></superviz-comments-topbar>
        </div>
        <superviz-comments-annotation-filter
          filter=${this.annotationFilter}
          @select=${this.setFilter}
        >
        </superviz-comments-annotation-filter>
        <superviz-comments-content
          annotations=${JSON.stringify(this.annotations)}
          annotationFilter=${this.annotationFilter}
          participantsList=${JSON.stringify(this.participantsList)}
          class="content"
        ></superviz-comments-content>
        ${this.poweredBy}
      </div>
    `;
  }
};
a.styles = O, a.properties = { open: { type: Boolean }, annotations: { type: Object }, annotationFilter: { type: String }, waterMarkState: { type: Boolean }, side: { type: String }, participantsList: { type: Object }, offset: { type: Object } }, a = m([w("superviz-comments")], a);
export {
  a as Comments
};
