// @superviz/react-sdk/dist/delete-comment-modal-JX3K2H44-XQyqkSFi.js
import { D as i } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { V as e, a as d, t as l } from "./chunk-WG46WHJF-JdNs68CV.js";
import { y as m, e as a } from "./index-CtPfqlcy.js";
m();
var s = i(d), p = [s.styles], t = class extends s {
  constructor() {
    super(...arguments), this.emitEventOpenModal = () => {
      window.document.body.dispatchEvent(new CustomEvent("superviz-modal--open", { detail: { title: "DELETE COMMENT", body: e`<span class="text text-big sv-gray-600"
            >Are you sure you want to delete this comment? <br />
            This action cannot be undone</span
          >`, confirmLabel: "DELETE", confirm: !0, cancel: !0 }, bubbles: !0, composed: !0 }));
    }, this.toggle = () => {
      this.open = !this.open;
    };
  }
  firstUpdated() {
    window.document.body.addEventListener("superviz-modal--close", () => {
      this.open && this.emitEvent("close", {}, { bubbles: !1, composed: !1 });
    }), window.document.body.addEventListener("superviz-modal--confirm", () => {
      this.open && (this.emitEvent("confirm", {}, { bubbles: !1, composed: !1 }), this.emitEventCloseModal());
    });
  }
  updated(o) {
    o.has("open") && (this.open && setTimeout(() => this.emitEventOpenModal()), this.open || setTimeout(() => this.emitEventCloseModal()));
  }
  emitEventCloseModal() {
    this.open && (window.document.body.dispatchEvent(new CustomEvent("superviz-modal--close", { detail: { open: !0 }, bubbles: !0, composed: !0 })), this.emitEvent("close", {}, { bubbles: !1, composed: !1 }));
  }
  render() {
    let o = () => {
      if (this.useSlot) return e`<slot name="modal-handler" @click=${this.toggle}></slot>`;
    }, n = () => {
      if (this.open) return e` <superviz-modal></superviz-modal> `;
    };
    return e` ${o()} ${n()} `;
  }
};
t.styles = p, t.properties = { open: { type: Boolean }, useSlot: { type: Boolean } }, t = a([l("superviz-comments-delete-comments-modal")], t);
export {
  t as DeleteCommentModal
};
