// @superviz/react-sdk/dist/superviz-sdk-react.es39.js
import { u as j, j as b } from "./room-BEOW-846.js";
import { useState as k, useCallback as x, useEffect as c } from "react";
import { F as E } from "./index-CtPfqlcy.js";
function w({
  children: p,
  elementId: n,
  onMount: i,
  onUnmount: u,
  ...t
}) {
  var l;
  const [a, m] = k(null), { room: r, component: e, addComponent: T } = j(
    "presence"
  ), P = {
    mount: i,
    unmount: u
  }, d = x(
    (o) => {
      var s, f;
      (s = t.callbacks) != null && s.onGoToPresence && ((f = t.callbacks) == null || f.onGoToPresence(o));
    },
    [(l = t.callbacks) == null ? void 0 : l.onGoToPresence]
  );
  return c(() => {
    e && Object.entries(P).forEach(([o, s]) => {
      e.unsubscribe(o, s), s && e.subscribe(o, s);
    });
  }, [e, r, i, u]), c(() => {
    if (!r || a || !document.getElementById(n)) return;
    const o = new E(n, {
      callbacks: { onGoToPresence: d }
    });
    T(o), m(Date.now());
  }, [r, n]), c(() => {
    !e && a && m(null);
  }, [e]), p ?? /* @__PURE__ */ b.jsx(b.Fragment, {});
}
export {
  w as MousePointers
};
