// @superviz/react-sdk/dist/superviz-sdk-react.es20.js
import { useState as d, useRef as h, useEffect as l, useCallback as f, useMemo as T } from "react";
import { u as A } from "./room-BEOW-846.js";
import { _ as a, S as y } from "./index-CtPfqlcy.js";
function P(m) {
  const [n, p] = d(!1), [c, b] = d(!1), { component: t, room: E, ...i } = A("realtime"), s = h({
    [a.JOINED_ROOM]: [],
    [a.LEAVE]: [],
    [a.UPDATE]: []
  }), e = h(null);
  l(() => {
    if (t) {
      if (t.state === y.STARTED) {
        p(!0);
        return;
      }
      t.subscribe("realtime-component.state-changed", (r) => {
        p(r === y.STARTED);
      });
    }
  }, [t, s.current]), l(() => {
    if (!n && !e.current || !t) return;
    (async () => {
      if (e.current = await t.connect(m), e.current.state === "CONNECTED") {
        b(!0);
        return;
      }
      e.current.subscribe("realtime-channel.state-changed", (u) => {
        b(u === "CONNECTED");
      });
    })();
  }, [n]), l(() => {
    e.current && !E && (e.current.disconnect(), e.current = null, b(!1), p(!1));
  }, [t, E]), l(() => {
    c && Object.keys(s.current).length && (Object.entries(s.current).forEach(([r, u]) => {
      u.forEach((o) => {
        e.current.participant.subscribe(r, o);
      });
    }), s.current = {
      [a.JOINED_ROOM]: [],
      [a.LEAVE]: [],
      [a.UPDATE]: []
    });
  }, [c]);
  const O = f(
    (r, u) => {
      if (!e.current) {
        const o = s.current[r] || [];
        s.current[r] = o.includes(u) ? o : [...o, u];
        return;
      }
      e.current.participant.subscribe(r, u);
    },
    [t, i, s.current]
  ), R = f(
    (r) => {
      !n || !c || e.current.participant.update(r);
    },
    [t, i, n]
  ), C = f(
    (r) => {
      !n || !c || e.current.participant.unsubscribe(r);
    },
    [t, i, n]
  ), S = f(async () => n ? c ? await e.current.participant.getAll() : (console.warn(`[SuperViz] Channel ${m} is not ready - try again later`), Promise.resolve([])) : (console.warn("[SuperViz] Realtime component is not ready - try again later"), Promise.resolve([])), [t, i, n]);
  return T(() => ({
    isReady: c && n,
    update: R,
    subscribe: O,
    unsubscribe: C,
    getAll: S
  }), [t, i, n, c]);
}
export {
  P as useRealtimeParticipant
};
