// @superviz/react-sdk/dist/superviz-sdk-react.es21.js
import { useState as h, useRef as y, useEffect as a, useCallback as l, useMemo as T } from "react";
import { u as g } from "./room-BEOW-846.js";
import { S as d } from "./index-CtPfqlcy.js";
function N(m) {
  const [n, f] = h(!1), [u, b] = h(!1), { component: t, room: p, ...i } = g("realtime"), c = y({}), r = y(null);
  a(() => {
    if (t) {
      if (t.state === d.STARTED) {
        f(!0);
        return;
      }
      t.subscribe("realtime-component.state-changed", (e) => {
        f(e === d.STARTED);
      });
    }
  }, [t, c.current]), a(() => {
    (async () => {
      if (!(!n && !r.current || !t)) {
        if (r.current = await t.connect(m), r.current.state === "CONNECTED") {
          b(!0);
          return;
        }
        r.current.subscribe("realtime-channel.state-changed", (e) => {
          b(e === "CONNECTED");
        });
      }
    })();
  }, [n]), a(() => {
    r.current && !p && (r.current.disconnect(), r.current = null, b(!1), f(!1));
  }, [t, p]), a(() => {
    u && Object.keys(c.current).length && (Object.entries(c.current).forEach(([e, s]) => {
      s.forEach((o) => {
        r.current.subscribe(e, o);
      });
    }), c.current = {});
  }, [u]);
  const E = l(
    (e, s) => {
      if (!r.current) {
        const o = c.current[e] || [];
        c.current[e] = o.includes(s) ? o : [...o, s];
        return;
      }
      r.current.subscribe(e, s);
    },
    [t, i, c.current]
  ), S = l(
    (e, s) => {
      !n || !u || r.current.publish(e, s);
    },
    [t, i, n]
  ), R = l(
    (e, s) => {
      !n || !u || r.current.unsubscribe(e, s);
    },
    [t, i, n]
  ), C = l(
    async (e) => n ? u ? await r.current.fetchHistory(e) : (console.warn(`[SuperViz] Channel ${m} is not ready - try again later`), Promise.resolve(null)) : (console.warn("[SuperViz] Realtime component is not ready - try again later"), Promise.resolve(null)),
    [t, i, n]
  );
  return T(() => ({
    isReady: u && n,
    publish: S,
    subscribe: E,
    unsubscribe: R,
    fetchHistory: C
  }), [t, i, n, u]);
}
export {
  N as useRealtime
};
