// @superviz/react-sdk/dist/superviz-sdk-react.es38.js
import { u as E, j as o } from "./room-BEOW-846.js";
import { v as F } from "./index-ad0DlQ-s.js";
import { useState as I, useEffect as a, useRef as j, useCallback as P } from "react";
function S({
  matterportSdkInstance: t,
  children: m,
  ...u
}) {
  const { room: r, component: e, addComponent: f, removeComponent: i } = E("presence3dMatterport"), [p, s] = I(null);
  return a(() => {
    if (!r || !t)
      return;
    const l = new F(
      t,
      u
    );
    f(l), s(Date.now());
  }, [r, t]), a(() => {
    !e && p && s(null);
  }, [e]), a(() => {
    (!t || !r) && e && i(e);
  }, [t, r]), m ?? /* @__PURE__ */ o.jsx(o.Fragment, {});
}
function R({
  isAvatarsEnabled: t,
  isLaserEnabled: m,
  isNameEnabled: u,
  avatarConfig: r,
  bundleUrl: e,
  matterportKey: f,
  onMpSdkLoaded: i,
  ...p
}) {
  const { room: s, removeComponent: l, component: z } = E("presence3dMatterport"), c = j(null), [w, d] = I(null), h = j("");
  a(() => {
    c.current && C();
  }, [e, c.current]), a(() => {
    var n;
    !s && w && (d(null), (n = c.current) == null || n.remove());
  }, [s]);
  const C = P(async () => {
    if (h.current === e)
      return;
    h.current = e, w && (d(null), l(z));
    const n = document.getElementById("showcase"), v = n.contentWindow;
    if (n.setAttribute("src", e), !v) {
      console.error("[SuperViz] Matterport Showcase iframe not found");
      return;
    }
    const g = await new Promise((y) => {
      const x = async () => {
        const M = await v.MP_SDK.connect(v, f);
        i && i({
          matterportSdkInstance: M,
          showcaseWindow: c.current
        }), y(M), n.removeEventListener("load", x);
      };
      n.addEventListener("load", x);
    });
    d(g);
  }, [e]);
  return /* @__PURE__ */ o.jsxs(o.Fragment, { children: [
    /* @__PURE__ */ o.jsx(
      S,
      {
        matterportSdkInstance: w,
        isAvatarsEnabled: t,
        isLaserEnabled: m,
        isNameEnabled: u,
        avatarConfig: r
      }
    ),
    /* @__PURE__ */ o.jsx("iframe", { ref: c, id: "showcase", ...p })
  ] });
}
export {
  R as MatterportIframe,
  S as MatterportPresence
};
