// @superviz/react-sdk/dist/superviz-sdk-react.es34.js
import { u as g, j as d } from "./room-BEOW-846.js";
import { R as $ } from "./index-BqPoy8ey.js";
import { useState as S, useEffect as h } from "react";
function U({ viewer: i, children: l, ...m }) {
  const { room: r, component: u, addComponent: p } = g("presence3dAutodesk"), [s, a] = S(null);
  return h(() => {
    if (!r || s || !i) return;
    const c = new $(i, m);
    p(c), a(Date.now());
  }, [r, i]), h(() => {
    !u && s && a(null);
  }, [u]), l ?? /* @__PURE__ */ d.jsx(d.Fragment, {});
}
function G({
  modelUrn: i,
  clientId: l,
  clientSecret: m,
  authUrl: r,
  initializeOptions: u,
  data: p,
  onDocumentLoadError: s,
  onDocumentLoadSuccess: a,
  onViewerInitialized: c,
  isAvatarsEnabled: v,
  isLaserEnabled: x,
  isNameEnabled: z,
  isMouseEnabled: I,
  avatarConfig: V,
  ...b
}) {
  const [f, A] = S(null), { hasJoinedRoom: y } = g("presence3dAutodesk"), j = r || "https://developer.api.autodesk.com/authentication/v2/token", R = btoa(`${l}:${m}`), F = {
    // eslint-disable-next-line camelcase
    grant_type: "client_credentials",
    scope: "data:read bucket:read",
    ...p
  };
  h(() => {
    if (y) {
      T();
      return;
    }
    f && (f.finish(), A(null));
  }, [y]);
  async function T() {
    let t = null;
    if (!window.Autodesk) {
      console.error("[Superviz] Autodesk not found");
      return;
    }
    await fetch(j, {
      method: "POST",
      headers: {
        "Content-Type": "application/x-www-form-urlencoded",
        Authorization: `Basic ${R}`
      },
      body: new URLSearchParams(F).toString()
    }).then((e) => e.json()).then((e) => {
      const o = `urn:${btoa(i)}`, w = {
        env: "AutodeskProduction2",
        api: "streamingV2",
        accessToken: e.access_token,
        ...u
      };
      window.Autodesk.Viewing.Initializer(w, async () => {
        const k = document.getElementById("autodesk-content");
        if (!k) {
          console.error("[Superviz] Autodesk container not found");
          return;
        }
        t = new window.Autodesk.Viewing.GuiViewer3D(k), await t.start(), c && c({
          viewer: t,
          container: k
        }), window.Autodesk.Viewing.Document.load(o, D, P);
      });
    });
    function D(e) {
      a && a(e);
      const n = e.getRoot().getDefaultGeometry();
      if (!n || !t) return;
      const o = {
        applyScaling: "meters"
      };
      t.loadDocumentNode(e, n, o).then(() => {
        t && A(t);
      }).catch((w) => {
        console.error("[SuperViz] Failed to load document node", w);
      });
    }
    function P(e, n, o) {
      s && s(e, n, o), console.error("[SuperViz] Failed to load document", e, n, o);
    }
  }
  return /* @__PURE__ */ d.jsx(
    U,
    {
      viewer: f,
      isAvatarsEnabled: v,
      isLaserEnabled: x,
      isNameEnabled: z,
      isMouseEnabled: I,
      avatarConfig: V,
      children: /* @__PURE__ */ d.jsx("div", { id: "autodesk-content", ...b })
    }
  );
}
export {
  U as AutodeskPresence,
  G as AutodeskViewer
};
