// @superviz/react-sdk/dist/who-is-online-4MQU4252-eujsIBCf.cjs
"use strict";Object.defineProperty(exports,Symbol.toStringTag,{value:"Module"});const m=require("./chunk-DX57ODX6-ClzWz6Mr.cjs"),p=require("./index-N3n6WSly.cjs"),c=require("./chunk-K5TMEGOR-CjzheANK.cjs");require("./chunk-3MEZQQ2J-Bai2OMk0.cjs");const g=require("./chunk-6U6BYUHV-h-2X1uK0.cjs"),_=require("./chunk-MVFOLHBC-C85clPX2.cjs"),n=require("./chunk-WG46WHJF-D0x9jAhS.cjs");p.y();p.y();p.y();p.y();var $=n.Ct`
  .who-is-online__participant-list {
    display: flex;
    align-items: center;
    gap: 4px;
    position: relative;
  }

  .who-is-online {
    display: flex;
    flex-direction: column;
    position: fixed;
    z-index: 99;
  }

  .who-is-online__presence-control-message__text {
    margin: 0;
  }

  .who-is-online__participant {
    border-radius: 50%;
    box-sizing: border-box;
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    user-select: none;
    cursor: pointer;
    transition: opacity 0.3s ease-in-out;
    border: 2px solid #878291;
    border-radius: 50%;
    max-width: 40px;
    flex: 1 0 40px;
  }

  .followed {
    border-style: dashed !important;
    animation: rotate 15s linear infinite;
  }

  .followed .who-is-online__participant__avatar {
    animation: nullifyRotate 15s linear infinite;
  }

  .private {
    opacity: 0.3;
  }

  @keyframes rotate {
    100% {
      transform: rotate(1turn);
    }
  }

  @keyframes nullifyRotate {
    100% {
      transform: rotate(-1turn);
    }
  }

  .who-is-online__participant.disable-dropdown {
    cursor: default;
  }

  .who-is-online__participant__avatar {
    width: 32px;
    height: 32px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: 'Open Sans';
    font-size: 14px;
    line-height: 14px;
    line-height: 14px;
    font-weight: bold;
    color: #26242a;
    object-fit: contain;
  }

  .superviz-who-is-online__excess {
    width: 32px;
    height: 32px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: Roboto;
    font-size: 14px;
    line-height: 16px;
    text-align: center;
    font-weight: bold;
    cursor: pointer;
    color: #aea9b8;
  }

  .excess_participants:hover,
  .excess_participants--open {
    background-color: #aea9b8 !important;
  }

  .excess_participants:hover > div,
  .excess_participants--open > div {
    color: #fff !important;
  }

  @media (max-width: 780px) {
    .who-is-online__participant {
      width: 32px;
      height: 32px;
    }

    .who-is-online__participant {
      flex: 1 0 32px;
      max-width: 32px;
    }

    .who-is-online__participant__avatar {
      width: 24px;
      height: 24px;
    }

    .who-is-online__participant-list {
      gap: 8px;
    }

    .superviz-who-is-online__excess {
      width: 24px;
      height: 24px;
      font-size: 12px;
      line-height: 12px;
    }
  }
`;p.y();var P=n.Ct`
  .dropdown {
    position: relative;
  }

  .who-is-online__extra-participant {
    display: flex;
    user-select: none;
    align-items: center;
    justify-items: start;
    gap: 4px;
    width: 100%;
    padding: 10px;
    border-radius: 2px;
    position: relative;
    cursor: pointer;
  }

  .who-is-online__extra-participant.disable-dropdown {
    cursor: default;
  }

  .who-is-online__extra-participant:hover,
  .who-is-online__extra-participant--selected {
    background-color: rgb(var(--sv-gray-200));
  }

  .who-is-online__participant {
    border-radius: 50%;
    box-sizing: border-box;
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 40px;
    height: 40px;
    border: 2px solid #878291;
    border-radius: 50%;
    max-width: 40px;
    flex: 1 0 40px;
  }

  .who-is-online__participant__avatar {
    width: 32px;
    height: 32px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: 'Open Sans';
    font-size: 14px;
    line-height: 14px;
    font-weight: bold;
    color: #26242a;
    object-fit: contain;
  }

  .dropdown-list {
    position: relative;
    display: flex;
    flex-direction: column;
    z-index: 100;
  }

  .dropdown-list > div {
    padding: 4px;
    width: 216px;
    box-sizing: border-box;
  }

  .who-is-online__extras-dropdown {
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
    opacity: 0;
    display: none;
    background: #fff;
    padding: 0;
    z-index: 1;
    transition: 0.2s;
    border-radius: 3px;
    max-height: 240px;
    overflow: auto;
  }

  .who-is-online__extras-dropdown superviz-dropdown:hover {
    z-index: 999;
    position: relative;
  }

  .who-is-online__extras__arrow-icon {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .menu--bottom {
    top: 4px;
    min-width: 103px;
    position: absolute;
    right: 0;
  }

  .menu--top {
    bottom: 44px;
    min-width: 103px;
    position: absolute;
    right: 0px;
  }

  .menu-open {
    display: block;
    opacity: 1;
  }

  .superviz-who-is-online-dropdown__tooltip {
    background-color: rgb(var(--sv-gray-600));
    padding: 8px;
    display: flex;
    flex-direction: column;
    gap: 8px;
    align-items: center;
    position: absolute;
    top: 52px;
    left: 50%;
    transform: translateX(-50%);
    border-radius: 2px;
    opacity: 0;
    cursor: default;
    display: none;
    transition: opacity 0.2s ease-in-out display 0s;
  }

  .superviz-who-is-online-dropdown__tooltip-arrow {
    width: 13px;
    height: 13px;
    position: absolute;
    top: 0px;
    left: 50%;
    transform: rotate(45deg) translateX(-50%);
    background-color: rgb(var(--sv-gray-600));
    border-top-left-radius: 3px;
  }

  .dropdown-content:hover + .superviz-who-is-online-dropdown__tooltip {
    opacity: 1;
    display: block;
  }

  .tooltip-content {
    margin: 0;
    font-family: roboto;
    white-space: nowrap;
    text-align: center;
    color: white;
    font-size: 14px;
  }

  .who-is-online__extras__username {
    font-size: 14px;
    line-height: 20px;
    font-family: 'Open sans';
    color: rgb(var(--sv-gray-600));
  }

  .icon {
    flex: 1;
    justify-content: flex-end;
    display: flex;
  }

  .hide-icon {
    display: none;
  }

  @media (max-width: 780px) {
    .sv-icon,
    .who-is-online__participant {
      width: 32px;
      height: 32px;
    }

    .who-is-online__participant {
      flex: 1 0 32px;
      max-width: 32px;
    }

    .who-is-online__participant__avatar {
      width: 24px;
      height: 24px;
    }

    .dropdown-list > div {
      width: 192px;
    }

    .menu--top {
      bottom: 36px;
    }

    .superviz-who-is-online-dropdown__tooltip {
      top: 44px;
    }
  }
`;p.y();var S=n.Ct`
  .who-is-online__controls-messages {
    position: absolute;
    right: 0;
    display: flex;
    flex-direction: column;
    gap: 5px;
    margin-top: 4px;
  }

  .who-is-online__presence-control-message {
    box-sizing: border-box;
    font-size: 12px;
    padding: 8px 10px;
    font-family: 'Roboto';
    border-radius: 6px;
    background-color: #fff;
    color: rgb(var(--sv-gray-700));
    border: 2px solid #e0e0e0;
    white-space: nowrap;
  }

  .who-is-online__pcm__text {
    margin: 0;
  }

  .who-is-online__presence-control-message span {
    margin-left: 3px;
    text-decoration: underline;
    cursor: pointer;
  }

  .left-side {
    left: 0;
  }

  .right-side {
    right: 0;
  }

  .left-side .who-is-online__presence-control-message {
    align-self: flex-start;
  }

  .right-side .who-is-online__presence-control-message {
    align-self: flex-end;
  }

  .bottom-side {
    bottom: auto;
    top: calc(100% + 5px);
  }

  .top-side {
    top: auto;
    bottom: calc(100% + 5px);
  }
`,x=_.D(n.C),z=[x.styles,$];exports.WhoIsOnline=class extends x{constructor(){super(),this.onClickOutDropdown=({detail:r})=>{this.open=r.open},this.toggleShowTooltip=()=>{this.showTooltip=!this.showTooltip},this.dropdownOptionsHandler=({detail:{label:r,participantId:l,source:a}})=>{switch(r){case"go to":this.handleGoTo(l);break;case"follow":this.handleLocalFollow(l,a);break;case"unfollow":this.handleLocalUnfollow();break;case"private mode":this.handlePrivate(l);break;case"leave private mode":this.handleCancelPrivate(l);break;case"everyone follows me":this.handleFollow(l,a);break;case"stop followers":this.handleStopFollow();break;case"gather all":this.handleGatherAll(l);break}},this.position="top: 20px; right: 40px;",this.showTooltip=!0,this.open=!1;let{localParticipant:e}=this.useStore("global-store"),{participants:i,following:o,extras:t,disablePresenceControls:s}=this.useStore("who-is-online-store");i.subscribe(r=>{this.participants=r}),o.subscribe(),s.subscribe(),t.subscribe(r=>{this.amountOfExtras=r.length}),e.subscribe(r=>{var l,a;let d=(l=r.activeComponents)==null?void 0:l.some(h=>h.toLowerCase().includes("presence"));this.localParticipantData={id:r.id,joinedPresence:(a=r.activeComponents)==null?void 0:a.some(h=>h.toLowerCase().includes("presence"))},this.disableDropdown=!d})}firstUpdated(e){super.firstUpdated(e),g.n.call(this,"who-is-online")}toggleOpen(){this.open=!this.open}dropdownPosition(e){if(this.participants.length===1)return"bottom-center";if(e===0)return"bottom-right";let i=this.participants.length>4,o=e+1===this.participants.length;return i||!o?"bottom-center":"bottom-left"}renderExtras(){var e;if(!this.amountOfExtras)return;let i={"who-is-online__participant":!0,excess_participants:!0,"excess_participants--open":this.open};return n.Vt`
      <superviz-who-is-online-dropdown
        @selected=${this.dropdownOptionsHandler}
        @clickout=${this.onClickOutDropdown}
        ?disableDropdown=${this.disableDropdown}
        ?showSeeMoreTooltip=${this.showTooltip}
        @toggle=${this.toggleOpen}
        @toggle-dropdown-state=${this.toggleShowTooltip}
        ?localParticipantJoinedPresence=${(e=this.localParticipantData)==null?void 0:e.joinedPresence}
        classesPrefix="who-is-online__controls"
        parentComponent="who-is-online"
        tooltipPrefix="who-is-online"
      >
        <div class=${c.u(i)} slot="dropdown">
          <div class="superviz-who-is-online__excess who-is-online__extras">
            +${this.amountOfExtras}
          </div>
        </div>
      </superviz-who-is-online-dropdown>
    `}getAvatar({color:e,imageUrl:i,firstLetter:o,letterColor:t}){return i?n.Vt` <img
        class="who-is-online__participant__avatar"
        style="background-color: ${e}"
        src=${i}
      />`:n.Vt`<div
      class="who-is-online__participant__avatar"
      style="background-color: ${e}; color: ${t}"
    >
      ${o}
    </div>`}cancelPrivate(){this.isPrivate=void 0,this.emitEvent("realtime.private-mode",{id:this.localParticipantData.id})}renderParticipants(){return this.participants.length?n.Vt`
      ${m.j(this.participants,e=>e.id,(e,i)=>{var o;let{avatar:t,id:s,name:r,tooltip:l,controls:a,disableDropdown:d,isLocalParticipant:h}=e,w=this.dropdownPosition(i),u=((o=this.following)==null?void 0:o.id)===s,b={"who-is-online__participant":!0,"disable-dropdown":d,followed:u||this.everyoneFollowsMe&&h,private:h&&this.isPrivate};return n.Vt`
            <superviz-dropdown
              options=${JSON.stringify(a)}
              returnData=${JSON.stringify({participantId:s,source:"participants"})}
              position=${w}
              name=${r}
              tooltipData=${JSON.stringify(l)}
              classesPrefix="who-is-online__controls"
              parentComponent="who-is-online"
              tooltipPrefix="who-is-online"
              ?disabled=${d}
              ?canShowTooltip=${this.showTooltip}
              @selected=${this.dropdownOptionsHandler}
              @toggle-dropdown-state=${this.toggleShowTooltip}
            >
              <div
                slot="dropdown"
                class=${c.u(b)}
                style="border-color: ${t.color}"
              >
                ${this.getAvatar(t)}
              </div>
            </superviz-dropdown>
          `})}
    `:n.Vt``}handleGoTo(e){this.emitEvent("realtime.go-to-participant",{id:e})}handleLocalFollow(e,i){let{following:o}=this.useStore("who-is-online-store"),t=this.useStore("who-is-online-store")[i].value,{id:s,name:r,avatar:{color:l}}=t.find(({id:a})=>a===e);this.everyoneFollowsMe&&this.handleStopFollow(),o.publish({name:r,id:s,color:l}),this.emitEvent("realtime.local-follow-participant",{id:s,source:i})}handleLocalUnfollow(){let{following:e}=this.useStore("who-is-online-store");e.publish(void 0),this.emitEvent("realtime.local-follow-participant",{id:void 0})}handlePrivate(e){this.everyoneFollowsMe&&this.handleStopFollow(),this.emitEvent("realtime.private-mode",{id:e,isPrivate:!0}),this.isPrivate=!0}handleCancelPrivate(e){this.emitEvent("realtime.private-mode",{id:e,isPrivate:!1}),this.isPrivate=!1}handleFollow(e,i){this.isPrivate&&this.cancelPrivate();let o=this.useStore("who-is-online-store")[i].value,{id:t,name:s,avatar:{color:r}}=o.find(({id:a})=>a===e);this.everyoneFollowsMe=!0;let{following:l}=this.useStore("who-is-online-store");l.publish(void 0),this.emitEvent("realtime.follow-participant",{id:t,name:s,color:r})}handleStopFollow(){this.everyoneFollowsMe=!1,this.emitEvent("realtime.follow-participant",void 0)}handleGatherAll(e){this.isPrivate&&this.cancelPrivate(),this.emitEvent("realtime.gather",{id:e})}updated(e){super.updated(e),this.updateComplete.then(()=>{let i=this.shadowRoot.querySelector(".who-is-online");if(!i)return;let o=this.position.includes("left")?"flex-start":"flex-end",t=`${this.position} align-items: ${o};`;i.setAttribute("style",t)})}render(){return n.Vt`<div class="who-is-online who-is-online">
      <div class="who-is-online__participant-list">
        ${this.renderParticipants()} ${this.renderExtras()}
      </div>
      <superviz-who-is-online-messages
        ?everyoneFollowsMe=${this.everyoneFollowsMe}
        ?isPrivate=${this.isPrivate}
        @stop-following=${this.handleLocalUnfollow}
        @cancel-private=${this.cancelPrivate}
        @stop-everyone-follows-me=${this.handleStopFollow}
      ></superviz-who-is-online-messages>
    </div> `}};exports.WhoIsOnline.styles=z,exports.WhoIsOnline.properties={position:{type:String},open:{type:Boolean},localParticipantColor:{type:String},isPrivate:{type:Boolean},everyoneFollowsMe:{type:Boolean},showTooltip:{type:Boolean}},exports.WhoIsOnline=p.F([n.ts("superviz-who-is-online")],exports.WhoIsOnline);p.y();p.y();p.y();var f=_.D(n.C),O=[f.styles,S];exports.WhoIsOnlineMessages=class extends f{constructor(){super(),this.repositionMessages=()=>{let{following:o,everyoneFollowsMe:t,isPrivate:s}=this;if(o||t||s){this.repositionInVerticalDirection(),this.repositionInHorizontalDirection(),this.animationFrame=window.requestAnimationFrame(this.repositionMessages);return}window.cancelAnimationFrame(this.animationFrame)};let{localParticipant:e}=this.useStore("global-store");e.subscribe(o=>{var t,s;this.participantColor=(s=(t=o.slot)==null?void 0:t.color)!=null?s:p.i.gray});let{following:i}=this.useStore("who-is-online-store");i.subscribe()}firstUpdated(e){super.firstUpdated(e),g.n.call(this,"who-is-online")}updated(e){super.updated(e),["following","everyoneFollowsMe","isPrivate"].some(i=>e.has(i))&&this.repositionMessages()}repositionInVerticalDirection(){let e=window.innerHeight/2,{top:i}=this.parentElement.getBoundingClientRect();if(i<e){this.verticalSide="bottom-side";return}this.verticalSide="top-side"}repositionInHorizontalDirection(){let e=window.innerWidth/2,{left:i}=this.parentElement.getBoundingClientRect();if(i<e){this.horizontalSide="left-side";return}this.horizontalSide="right-side"}stopFollowing(){this.emitEvent("stop-following",{})}cancelPrivate(){this.emitEvent("cancel-private",{})}stopEveryoneFollowsMe(){this.emitEvent("stop-everyone-follows-me",{})}followingMessage(){if(!this.following)return"";let{name:e,color:i}=this.following;return n.Vt`<div
      class="who-is-online__following-message who-is-online__presence-control-message who-is-online__pcm"
      style="border-color: ${i}"
    >
      <p class="who-is-online__presence-control-message__text who-is-online__pcm__text">
        Following: ${e}
        <span
          class="who-is-online__presence-control-message__cancel-action-button who-is-online__pcm__cancel-action-button"
          @click=${this.stopFollowing}
          >Stop</span
        >
      </p>
    </div>`}everyoneFollowsMeMessage(){return this.everyoneFollowsMe?n.Vt`<div
      class="who-is-online__follow-me-message who-is-online__presence-control-message who-is-online__pcm"
      style="border-color: ${this.participantColor}"
    >
      <p class="who-is-online__presence-control-message__text who-is-online__pcm__text">
        Everyone is following you
        <span
          class="who-is-online__presence-control-message__cancel-action-button who-is-online__pcm__cancel-action-button"
          @click=${this.stopEveryoneFollowsMe}
          >Stop</span
        >
      </p>
    </div>`:""}privateMessage(){return this.isPrivate?n.Vt`<div
      class="who-is-online__private-mode-message who-is-online__presence-control-message who-is-online__pcm"
      style="border-color: ${this.participantColor}"
    >
      <p class="who-is-online__presence-control-message__text who-is-online__pcm__text">
        You are in Private Mode
        <span
          class="who-is-online__presence-control-message__cancel-action-button who-is-online__pcm__cancel-action-button"
          @click=${this.cancelPrivate}
          >Cancel</span
        >
      </p>
    </div>`:""}render(){let e={"who-is-online__controls-messages":!0,[this.verticalSide]:!0,[this.horizontalSide]:!0};return n.Vt` <div class=${c.u(e)}>
      ${this.followingMessage()} ${this.everyoneFollowsMeMessage()} ${this.privateMessage()}
    </div>`}};exports.WhoIsOnlineMessages.styles=O,exports.WhoIsOnlineMessages.properties={everyoneFollowsMe:{type:Boolean},isPrivate:{type:Boolean},verticalSide:{type:String},horizontalSide:{type:String}},exports.WhoIsOnlineMessages=p.F([n.ts("superviz-who-is-online-messages")],exports.WhoIsOnlineMessages);p.y();var y=_.D(n.C),k=[y.styles,P];exports.WhoIsOnlineDropdown=class extends y{constructor(){super(),this.onClickOutDropdown=i=>{if(i.stopPropagation(),!this.open)return;let o=i.composedPath(),t=this.shadowRoot.querySelector(".dropdown-content"),s=this.shadowRoot.querySelector(".dropdown-list"),r=this.shadowRoot.querySelector('slot[name="dropdown"]').assignedElements()[0],l=o.includes(t),a=o.includes(s),d=o.includes(r);l||a||d||this.close()},this.close=()=>{this.open=!1,this.selected="",this.emitEvent("clickout",{bubbles:!1,composed:!1})},this.selectParticipant=(i,o)=>()=>{o||(this.selected=i)},this.toggleShowTooltip=({detail:{open:i}})=>{this.showParticipantTooltip=!this.showParticipantTooltip,i||(this.selected="")},this.tooltip=()=>this.showSeeMoreTooltip?n.Vt`<superviz-tooltip
      tooltipData=${JSON.stringify({name:"See more"})}
      classesPrefix="who-is-online__tooltip"
      parentComponent="who-is-online"
    ></superviz-tooltip>`:"",this.repositionDropdown=()=>{if(!this.open||!this.parentElement){window.cancelAnimationFrame(this.animationFrame);return}this.repositionInVerticalDirection(),this.repositionInHorizontalDirection(),this.animationFrame=window.requestAnimationFrame(this.repositionDropdown)},this.repositionInVerticalDirection=()=>{let{bottom:i,top:o,height:t}=this.parentElement.getBoundingClientRect(),s=window.innerHeight/2;if(o+t/2>s){this.dropdownList.style.setProperty("bottom",`${window.innerHeight-o+8}px`),this.dropdownList.style.setProperty("top","");return}this.dropdownList.style.setProperty("top",`${i+8}px`),this.dropdownList.style.setProperty("bottom","")},this.repositionInHorizontalDirection=()=>{let{right:i,left:o}=this.parentElement.getBoundingClientRect();this.dropdownList.style.setProperty("right",`${window.innerWidth-i}px`),this.dropdownList.style.setProperty("left",`${o}px`)},this.selected="",this.showParticipantTooltip=!0;let{extras:e}=this.useStore("who-is-online-store");e.subscribe(i=>{this.participants=i})}firstUpdated(e){this.shadowRoot.querySelector(".who-is-online__extras-dropdown").scrollTop=0,g.n.call(this,"who-is-online"),this.dropdownList=this.shadowRoot.querySelector(".dropdown-list"),this.dropdownList.style.position="fixed"}updated(e){if(e.has("open")){if(this.emitEvent("toggle-dropdown-state",{open:this.open,font:"toggle"}),this.open){document.addEventListener("click",this.onClickOutDropdown);return}document.removeEventListener("click",this.onClickOutDropdown)}}getAvatar({color:e,imageUrl:i,firstLetter:o,letterColor:t}){return i?n.Vt` <img
        class="who-is-online__participant__avatar"
        style="background-color: ${e}"
        src=${i}
      />`:n.Vt`<div
      class="who-is-online__participant__avatar"
      style="background-color: ${e}; color: ${t}"
    >
      ${o}
    </div>`}renderParticipants(){if(!this.participants)return;let e=this.participants.length-1;return m.j(this.participants,i=>i.id,(i,o)=>{let{disableDropdown:t,id:s,avatar:r,controls:l,tooltip:a,name:d}=i,h={"who-is-online__extra-participant":!0,"who-is-online__extra-participant--selected":this.selected===s,"disable-dropdown":t},w={"who-is-online__extras__arrow-icon":!0,"hide-icon":t},u=o===e;return n.Vt`
        <superviz-dropdown
          options=${JSON.stringify(l)}
          returnData=${JSON.stringify({participantId:s,source:"extras"})}
          tooltipData=${JSON.stringify(a)}
          position="bottom-right"
          classesPrefix="who-is-online__controls"
          parentComponent="who-is-online"
          tooltipPrefix="who-is-online"
          ?disabled=${t}
          ?canShowTooltip=${this.showParticipantTooltip}
          ?shiftTooltipLeft=${!0}
          ?lastParticipant=${u}
          @toggle-dropdown-state=${this.toggleShowTooltip}
          @selected=${this.close}
        >
        <div 
          class=${c.u(h)} 
          @click=${this.selectParticipant(s,t)} slot="dropdown">
            <div class="who-is-online__participant" style="border-color: 
            ${r.color}">
              ${this.getAvatar(r)}
            </div>
            <span class="who-is-online__extras__username">${d}</span>
            <superviz-icon 
              class=${c.u(w)} 
              name="right" 
              color="var(--sv-gray-600)"
              size="sm"
            >
          </superviz-icon>
          </div>
        </div>
      </superviz-dropdown>
      `})}toggle(){this.open=!this.open,this.open&&(this.selected="",this.repositionDropdown())}get menuClasses(){return{"who-is-online__extras-dropdown":!0,"menu--bottom":this.position==="bottom-side","menu--top":this.position==="top-side","menu-open":this.open}}render(){return n.Vt`
      <div class="dropdown">
        <div class="dropdown-content" @click=${this.toggle}>
          <slot name="dropdown"></slot>
        </div>
        ${this.tooltip()}
      </div>
      <div class="dropdown-list">
        <div class=${c.u(this.menuClasses)}>${this.renderParticipants()}</div>
      </div>
    `}};exports.WhoIsOnlineDropdown.styles=k,exports.WhoIsOnlineDropdown.properties={open:{type:Boolean},align:{type:String},position:{type:String},selected:{type:String},disableDropdown:{type:Boolean},showSeeMoreTooltip:{type:Boolean},showParticipantTooltip:{type:Boolean},localParticipantJoinedPresence:{type:Boolean},dropdownList:{type:Object}},exports.WhoIsOnlineDropdown=p.F([n.ts("superviz-who-is-online-dropdown")],exports.WhoIsOnlineDropdown);
