// @superviz/react-sdk/dist/annotation-resolved-6VJKFGWE-DyTa30u8.js
import { m as o } from "./chunk-SDMTFDEM-n5DHnLs2.js";
import { n } from "./chunk-6U6BYUHV-TZ4JwLT4.js";
import { D as m } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { V as t, a as d, t as r } from "./chunk-WG46WHJF-JdNs68CV.js";
import { y as a, e as c } from "./index-CtPfqlcy.js";
a();
var s = m(d), l = [s.styles, o], u = 10 * 1e3, e = class extends s {
  constructor() {
    super(), this.setTimer = () => {
      clearTimeout(this.timeout), this.isCanceled = !1, this.timeout = setTimeout(() => {
        this.isCanceled || (this.timeToHide = 0, this.isCanceled = !1, this.hide());
      }, this.timeToHide);
    }, this.timeToHide = u, this.isCanceled = !1;
  }
  firstUpdated(i) {
    super.firstUpdated(i), this.updateComplete.then(() => {
      n.call(this, ["comments"]), this.setTimer();
    });
  }
  hide() {
    this.emitEvent("hide", {}, { bubbles: !1, composed: !1 });
  }
  undone() {
    this.isCanceled = !0, this.hide(), this.emitEvent("undo-resolve", { type: "undo-resolve", resolved: !1 }, { bubbles: !1, composed: !1 }), clearTimeout(this.timeout);
  }
  render() {
    return this.timeToHide === 0 ? t`` : this.isCanceled ? t`` : t`
      <div class="comments__annotation-resolved">
        <span class="text text-big sv-gray-700 comments__annotation-resolved__text"
          >You resolved this comment</span
        >
        <button
          id="undone"
          @click=${this.undone}
          class="icon-button icon-button--clickable icon-button--medium comments__annotation-resolved__unresolve_button"
        >
          <superviz-icon name="undo" size="md"></superviz-icon>
        </button>
      </div>
    `;
  }
};
e.styles = l, e.properties = { timeToHide: { type: Number }, isCanceled: { type: Boolean } }, e = c([r("superviz-comments-annotation-resolved")], e);
export {
  e as CommentsAnnotationResolved
};
