// @superviz/react-sdk/dist/chunk-WG46WHJF-JdNs68CV.js
import { y as u, G as mt } from "./index-CtPfqlcy.js";
u();
var H = globalThis, z = H.ShadowRoot && (H.ShadyCSS === void 0 || H.ShadyCSS.nativeShadow) && "adoptedStyleSheets" in Document.prototype && "replace" in CSSStyleSheet.prototype, j = Symbol(), D = /* @__PURE__ */ new WeakMap(), lt = class {
  constructor(t, e, i) {
    if (this._$cssResult$ = !0, i !== j) throw Error("CSSResult is not constructable. Use `unsafeCSS` or `css` instead.");
    this.cssText = t, this.t = e;
  }
  get styleSheet() {
    let t = this.o, e = this.t;
    if (z && t === void 0) {
      let i = e !== void 0 && e.length === 1;
      i && (t = D.get(e)), t === void 0 && ((this.o = t = new CSSStyleSheet()).replaceSync(this.cssText), i && D.set(e, t));
    }
    return t;
  }
  toString() {
    return this.cssText;
  }
}, gt = (t) => new lt(typeof t == "string" ? t : t + "", void 0, j), Rt = (t, ...e) => {
  let i = t.length === 1 ? t[0] : e.reduce((s, n, r) => s + ((l) => {
    if (l._$cssResult$ === !0) return l.cssText;
    if (typeof l == "number") return l;
    throw Error("Value passed to 'css' function must be a 'css' function result: " + l + ". Use 'unsafeCSS' to pass non-literal values, but take care to ensure page security.");
  })(n) + t[r + 1], t[0]);
  return new lt(i, t, j);
}, yt = (t, e) => {
  if (z) t.adoptedStyleSheets = e.map((i) => i instanceof CSSStyleSheet ? i : i.styleSheet);
  else for (let i of e) {
    let s = document.createElement("style"), n = H.litNonce;
    n !== void 0 && s.setAttribute("nonce", n), s.textContent = i.cssText, t.appendChild(s);
  }
}, W = z ? (t) => t : (t) => t instanceof CSSStyleSheet ? ((e) => {
  let i = "";
  for (let s of e.cssRules) i += s.cssText;
  return gt(i);
})(t) : t;
u();
var { is: Et, defineProperty: St, getOwnPropertyDescriptor: bt, getOwnPropertyNames: wt, getOwnPropertySymbols: Pt, getPrototypeOf: Ct } = Object, E = globalThis, J = E.trustedTypes, Ut = J ? J.emptyScript : "", q = E.reactiveElementPolyfillSupport, w = (t, e) => t, k = { toAttribute(t, e) {
  switch (e) {
    case Boolean:
      t = t ? Ut : null;
      break;
    case Object:
    case Array:
      t = t == null ? t : JSON.stringify(t);
  }
  return t;
}, fromAttribute(t, e) {
  let i = t;
  switch (e) {
    case Boolean:
      i = t !== null;
      break;
    case Number:
      i = t === null ? null : Number(t);
      break;
    case Object:
    case Array:
      try {
        i = JSON.parse(t);
      } catch {
        i = null;
      }
  }
  return i;
} }, ot = (t, e) => !Et(t, e), G = { attribute: !0, type: String, converter: k, reflect: !1, hasChanged: ot };
Symbol.metadata != null || (Symbol.metadata = Symbol("metadata")), E.litPropertyMetadata != null || (E.litPropertyMetadata = /* @__PURE__ */ new WeakMap());
var y = class extends HTMLElement {
  static addInitializer(t) {
    var e;
    this._$Ei(), ((e = this.l) != null ? e : this.l = []).push(t);
  }
  static get observedAttributes() {
    return this.finalize(), this._$Eh && [...this._$Eh.keys()];
  }
  static createProperty(t, e = G) {
    if (e.state && (e.attribute = !1), this._$Ei(), this.elementProperties.set(t, e), !e.noAccessor) {
      let i = Symbol(), s = this.getPropertyDescriptor(t, i, e);
      s !== void 0 && St(this.prototype, t, s);
    }
  }
  static getPropertyDescriptor(t, e, i) {
    var s;
    let { get: n, set: r } = (s = bt(this.prototype, t)) != null ? s : { get() {
      return this[e];
    }, set(l) {
      this[e] = l;
    } };
    return { get() {
      return n == null ? void 0 : n.call(this);
    }, set(l) {
      let o = n == null ? void 0 : n.call(this);
      r.call(this, l), this.requestUpdate(t, o, i);
    }, configurable: !0, enumerable: !0 };
  }
  static getPropertyOptions(t) {
    var e;
    return (e = this.elementProperties.get(t)) != null ? e : G;
  }
  static _$Ei() {
    if (this.hasOwnProperty(w("elementProperties"))) return;
    let t = Ct(this);
    t.finalize(), t.l !== void 0 && (this.l = [...t.l]), this.elementProperties = new Map(t.elementProperties);
  }
  static finalize() {
    if (this.hasOwnProperty(w("finalized"))) return;
    if (this.finalized = !0, this._$Ei(), this.hasOwnProperty(w("properties"))) {
      let e = this.properties, i = [...wt(e), ...Pt(e)];
      for (let s of i) this.createProperty(s, e[s]);
    }
    let t = this[Symbol.metadata];
    if (t !== null) {
      let e = litPropertyMetadata.get(t);
      if (e !== void 0) for (let [i, s] of e) this.elementProperties.set(i, s);
    }
    this._$Eh = /* @__PURE__ */ new Map();
    for (let [e, i] of this.elementProperties) {
      let s = this._$Eu(e, i);
      s !== void 0 && this._$Eh.set(s, e);
    }
    this.elementStyles = this.finalizeStyles(this.styles);
  }
  static finalizeStyles(t) {
    let e = [];
    if (Array.isArray(t)) {
      let i = new Set(t.flat(1 / 0).reverse());
      for (let s of i) e.unshift(W(s));
    } else t !== void 0 && e.push(W(t));
    return e;
  }
  static _$Eu(t, e) {
    let i = e.attribute;
    return i === !1 ? void 0 : typeof i == "string" ? i : typeof t == "string" ? t.toLowerCase() : void 0;
  }
  constructor() {
    super(), this._$Ep = void 0, this.isUpdatePending = !1, this.hasUpdated = !1, this._$Em = null, this._$Ev();
  }
  _$Ev() {
    var t;
    this._$ES = new Promise((e) => this.enableUpdating = e), this._$AL = /* @__PURE__ */ new Map(), this._$E_(), this.requestUpdate(), (t = this.constructor.l) == null || t.forEach((e) => e(this));
  }
  addController(t) {
    var e, i;
    ((e = this._$EO) != null ? e : this._$EO = /* @__PURE__ */ new Set()).add(t), this.renderRoot !== void 0 && this.isConnected && ((i = t.hostConnected) == null || i.call(t));
  }
  removeController(t) {
    var e;
    (e = this._$EO) == null || e.delete(t);
  }
  _$E_() {
    let t = /* @__PURE__ */ new Map(), e = this.constructor.elementProperties;
    for (let i of e.keys()) this.hasOwnProperty(i) && (t.set(i, this[i]), delete this[i]);
    t.size > 0 && (this._$Ep = t);
  }
  createRenderRoot() {
    var t;
    let e = (t = this.shadowRoot) != null ? t : this.attachShadow(this.constructor.shadowRootOptions);
    return yt(e, this.constructor.elementStyles), e;
  }
  connectedCallback() {
    var t;
    this.renderRoot != null || (this.renderRoot = this.createRenderRoot()), this.enableUpdating(!0), (t = this._$EO) == null || t.forEach((e) => {
      var i;
      return (i = e.hostConnected) == null ? void 0 : i.call(e);
    });
  }
  enableUpdating(t) {
  }
  disconnectedCallback() {
    var t;
    (t = this._$EO) == null || t.forEach((e) => {
      var i;
      return (i = e.hostDisconnected) == null ? void 0 : i.call(e);
    });
  }
  attributeChangedCallback(t, e, i) {
    this._$AK(t, i);
  }
  _$EC(t, e) {
    var i;
    let s = this.constructor.elementProperties.get(t), n = this.constructor._$Eu(t, s);
    if (n !== void 0 && s.reflect === !0) {
      let r = (((i = s.converter) == null ? void 0 : i.toAttribute) !== void 0 ? s.converter : k).toAttribute(e, s.type);
      this._$Em = t, r == null ? this.removeAttribute(n) : this.setAttribute(n, r), this._$Em = null;
    }
  }
  _$AK(t, e) {
    var i;
    let s = this.constructor, n = s._$Eh.get(t);
    if (n !== void 0 && this._$Em !== n) {
      let r = s.getPropertyOptions(n), l = typeof r.converter == "function" ? { fromAttribute: r.converter } : ((i = r.converter) == null ? void 0 : i.fromAttribute) !== void 0 ? r.converter : k;
      this._$Em = n, this[n] = l.fromAttribute(e, r.type), this._$Em = null;
    }
  }
  requestUpdate(t, e, i) {
    var s;
    if (t !== void 0) {
      if (i != null || (i = this.constructor.getPropertyOptions(t)), !((s = i.hasChanged) != null ? s : ot)(this[t], e)) return;
      this.P(t, e, i);
    }
    this.isUpdatePending === !1 && (this._$ES = this._$ET());
  }
  P(t, e, i) {
    var s;
    this._$AL.has(t) || this._$AL.set(t, e), i.reflect === !0 && this._$Em !== t && ((s = this._$Ej) != null ? s : this._$Ej = /* @__PURE__ */ new Set()).add(t);
  }
  _$ET() {
    return mt(this, null, function* () {
      this.isUpdatePending = !0;
      try {
        yield this._$ES;
      } catch (e) {
        Promise.reject(e);
      }
      let t = this.scheduleUpdate();
      return t != null && (yield t), !this.isUpdatePending;
    });
  }
  scheduleUpdate() {
    return this.performUpdate();
  }
  performUpdate() {
    var t;
    if (!this.isUpdatePending) return;
    if (!this.hasUpdated) {
      if (this.renderRoot != null || (this.renderRoot = this.createRenderRoot()), this._$Ep) {
        for (let [n, r] of this._$Ep) this[n] = r;
        this._$Ep = void 0;
      }
      let s = this.constructor.elementProperties;
      if (s.size > 0) for (let [n, r] of s) r.wrapped !== !0 || this._$AL.has(n) || this[n] === void 0 || this.P(n, this[n], r);
    }
    let e = !1, i = this._$AL;
    try {
      e = this.shouldUpdate(i), e ? (this.willUpdate(i), (t = this._$EO) == null || t.forEach((s) => {
        var n;
        return (n = s.hostUpdate) == null ? void 0 : n.call(s);
      }), this.update(i)) : this._$EU();
    } catch (s) {
      throw e = !1, this._$EU(), s;
    }
    e && this._$AE(i);
  }
  willUpdate(t) {
  }
  _$AE(t) {
    var e;
    (e = this._$EO) == null || e.forEach((i) => {
      var s;
      return (s = i.hostUpdated) == null ? void 0 : s.call(i);
    }), this.hasUpdated || (this.hasUpdated = !0, this.firstUpdated(t)), this.updated(t);
  }
  _$EU() {
    this._$AL = /* @__PURE__ */ new Map(), this.isUpdatePending = !1;
  }
  get updateComplete() {
    return this.getUpdateComplete();
  }
  getUpdateComplete() {
    return this._$ES;
  }
  shouldUpdate(t) {
    return !0;
  }
  update(t) {
    this._$Ej && (this._$Ej = this._$Ej.forEach((e) => this._$EC(e, this[e]))), this._$EU();
  }
  updated(t) {
  }
  firstUpdated(t) {
  }
}, Z;
y.elementStyles = [], y.shadowRootOptions = { mode: "open" }, y[w("elementProperties")] = /* @__PURE__ */ new Map(), y[w("finalized")] = /* @__PURE__ */ new Map(), q == null || q({ ReactiveElement: y }), ((Z = E.reactiveElementVersions) != null ? Z : E.reactiveElementVersions = []).push("2.0.4");
u();
var O = globalThis, N = O.trustedTypes, F = N ? N.createPolicy("lit-html", { createHTML: (t) => t }) : void 0, B = "$lit$", _ = `lit$${Math.random().toFixed(9).slice(2)}$`, I = "?" + _, xt = `<${I}>`, m = document, P = () => m.createComment(""), C = (t) => t === null || typeof t != "object" && typeof t != "function", V = Array.isArray, ht = (t) => V(t) || typeof (t == null ? void 0 : t[Symbol.iterator]) == "function", M = `[ 	
\f\r]`, b = /<(?:(!--|\/[^a-zA-Z])|(\/?[a-zA-Z][^>\s]*)|(\/?$))/g, K = /-->/g, Q = />/g, f = RegExp(`>|${M}(?:([^\\s"'>=/]+)(${M}*=${M}*(?:[^ 	
\f\r"'\`<>=]|("|')|))|$)`, "g"), X = /'/g, Y = /"/g, at = /^(?:script|style|textarea|title)$/i, Ht = (t) => (e, ...i) => ({ _$litType$: t, strings: e, values: i }), Mt = Ht(1), S = Symbol.for("lit-noChange"), d = Symbol.for("lit-nothing"), tt = /* @__PURE__ */ new WeakMap(), A = m.createTreeWalker(m, 129);
function dt(t, e) {
  if (!V(t) || !t.hasOwnProperty("raw")) throw Error("invalid template strings array");
  return F !== void 0 ? F.createHTML(e) : e;
}
var ut = (t, e) => {
  let i = t.length - 1, s = [], n, r = e === 2 ? "<svg>" : e === 3 ? "<math>" : "", l = b;
  for (let o = 0; o < i; o++) {
    let h = t[o], p, c, a = -1, $ = 0;
    for (; $ < h.length && (l.lastIndex = $, c = l.exec(h), c !== null); ) $ = l.lastIndex, l === b ? c[1] === "!--" ? l = K : c[1] !== void 0 ? l = Q : c[2] !== void 0 ? (at.test(c[2]) && (n = RegExp("</" + c[2], "g")), l = f) : c[3] !== void 0 && (l = f) : l === f ? c[0] === ">" ? (l = n ?? b, a = -1) : c[1] === void 0 ? a = -2 : (a = l.lastIndex - c[2].length, p = c[1], l = c[3] === void 0 ? f : c[3] === '"' ? Y : X) : l === Y || l === X ? l = f : l === K || l === Q ? l = b : (l = f, n = void 0);
    let v = l === f && t[o + 1].startsWith("/>") ? " " : "";
    r += l === b ? h + xt : a >= 0 ? (s.push(p), h.slice(0, a) + B + h.slice(a) + _ + v) : h + _ + (a === -2 ? o : v);
  }
  return [dt(t, r + (t[i] || "<?>") + (e === 2 ? "</svg>" : e === 3 ? "</math>" : "")), s];
}, L = class ct {
  constructor({ strings: e, _$litType$: i }, s) {
    let n;
    this.parts = [];
    let r = 0, l = 0, o = e.length - 1, h = this.parts, [p, c] = ut(e, i);
    if (this.el = ct.createElement(p, s), A.currentNode = this.el.content, i === 2 || i === 3) {
      let a = this.el.content.firstChild;
      a.replaceWith(...a.childNodes);
    }
    for (; (n = A.nextNode()) !== null && h.length < o; ) {
      if (n.nodeType === 1) {
        if (n.hasAttributes()) for (let a of n.getAttributeNames()) if (a.endsWith(B)) {
          let $ = c[l++], v = n.getAttribute(a).split(_), x = /([.?@])?(.*)/.exec($);
          h.push({ type: 1, index: r, name: x[2], strings: v, ctor: x[1] === "." ? _t : x[1] === "?" ? vt : x[1] === "@" ? ft : U }), n.removeAttribute(a);
        } else a.startsWith(_) && (h.push({ type: 6, index: r }), n.removeAttribute(a));
        if (at.test(n.tagName)) {
          let a = n.textContent.split(_), $ = a.length - 1;
          if ($ > 0) {
            n.textContent = N ? N.emptyScript : "";
            for (let v = 0; v < $; v++) n.append(a[v], P()), A.nextNode(), h.push({ type: 2, index: ++r });
            n.append(a[$], P());
          }
        }
      } else if (n.nodeType === 8) if (n.data === I) h.push({ type: 2, index: r });
      else {
        let a = -1;
        for (; (a = n.data.indexOf(_, a + 1)) !== -1; ) h.push({ type: 7, index: r }), a += _.length - 1;
      }
      r++;
    }
  }
  static createElement(e, i) {
    let s = m.createElement("template");
    return s.innerHTML = e, s;
  }
};
function g(t, e, i = t, s) {
  var n, r, l;
  if (e === S) return e;
  let o = s !== void 0 ? (n = i.o) == null ? void 0 : n[s] : i.l, h = C(e) ? void 0 : e._$litDirective$;
  return (o == null ? void 0 : o.constructor) !== h && ((r = o == null ? void 0 : o._$AO) == null || r.call(o, !1), h === void 0 ? o = void 0 : (o = new h(t), o._$AT(t, i, s)), s !== void 0 ? ((l = i.o) != null ? l : i.o = [])[s] = o : i.l = o), o !== void 0 && (e = g(t, o._$AS(t, e.values), o, s)), e;
}
var pt = class {
  constructor(t, e) {
    this._$AV = [], this._$AN = void 0, this._$AD = t, this._$AM = e;
  }
  get parentNode() {
    return this._$AM.parentNode;
  }
  get _$AU() {
    return this._$AM._$AU;
  }
  u(t) {
    var e;
    let { el: { content: i }, parts: s } = this._$AD, n = ((e = t == null ? void 0 : t.creationScope) != null ? e : m).importNode(i, !0);
    A.currentNode = n;
    let r = A.nextNode(), l = 0, o = 0, h = s[0];
    for (; h !== void 0; ) {
      if (l === h.index) {
        let p;
        h.type === 2 ? p = new R(r, r.nextSibling, this, t) : h.type === 1 ? p = new h.ctor(r, h.name, h.strings, this, t) : h.type === 6 && (p = new At(r, this, t)), this._$AV.push(p), h = s[++o];
      }
      l !== (h == null ? void 0 : h.index) && (r = A.nextNode(), l++);
    }
    return A.currentNode = m, n;
  }
  p(t) {
    let e = 0;
    for (let i of this._$AV) i !== void 0 && (i.strings !== void 0 ? (i._$AI(t, i, e), e += i.strings.length - 2) : i._$AI(t[e])), e++;
  }
}, R = class $t {
  get _$AU() {
    var e, i;
    return (i = (e = this._$AM) == null ? void 0 : e._$AU) != null ? i : this.v;
  }
  constructor(e, i, s, n) {
    var r;
    this.type = 2, this._$AH = d, this._$AN = void 0, this._$AA = e, this._$AB = i, this._$AM = s, this.options = n, this.v = (r = n == null ? void 0 : n.isConnected) != null ? r : !0;
  }
  get parentNode() {
    let e = this._$AA.parentNode, i = this._$AM;
    return i !== void 0 && (e == null ? void 0 : e.nodeType) === 11 && (e = i.parentNode), e;
  }
  get startNode() {
    return this._$AA;
  }
  get endNode() {
    return this._$AB;
  }
  _$AI(e, i = this) {
    e = g(this, e, i), C(e) ? e === d || e == null || e === "" ? (this._$AH !== d && this._$AR(), this._$AH = d) : e !== this._$AH && e !== S && this._(e) : e._$litType$ !== void 0 ? this.$(e) : e.nodeType !== void 0 ? this.T(e) : ht(e) ? this.k(e) : this._(e);
  }
  O(e) {
    return this._$AA.parentNode.insertBefore(e, this._$AB);
  }
  T(e) {
    this._$AH !== e && (this._$AR(), this._$AH = this.O(e));
  }
  _(e) {
    this._$AH !== d && C(this._$AH) ? this._$AA.nextSibling.data = e : this.T(m.createTextNode(e)), this._$AH = e;
  }
  $(e) {
    var i;
    let { values: s, _$litType$: n } = e, r = typeof n == "number" ? this._$AC(e) : (n.el === void 0 && (n.el = L.createElement(dt(n.h, n.h[0]), this.options)), n);
    if (((i = this._$AH) == null ? void 0 : i._$AD) === r) this._$AH.p(s);
    else {
      let l = new pt(r, this), o = l.u(this.options);
      l.p(s), this.T(o), this._$AH = l;
    }
  }
  _$AC(e) {
    let i = tt.get(e.strings);
    return i === void 0 && tt.set(e.strings, i = new L(e)), i;
  }
  k(e) {
    V(this._$AH) || (this._$AH = [], this._$AR());
    let i = this._$AH, s, n = 0;
    for (let r of e) n === i.length ? i.push(s = new $t(this.O(P()), this.O(P()), this, this.options)) : s = i[n], s._$AI(r), n++;
    n < i.length && (this._$AR(s && s._$AB.nextSibling, n), i.length = n);
  }
  _$AR(e = this._$AA.nextSibling, i) {
    var s;
    for ((s = this._$AP) == null ? void 0 : s.call(this, !1, !0, i); e && e !== this._$AB; ) {
      let n = e.nextSibling;
      e.remove(), e = n;
    }
  }
  setConnected(e) {
    var i;
    this._$AM === void 0 && (this.v = e, (i = this._$AP) == null || i.call(this, e));
  }
}, U = class {
  get tagName() {
    return this.element.tagName;
  }
  get _$AU() {
    return this._$AM._$AU;
  }
  constructor(t, e, i, s, n) {
    this.type = 1, this._$AH = d, this._$AN = void 0, this.element = t, this.name = e, this._$AM = s, this.options = n, i.length > 2 || i[0] !== "" || i[1] !== "" ? (this._$AH = Array(i.length - 1).fill(new String()), this.strings = i) : this._$AH = d;
  }
  _$AI(t, e = this, i, s) {
    let n = this.strings, r = !1;
    if (n === void 0) t = g(this, t, e, 0), r = !C(t) || t !== this._$AH && t !== S, r && (this._$AH = t);
    else {
      let l = t, o, h;
      for (t = n[0], o = 0; o < n.length - 1; o++) h = g(this, l[i + o], e, o), h === S && (h = this._$AH[o]), r || (r = !C(h) || h !== this._$AH[o]), h === d ? t = d : t !== d && (t += (h ?? "") + n[o + 1]), this._$AH[o] = h;
    }
    r && !s && this.j(t);
  }
  j(t) {
    t === d ? this.element.removeAttribute(this.name) : this.element.setAttribute(this.name, t ?? "");
  }
}, _t = class extends U {
  constructor() {
    super(...arguments), this.type = 3;
  }
  j(t) {
    this.element[this.name] = t === d ? void 0 : t;
  }
}, vt = class extends U {
  constructor() {
    super(...arguments), this.type = 4;
  }
  j(t) {
    this.element.toggleAttribute(this.name, !!t && t !== d);
  }
}, ft = class extends U {
  constructor(t, e, i, s, n) {
    super(t, e, i, s, n), this.type = 5;
  }
  _$AI(t, e = this) {
    var i;
    if ((t = (i = g(this, t, e, 0)) != null ? i : d) === S) return;
    let s = this._$AH, n = t === d && s !== d || t.capture !== s.capture || t.once !== s.once || t.passive !== s.passive, r = t !== d && (s === d || n);
    n && this.element.removeEventListener(this.name, this, s), r && this.element.addEventListener(this.name, this, t), this._$AH = t;
  }
  handleEvent(t) {
    var e, i;
    typeof this._$AH == "function" ? this._$AH.call((i = (e = this.options) == null ? void 0 : e.host) != null ? i : this.element, t) : this._$AH.handleEvent(t);
  }
}, At = class {
  constructor(t, e, i) {
    this.element = t, this.type = 6, this._$AN = void 0, this._$AM = e, this.options = i;
  }
  get _$AU() {
    return this._$AM._$AU;
  }
  _$AI(t) {
    g(this, t);
  }
}, kt = { M: B, P: _, A: I, C: 1, L: ut, R: pt, D: ht, V: g, I: R, H: U, N: vt, U: ft, B: _t, F: At }, et = O.litHtmlPolyfillSupport, it;
et == null || et(L, R), ((it = O.litHtmlVersions) != null ? it : O.litHtmlVersions = []).push("3.2.0");
var Tt = (t, e, i) => {
  var s, n;
  let r = (s = i == null ? void 0 : i.renderBefore) != null ? s : e, l = r._$litPart$;
  if (l === void 0) {
    let o = (n = i == null ? void 0 : i.renderBefore) != null ? n : null;
    r._$litPart$ = l = new R(e.insertBefore(P(), o), o, void 0, i ?? {});
  }
  return l._$AI(t), l;
};
u();
var T = class extends y {
  constructor() {
    super(...arguments), this.renderOptions = { host: this }, this.o = void 0;
  }
  createRenderRoot() {
    var t;
    let e = super.createRenderRoot();
    return (t = this.renderOptions).renderBefore != null || (t.renderBefore = e.firstChild), e;
  }
  update(t) {
    let e = this.render();
    this.hasUpdated || (this.renderOptions.isConnected = this.isConnected), super.update(t), this.o = Tt(e, this.renderRoot, this.renderOptions);
  }
  connectedCallback() {
    var t;
    super.connectedCallback(), (t = this.o) == null || t.setConnected(!0);
  }
  disconnectedCallback() {
    var t;
    super.disconnectedCallback(), (t = this.o) == null || t.setConnected(!1);
  }
  render() {
    return S;
  }
}, st;
T._$litElement$ = !0, T.finalized = !0, (st = globalThis.litElementHydrateSupport) == null || st.call(globalThis, { LitElement: T });
var nt = globalThis.litElementPolyfillSupport;
nt == null || nt({ LitElement: T });
var rt;
((rt = globalThis.litElementVersions) != null ? rt : globalThis.litElementVersions = []).push("4.1.0");
u();
u();
u();
u();
u();
var Ot = (t) => (e, i) => {
  i !== void 0 ? i.addInitializer(() => {
    customElements.define(t, e);
  }) : customElements.define(t, e);
};
u();
u();
u();
u();
u();
u();
u();
u();
u();
function Lt(t) {
  return function(e) {
    let i = !!customElements.get(t);
    typeof window > "u" || typeof HTMLElement > "u" || i || Ot(t)(e);
  };
}
/*! Bundled license information:

@lit/reactive-element/css-tag.js:
  (**
   * @license
   * Copyright 2019 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/reactive-element.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

lit-html/lit-html.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

lit-element/lit-element.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

lit-html/is-server.js:
  (**
   * @license
   * Copyright 2022 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/custom-element.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/property.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/state.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/event-options.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/base.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/query.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/query-all.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/query-async.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/query-assigned-elements.js:
  (**
   * @license
   * Copyright 2021 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

@lit/reactive-element/decorators/query-assigned-nodes.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)
*/
export {
  Rt as C,
  S as E,
  kt as J,
  Mt as V,
  T as a,
  Lt as t
};
