// @superviz/react-sdk/dist/chunk-DX57ODX6-DAIpi7BW.js
import { a as S, i as E, E as H } from "./chunk-3MEZQQ2J-CHe_tQrN.js";
import { E as M, J as P } from "./chunk-WG46WHJF-JdNs68CV.js";
import { y as g } from "./index-CtPfqlcy.js";
g();
g();
var { I: w } = P, B = () => document.createComment(""), h = (e, t, l) => {
  var A;
  let f = e._$AA.parentNode, r = t === void 0 ? e._$AB : t._$AA;
  if (l === void 0) {
    let i = f.insertBefore(B(), r), a = f.insertBefore(B(), r);
    l = new w(i, a, e, e.options);
  } else {
    let i = l._$AB.nextSibling, a = l._$AM, v = a !== e;
    if (v) {
      let u;
      (A = l._$AQ) == null || A.call(l, e), l._$AM = e, l._$AP !== void 0 && (u = e._$AU) !== a._$AU && l._$AP(u);
    }
    if (i !== r || v) {
      let u = l._$AA;
      for (; u !== i; ) {
        let p = u.nextSibling;
        f.insertBefore(u, r), u = p;
      }
    }
  }
  return l;
}, d = (e, t, l = e) => (e._$AI(t, l), e), I = {}, k = (e, t = I) => e._$AH = t, C = (e) => e._$AH, c = (e) => {
  var t;
  (t = e._$AP) == null || t.call(e, !1, !0);
  let l = e._$AA, A = e._$AB.nextSibling;
  for (; l !== A; ) {
    let f = l.nextSibling;
    l.remove(), l = f;
  }
}, b = (e, t, l) => {
  let A = /* @__PURE__ */ new Map();
  for (let f = t; f <= l; f++) A.set(e[f], f);
  return A;
}, j = S(class extends E {
  constructor(e) {
    if (super(e), e.type !== H.CHILD) throw Error("repeat() can only be used in text expressions");
  }
  dt(e, t, l) {
    let A;
    l === void 0 ? l = t : t !== void 0 && (A = t);
    let f = [], r = [], i = 0;
    for (let a of e) f[i] = A ? A(a, i) : i, r[i] = l(a, i), i++;
    return { values: r, keys: f };
  }
  render(e, t, l) {
    return this.dt(e, t, l).values;
  }
  update(e, [t, l, A]) {
    var f;
    let r = C(e), { values: i, keys: a } = this.dt(t, l, A);
    if (!Array.isArray(r)) return this.ut = a, i;
    let v = (f = this.ut) != null ? f : this.ut = [], u = [], p, x, n = 0, s = r.length - 1, o = 0, $ = i.length - 1;
    for (; n <= s && o <= $; ) if (r[n] === null) n++;
    else if (r[s] === null) s--;
    else if (v[n] === a[o]) u[o] = d(r[n], i[o]), n++, o++;
    else if (v[s] === a[$]) u[$] = d(r[s], i[$]), s--, $--;
    else if (v[n] === a[$]) u[$] = d(r[n], i[$]), h(e, u[$ + 1], r[n]), n++, $--;
    else if (v[s] === a[o]) u[o] = d(r[s], i[o]), h(e, r[n], r[s]), s--, o++;
    else if (p === void 0 && (p = b(a, o, $), x = b(v, n, s)), p.has(v[n])) if (p.has(v[s])) {
      let _ = x.get(a[o]), m = _ !== void 0 ? r[_] : null;
      if (m === null) {
        let y = h(e, r[n]);
        d(y, i[o]), u[o] = y;
      } else u[o] = d(m, i[o]), h(e, r[n], m), r[_] = null;
      o++;
    } else c(r[s]), s--;
    else c(r[n]), n++;
    for (; o <= $; ) {
      let _ = h(e, u[$ + 1]);
      d(_, i[o]), u[o++] = _;
    }
    for (; n <= s; ) {
      let _ = r[n++];
      _ !== null && c(_);
    }
    return this.ut = a, k(e, u), M;
  }
});
g();
/*! Bundled license information:

lit-html/directive-helpers.js:
  (**
   * @license
   * Copyright 2020 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)

lit-html/directives/repeat.js:
  (**
   * @license
   * Copyright 2017 Google LLC
   * SPDX-License-Identifier: BSD-3-Clause
   *)
*/
export {
  j
};
