// @superviz/react-sdk/dist/content-BR3JBVI3-DcusmrRY.js
import { j as c } from "./chunk-DX57ODX6-DAIpi7BW.js";
import { y as d, e as u } from "./index-CtPfqlcy.js";
import "./chunk-3MEZQQ2J-CHe_tQrN.js";
import { a as m } from "./chunk-SDMTFDEM-n5DHnLs2.js";
import { D as h } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { V as l, a as p, t as y } from "./chunk-WG46WHJF-JdNs68CV.js";
d();
var r = h(p), v = [r.styles, m], s = class extends r {
  constructor() {
    super(), this.unselectAnnotation = () => {
      this.selectedAnnotation = null;
    }, this.unselectAnnotationEsc = (t) => {
      t && (t == null ? void 0 : t.key) !== "Escape" || (this.selectedAnnotation = null);
    }, this.selectAnnotation = ({ detail: t }) => {
      let { uuid: n } = t;
      if (this.selectedAnnotation === n) {
        this.selectedAnnotation = null;
        return;
      }
      this.selectedAnnotation = n;
    }, this.checkLastAnnotation = (t) => {
      var n, i;
      let o = this.annotations.filter((a) => a.resolved), e = this.annotations.filter((a) => !a.resolved);
      return this.annotationFilter === "All comments" ? t === ((n = o[o.length - 1]) == null ? void 0 : n.uuid) : t === ((i = e[e.length - 1]) == null ? void 0 : i.uuid);
    }, this.annotations = [], this.participantsList = [];
  }
  updated(t) {
    super.updated(t), this.updateComplete.then(() => {
      let n = this.shadowRoot.querySelector(`[uuid='${this.selectedAnnotation}']`);
      if (!n) return;
      let i = this.lastCommentId === this.selectedAnnotation, o = i ? 0 : -150, e = n.getClientRects()[0];
      e && (this.scrollBy(0, e.y + o), i && setTimeout(() => {
        this.scrollBy(0, e.y + o);
      }, 200));
    });
  }
  connectedCallback() {
    super.connectedCallback(), window.document.body.addEventListener("select-annotation", this.selectAnnotation), window.document.body.addEventListener("keyup", this.unselectAnnotationEsc), window.document.body.addEventListener("unselect-annotation", this.unselectAnnotation);
  }
  disconnectedCallback() {
    super.disconnectedCallback(), window.document.body.removeEventListener("select-annotation", this.selectAnnotation), window.document.body.removeEventListener("keyup", this.unselectAnnotationEsc), window.document.body.removeEventListener("unselect-annotation", this.unselectAnnotation);
  }
  render() {
    return l` ${c(this.annotations.filter((t) => {
      var n;
      return (n = t.comments) == null ? void 0 : n.length;
    }), (t) => t.uuid, (t) => l`
        <superviz-comments-annotation-item
          annotation=${JSON.stringify(t)}
          selected="${this.selectedAnnotation}"
          annotationFilter=${this.annotationFilter}
          uuid=${t.uuid}
          ?resolved=${t.resolved}
          isLastComment=${this.checkLastAnnotation(t.uuid)}
          participantsList=${JSON.stringify(this.participantsList)}
        >
        </superviz-comments-annotation-item>
      `)}`;
  }
};
s.styles = v, s.properties = { annotations: { type: Object }, selectedAnnotation: { type: String }, annotationFilter: { type: String }, participantsList: { type: Object } }, s = u([y("superviz-comments-content")], s);
export {
  s as CommentsContent
};
