// @superviz/react-sdk/dist/dropdown-OIZAL2LO-CPX8Gnyu.js
import { u as p } from "./chunk-K5TMEGOR-DaSvC8Mu.js";
import "./chunk-3MEZQQ2J-CHe_tQrN.js";
import { n as m } from "./chunk-6U6BYUHV-TZ4JwLT4.js";
import { D as u } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { C as g, V as n, a as f, t as y } from "./chunk-WG46WHJF-JdNs68CV.js";
import { y as h, g as w, A as v, e as b } from "./index-CtPfqlcy.js";
h();
h();
var x = g`
  .dropdown {
    position: relative;
    z-index: 100;
  }

  .dropdown-content {
    display: flex;
  }

  .select-clicked {
    border: 2px #26489a solid;
  }

  .dropdown-list {
    position: relative;
    z-index: 101;
  }

  .header {
    display: grid;
    grid-template-rows: 1fr 1px;
    align-items: center;
    padding: 0 10px;
    min-height: 42px;
    font-size: 16px;
    color: rgb(var(--sv-gray-600));
  }

  .menu {
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
    opacity: 0;
    display: none;
    background: #fff;
    padding: 0;
    z-index: 1;
    transition: 0.2s;
    border-radius: 3px;
  }

  .items {
    list-style: none;
    padding: 0;
    color: #9fa5b5;
    margin: 0;
  }

  .text.username {
    font-size: 14px;
    text-align: left;

    display: inline-block;
    vertical-align: middle;
    line-height: normal;
    padding: 8px 0;
  }

  .items li {
    color: rgb(var(--sv-gray-600));
    text-transform: uppercase;
    padding: 0 10px;
    cursor: pointer;
    min-width: 103px;
    min-height: 32px;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    gap: 5px;
    height: 42px;
  }

  .sv-icon {
    width: 40px;
    height: 40px;
    background-color: red;
  }

  .active {
    background: rgb(var(--sv-gray-200));
  }

  .items li:hover {
    background: rgb(var(--sv-gray-200));
  }

  .menu-open {
    display: block;
    opacity: 1;
    position: fixed;
  }

  .sv-hr {
    width: 100%;
    height: 1px;
    background: rgb(var(--sv-gray-300));
    padding: 0px;
    margin: 0px;
    justify-self: flex-end;
  }

  .option-label {
    white-space: nowrap;
    line-height: 1;
  }

  .who-is-online__controls__item__icon {
    line-height: 1;
  }

  .who-is-online__controls__item {
    align-items: center;
  }

  @media (max-width: 780px) {
    .menu--top-left,
    .menu--top-center,
    .menu--top-right {
      bottom: 36px;
    }
  }
`, d = u(f), $ = [d.styles, x], a = class extends d {
  constructor() {
    super(), this.menu = void 0, this.onClickOutDropdown = (t) => {
      if (t.stopPropagation(), !this.open) return;
      let e = t.composedPath(), i = this.shadowRoot.querySelector(".dropdown-content"), o = this.shadowRoot.querySelector(".dropdown-list"), s = this.shadowRoot.querySelector('slot[name="dropdown"]').assignedElements()[0], r = e.includes(i), l = e.includes(o), c = e.includes(s);
      r || l || c || this.close();
    }, this.close = () => {
      this.open = !1, this.emitEvent("close", { bubbles: !1, composed: !1 });
    }, this.callbackSelected = ({ label: t }) => {
      this.open = !1, this.emitEvent("selected", w(v({}, this.returnData), { label: t }), { bubbles: !1, composed: !0 });
    }, this.adjustPosition = () => {
      if (!this.open) {
        cancelAnimationFrame(this.animationFrame);
        return;
      }
      this.setHorizontalPosition(), this.setPositionVertical(), this.animationFrame = requestAnimationFrame(this.adjustPosition);
    }, this.tooltip = () => {
      if (!this.canShowTooltip) return "";
      let t = this.lastParticipant ? "tooltip-top" : "tooltip-bottom";
      return n` <superviz-tooltip
      tooltipData=${JSON.stringify(this.tooltipData)}
      ?shiftTooltipLeft=${this.shiftTooltipLeft}
      tooltipVerticalPosition=${t}
      classesPrefix="${this.tooltipPrefix}__tooltip"
      parentComponent=${this.parentComponent}
    ></superviz-tooltip>`;
    }, this.showTooltip = !1, this.returnData = {};
  }
  firstUpdated(t) {
    super.firstUpdated(t), this.updateComplete.then(() => {
      this.menu = this.shadowRoot.querySelector(".menu"), m.call(this, [this.parentComponent]);
    });
  }
  disconnectedCallback() {
    super.disconnectedCallback(), document.removeEventListener("click", this.onClickOutDropdown);
    let t = this.shadowRoot.querySelector(".dropdown");
    t == null || t.removeEventListener("mouseenter", () => {
      this.showTooltip = !0;
    }), t == null || t.removeEventListener("mouseleave", () => {
      this.showTooltip = !1;
    });
  }
  updated(t) {
    if (t.has("open")) {
      if (this.emitEvent("toggle-dropdown-state", { open: this.open, font: this.name }, { bubbles: !1, composed: !1 }), this.open) {
        document.addEventListener("click", this.onClickOutDropdown), this.animationFrame = requestAnimationFrame(this.adjustPosition);
        return;
      }
      document.removeEventListener("click", this.onClickOutDropdown);
    }
  }
  setHorizontalPosition() {
    let t = this.shadowRoot.querySelector('slot[name="dropdown"]'), { left: e, right: i, width: o } = t.parentElement.getBoundingClientRect(), { width: s } = this.menu.getBoundingClientRect(), r = e - (s - o) / 2 < 0, l = i + (s - o) / 2 > window.innerWidth;
    if (this.shiftTooltipLeft) {
      this.menu.style.left = `${e + 8}px`, this.menu.style.right = "", this.menu.style.transform = "translate(0, 0)";
      return;
    }
    if (!r && !l) {
      this.menu.style.left = `${e}px`, this.menu.style.right = "", this.menu.style.transform = `translate(calc(-50% + ${o / 2}px), 0)`;
      return;
    }
    if (!r) {
      this.menu.style.left = "", this.menu.style.right = `${window.innerWidth - i}px`, this.menu.style.transform = "translate(0, 0)";
      return;
    }
    l || (this.menu.style.right = "", this.menu.style.left = `${i - o}px`, this.menu.style.transform = "translate(0, 0)");
  }
  setPositionVertical() {
    let t = this.shadowRoot.querySelector('slot[name="dropdown"]'), { top: e, bottom: i } = t.parentElement.getBoundingClientRect(), { height: o } = this.menu.getBoundingClientRect(), s = this.shiftTooltipLeft ? -2 : 8;
    if (!(i + o + s > window.innerHeight)) {
      this.menu.style.bottom = "", this.menu.style.top = `${i + s}px`;
      return;
    }
    this.menu.style.top = "auto", this.menu.style.bottom = `${window.innerHeight - e + s}px`;
  }
  get renderHeader() {
    return this.name ? n` <div class="header ${this.getClass("header")}">
      <span class="text username ${this.getClass("title")}">${this.name}</span>
      <span class="sv-hr ${this.getClass("divisor")}"></span>
    </div>` : n``;
  }
  toggle() {
    this.disabled || (this.open = !this.open, this.emitEvent("open", { open: this.open }));
  }
  getIcon(t) {
    if (t) return n`
      <span class=${this.getClass("item__icon")}>
        <superviz-icon name="${t}" size="sm"></superviz-icon>
      </span>
    `;
  }
  getLabel(t) {
    return n`<span class="option-label ${this.getClass("item__label")}">${t}</span>`;
  }
  get listOptions() {
    return this.options.map(({ label: t, icon: e, active: i }) => {
      let o = { text: !0, [this.getClass("item")]: !0, "text-bold": !0, active: i };
      return n`<li @click=${() => this.callbackSelected({ label: t })} class=${p(o)}>
        ${this.getIcon(e)} ${this.getLabel(t)}
      </li>`;
    });
  }
  getClass(t) {
    return `${this.classesPrefix}__${t}`;
  }
  render() {
    let t = { menu: !0, "menu-open": this.open, "menu-left": this.align === "left", "menu-right": this.align === "right", "who-is-online-dropdown": this.name };
    return n`
      <div class="dropdown">
        <div class="dropdown-content" @click=${this.toggle}>
          <slot name="dropdown"></slot>
        </div>
        ${this.tooltip()}
      </div>
      <div class="dropdown-list">
        <div class=${p(t)}>
          ${this.renderHeader}
          <ul class="${this.getClass("items")} items">
            ${this.listOptions}
          </ul>
        </div>
      </div>
    `;
  }
};
a.styles = $, a.properties = { open: { type: Boolean }, disabled: { type: Boolean }, align: { type: String }, options: { type: Array }, icons: { type: Array }, name: { type: String }, tooltipData: { type: Object }, showTooltip: { type: Boolean }, dropdown: { type: Object }, canShowTooltip: { type: Boolean }, drodpdownSizes: { type: Object }, shiftTooltipLeft: { type: Boolean }, lastParticipant: { type: Boolean }, classesPrefix: { type: String }, parentComponent: { type: String }, tooltipPrefix: { type: String }, returnData: { type: Object } }, a = b([y("superviz-dropdown")], a);
export {
  a as Dropdown
};
