// @superviz/react-sdk/dist/mention-list-PEJOF3VQ-Cqe2LibN.js
import { j as n } from "./chunk-DX57ODX6-DAIpi7BW.js";
import "./chunk-3MEZQQ2J-CHe_tQrN.js";
import { g as r } from "./chunk-SDMTFDEM-n5DHnLs2.js";
import { D as l } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { V as i, a as p, t as d } from "./chunk-WG46WHJF-JdNs68CV.js";
import { y as m, e as h } from "./index-CtPfqlcy.js";
m();
var a = l(p), v = [a.styles, r], s = class extends a {
  constructor() {
    super(), this.showMentionList = () => {
      var e;
      let t = (e = this.shadowRoot) == null ? void 0 : e.getElementById("mention-list");
      t == null || t.style.setProperty("display", "block"), t == null || t.style.setProperty("margin-top", "1px"), t.addEventListener("wheel", this.stopHandleZoom);
    }, this.hideMentionList = () => {
      var e;
      let t = (e = this.shadowRoot) == null ? void 0 : e.getElementById("mention-list");
      t.removeEventListener("wheel", this.stopHandleZoom), t == null || t.style.setProperty("display", "none");
    }, this.selectParticipant = (e) => {
      this.emitEvent("participant-selected", e, { bubbles: !1, composed: !1 }), this.hideMentionList();
    }, this.stopHandleZoom = (e) => {
      var t;
      let o = (t = this.shadowRoot) == null ? void 0 : t.getElementById("mention-list");
      o.scrollTop += e.deltaY, e.preventDefault();
    }, this.participants = [];
  }
  updated(e) {
    if (e.has("participants") && this.participants.length > 0) {
      this.showMentionList();
      return;
    }
    this.hideMentionList();
  }
  getAvatar(e) {
    var t;
    return e.avatar ? i`<img class="avatar" src=${e.avatar} />` : i`<div class="default-avatar">
      <p class="text text-bold">${((t = e.name[0]) == null ? void 0 : t.toUpperCase()) || "A"}</p>
    </div>`;
  }
  render() {
    let e = (t) => i`
      <div class="mention-item" @click=${() => this.selectParticipant(t)}>
        ${this.getAvatar(t)}
        <div class="avatar-type">${t.name}</div>
      </div>
    `;
    return i`
      <div id="mention-list">
        ${n(this.participants, (t) => t.id, (t) => i` ${e(t)} `)}
      </div>
    `;
  }
};
s.styles = v, s.properties = { participants: { type: Object } }, s = h([d("superviz-comments-mention-list")], s);
export {
  s as CommentsMentionList
};
