// @superviz/react-sdk/dist/modal-YJCZEQ42-CoycY_Js.js
import { D as d } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { C as h, V as t, a as s, t as r } from "./chunk-WG46WHJF-JdNs68CV.js";
import { y as e, e as l } from "./index-CtPfqlcy.js";
e();
e();
e();
var c = h`
  .modal--overlay {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: rgba(var(--sv-gray-400), 0.8);

    z-index: 999;
  }

  .modal--container {
    display: flex;
    align-items: center;
    justify-content: center;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: transparent;

    z-index: 1000;
  }

  .modal--container > .modal {
    background: #fff;
    min-width: 370px;
    min-height: 150px;
    border-radius: 6px;
  }

  .modal--container > .modal > .modal--header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 8px 16px;
    background: rgb(var(--sv-gray-200));
    border-radius: 6px 6px 0 0;
  }

  .modal--container > .modal > .modal--header > .text {
    display: flex;
    align-items: center;
  }

  .modal--container > .modal > .modal--header > .close {
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 24px;
    color: rgb(var(--sv-gray-600));
    cursor: pointer;
  }

  .modal--container > .modal > .modal--body {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 24px 24px 0 24px;
  }

  .modal--container > .modal > .modal--body > .modal--body-content {
    padding: 8px 0;
    text-align: center;
  }

  .modal--container > .modal > .modal--footer {
    padding: 8px;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 24px;
    padding: 24px;
  }

  .btn {
    width: 150px;
    height: 40px;
    text-transform: uppercase;
  }

  .btn--confirm {
    background: rgb(var(--sv-primary));
    border-radius: 4px;
    border: 1px solid rgb(var(--sv-primary));
    color: #fff;
    cursor: pointer;
  }

  .btn--cancel {
    background: transparent;
    border-radius: 4px;
    border: 1px solid rgb(var(--sv-gray-400));
    color: rgb(var(--sv-gray-400));
    cursor: pointer;
  }
`, m = d(s), x = [m.styles, c], a = class extends m {
  constructor() {
    super(...arguments), this.getContainer = () => window.document.querySelector("superviz-modal-container"), this.createContainer = (o) => {
      this.modal = document.createElement("superviz-modal-container"), this.modal.setOptions(o);
    }, this.createModal = ({ detail: o }) => {
      this.createContainer(o), window.document.body.appendChild(this.modal);
    }, this.destroyModal = () => {
      var o;
      this.modal = void 0, (o = this.getContainer()) == null || o.remove();
    };
  }
  firstUpdated() {
    window.document.body.addEventListener("superviz-modal--open", this.createModal), window.document.body.addEventListener("superviz-modal--close", this.destroyModal);
  }
  disconnectedCallback() {
    this.destroyModal(), window.document.body.removeEventListener("superviz-modal--open", this.createModal), window.document.body.removeEventListener("superviz-modal--close", this.destroyModal);
  }
};
a.styles = x, a = l([r("superviz-modal")], a);
e();
var p = d(s), y = [p.styles, c], i = class extends p {
  constructor() {
    super(...arguments), this.closeModal = () => {
      window.document.body.dispatchEvent(new CustomEvent("superviz-modal--close", { bubbles: !0, composed: !0 }));
    }, this.confirmModal = () => {
      window.document.body.dispatchEvent(new CustomEvent("superviz-modal--confirm", { bubbles: !0, composed: !0 }));
    };
  }
  setOptions(o) {
    this.options = o;
  }
  render() {
    let o = () => t`
        <div class="modal--header">
          <span class="text text-bold sv-gray-600">${this.options.title}</span>
          <div class="close" @click=${this.closeModal}>
            <superviz-icon name="close" size="md"></superviz-icon>
          </div>
        </div>
      `, v = () => t`
        <div class="modal--body">
          <div class="modal--body-content">${this.options.body}</div>
        </div>
      `, b = () => {
      if (this.options.footer) return t` <div class="modal--footer">${this.options.footer}</div> `;
      let n = this.options.confirmLabel || "OK", u = this.options.cancelLabel || "Cancel";
      return this.options.confirm && this.options.cancel ? t`
          <div class="modal--footer">
            <button class="text text-bold btn btn--cancel" @click=${this.closeModal}>
              ${u}
            </button>
            <button class="text text-bold btn btn--confirm" @click=${this.confirmModal}>
              ${n}
            </button>
          </div>
        ` : t`
        <div class="modal--footer">
          <button class="text text-bold btn btn--confirm" @click=${this.confirmModal}>
            ${n}
          </button>
        </div>
      `;
    };
    return t`
      <div class="modal--overlay"></div>
      <div class="modal--container">
        <div class="modal">${o()} ${v()} ${b()}</div>
      </div>
    `;
  }
};
i.styles = y, i = l([r("superviz-modal-container")], i);
export {
  a as Modal,
  i as ModalContainer
};
