// @superviz/react-sdk/dist/room-BEOW-846.js
import aa, { createContext as p_, useContext as __, useEffect as li, useState as si, useRef as v_, useMemo as d_ } from "react";
import { a as ce, u as w_ } from "./index-CtPfqlcy.js";
var lt = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, Pu = { exports: {} }, ct = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ua;
function x_() {
  if (ua) return ct;
  ua = 1;
  var Q = aa, yn = Symbol.for("react.element"), o = Symbol.for("react.fragment"), he = Object.prototype.hasOwnProperty, ge = Q.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, We = { key: !0, ref: !0, __self: !0, __source: !0 };
  function nn(Kn, en, pe) {
    var fn, on = {}, zn = null, Fn = null;
    pe !== void 0 && (zn = "" + pe), en.key !== void 0 && (zn = "" + en.key), en.ref !== void 0 && (Fn = en.ref);
    for (fn in en) he.call(en, fn) && !We.hasOwnProperty(fn) && (on[fn] = en[fn]);
    if (Kn && Kn.defaultProps) for (fn in en = Kn.defaultProps, en) on[fn] === void 0 && (on[fn] = en[fn]);
    return { $$typeof: yn, type: Kn, key: zn, ref: Fn, props: on, _owner: ge.current };
  }
  return ct.Fragment = o, ct.jsx = nn, ct.jsxs = nn, ct;
}
var ht = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var fa;
function R_() {
  return fa || (fa = 1, process.env.NODE_ENV !== "production" && function() {
    var Q = aa, yn = Symbol.for("react.element"), o = Symbol.for("react.portal"), he = Symbol.for("react.fragment"), ge = Symbol.for("react.strict_mode"), We = Symbol.for("react.profiler"), nn = Symbol.for("react.provider"), Kn = Symbol.for("react.context"), en = Symbol.for("react.forward_ref"), pe = Symbol.for("react.suspense"), fn = Symbol.for("react.suspense_list"), on = Symbol.for("react.memo"), zn = Symbol.for("react.lazy"), Fn = Symbol.for("react.offscreen"), Wn = Symbol.iterator, Ue = "@@iterator";
    function rn(s) {
      if (s === null || typeof s != "object")
        return null;
      var w = Wn && s[Wn] || s[Ue];
      return typeof w == "function" ? w : null;
    }
    var an = Q.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function b(s) {
      {
        for (var w = arguments.length, R = new Array(w > 1 ? w - 1 : 0), C = 1; C < w; C++)
          R[C - 1] = arguments[C];
        En("error", s, R);
      }
    }
    function En(s, w, R) {
      {
        var C = an.ReactDebugCurrentFrame, N = C.getStackAddendum();
        N !== "" && (w += "%s", R = R.concat([N]));
        var H = R.map(function(U) {
          return String(U);
        });
        H.unshift("Warning: " + w), Function.prototype.apply.call(console[s], console, H);
      }
    }
    var _e = !1, Sn = !1, An = !1, dn = !1, Jn = !1, ye;
    ye = Symbol.for("react.module.reference");
    function Qe(s) {
      return !!(typeof s == "string" || typeof s == "function" || s === he || s === We || Jn || s === ge || s === pe || s === fn || dn || s === Fn || _e || Sn || An || typeof s == "object" && s !== null && (s.$$typeof === zn || s.$$typeof === on || s.$$typeof === nn || s.$$typeof === Kn || s.$$typeof === en || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      s.$$typeof === ye || s.getModuleId !== void 0));
    }
    function ke(s, w, R) {
      var C = s.displayName;
      if (C)
        return C;
      var N = w.displayName || w.name || "";
      return N !== "" ? R + "(" + N + ")" : R;
    }
    function gr(s) {
      return s.displayName || "Context";
    }
    function bn(s) {
      if (s == null)
        return null;
      if (typeof s.tag == "number" && b("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof s == "function")
        return s.displayName || s.name || null;
      if (typeof s == "string")
        return s;
      switch (s) {
        case he:
          return "Fragment";
        case o:
          return "Portal";
        case We:
          return "Profiler";
        case ge:
          return "StrictMode";
        case pe:
          return "Suspense";
        case fn:
          return "SuspenseList";
      }
      if (typeof s == "object")
        switch (s.$$typeof) {
          case Kn:
            var w = s;
            return gr(w) + ".Consumer";
          case nn:
            var R = s;
            return gr(R._context) + ".Provider";
          case en:
            return ke(s, s.render, "ForwardRef");
          case on:
            var C = s.displayName || null;
            return C !== null ? C : bn(s.type) || "Memo";
          case zn: {
            var N = s, H = N._payload, U = N._init;
            try {
              return bn(U(H));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var Un = Object.assign, Ee = 0, pr, Zn, L, On, cn, Z, ue;
    function je() {
    }
    je.__reactDisabledLog = !0;
    function hi() {
      {
        if (Ee === 0) {
          pr = console.log, Zn = console.info, L = console.warn, On = console.error, cn = console.group, Z = console.groupCollapsed, ue = console.groupEnd;
          var s = {
            configurable: !0,
            enumerable: !0,
            value: je,
            writable: !0
          };
          Object.defineProperties(console, {
            info: s,
            log: s,
            warn: s,
            error: s,
            group: s,
            groupCollapsed: s,
            groupEnd: s
          });
        }
        Ee++;
      }
    }
    function Me() {
      {
        if (Ee--, Ee === 0) {
          var s = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: Un({}, s, {
              value: pr
            }),
            info: Un({}, s, {
              value: Zn
            }),
            warn: Un({}, s, {
              value: L
            }),
            error: Un({}, s, {
              value: On
            }),
            group: Un({}, s, {
              value: cn
            }),
            groupCollapsed: Un({}, s, {
              value: Z
            }),
            groupEnd: Un({}, s, {
              value: ue
            })
          });
        }
        Ee < 0 && b("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var Be = an.ReactCurrentDispatcher, Mr;
    function ve(s, w, R) {
      {
        if (Mr === void 0)
          try {
            throw Error();
          } catch (N) {
            var C = N.stack.trim().match(/\n( *(at )?)/);
            Mr = C && C[1] || "";
          }
        return `
` + Mr + s;
      }
    }
    var Ae = !1, _r;
    {
      var vr = typeof WeakMap == "function" ? WeakMap : Map;
      _r = new vr();
    }
    function nr(s, w) {
      if (!s || Ae)
        return "";
      {
        var R = _r.get(s);
        if (R !== void 0)
          return R;
      }
      var C;
      Ae = !0;
      var N = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var H;
      H = Be.current, Be.current = null, hi();
      try {
        if (w) {
          var U = function() {
            throw Error();
          };
          if (Object.defineProperty(U.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(U, []);
            } catch (wn) {
              C = wn;
            }
            Reflect.construct(s, [], U);
          } else {
            try {
              U.call();
            } catch (wn) {
              C = wn;
            }
            s.call(U.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (wn) {
            C = wn;
          }
          s();
        }
      } catch (wn) {
        if (wn && C && typeof wn.stack == "string") {
          for (var F = wn.stack.split(`
`), gn = C.stack.split(`
`), X = F.length - 1, k = gn.length - 1; X >= 1 && k >= 0 && F[X] !== gn[k]; )
            k--;
          for (; X >= 1 && k >= 0; X--, k--)
            if (F[X] !== gn[k]) {
              if (X !== 1 || k !== 1)
                do
                  if (X--, k--, k < 0 || F[X] !== gn[k]) {
                    var pn = `
` + F[X].replace(" at new ", " at ");
                    return s.displayName && pn.includes("<anonymous>") && (pn = pn.replace("<anonymous>", s.displayName)), typeof s == "function" && _r.set(s, pn), pn;
                  }
                while (X >= 1 && k >= 0);
              break;
            }
        }
      } finally {
        Ae = !1, Be.current = H, Me(), Error.prepareStackTrace = N;
      }
      var Ge = s ? s.displayName || s.name : "", Vn = Ge ? ve(Ge) : "";
      return typeof s == "function" && _r.set(s, Vn), Vn;
    }
    function gt(s, w, R) {
      return nr(s, !1);
    }
    function Mn(s) {
      var w = s.prototype;
      return !!(w && w.isReactComponent);
    }
    function de(s, w, R) {
      if (s == null)
        return "";
      if (typeof s == "function")
        return nr(s, Mn(s));
      if (typeof s == "string")
        return ve(s);
      switch (s) {
        case pe:
          return ve("Suspense");
        case fn:
          return ve("SuspenseList");
      }
      if (typeof s == "object")
        switch (s.$$typeof) {
          case en:
            return gt(s.render);
          case on:
            return de(s.type, w, R);
          case zn: {
            var C = s, N = C._payload, H = C._init;
            try {
              return de(H(N), w, R);
            } catch {
            }
          }
        }
      return "";
    }
    var er = Object.prototype.hasOwnProperty, Xn = {}, Br = an.ReactDebugCurrentFrame;
    function dr(s) {
      if (s) {
        var w = s._owner, R = de(s.type, s._source, w ? w.type : null);
        Br.setExtraStackFrame(R);
      } else
        Br.setExtraStackFrame(null);
    }
    function rr(s, w, R, C, N) {
      {
        var H = Function.call.bind(er);
        for (var U in s)
          if (H(s, U)) {
            var F = void 0;
            try {
              if (typeof s[U] != "function") {
                var gn = Error((C || "React class") + ": " + R + " type `" + U + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof s[U] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw gn.name = "Invariant Violation", gn;
              }
              F = s[U](w, U, C, R, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (X) {
              F = X;
            }
            F && !(F instanceof Error) && (dr(N), b("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", C || "React class", R, U, typeof F), dr(null)), F instanceof Error && !(F.message in Xn) && (Xn[F.message] = !0, dr(N), b("Failed %s type: %s", R, F.message), dr(null));
          }
      }
    }
    var Bn = Array.isArray;
    function me(s) {
      return Bn(s);
    }
    function wr(s) {
      {
        var w = typeof Symbol == "function" && Symbol.toStringTag, R = w && s[Symbol.toStringTag] || s.constructor.name || "Object";
        return R;
      }
    }
    function gi(s) {
      try {
        return Ne(s), !1;
      } catch {
        return !0;
      }
    }
    function Ne(s) {
      return "" + s;
    }
    function pt(s) {
      if (gi(s))
        return b("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", wr(s)), Ne(s);
    }
    var fe = an.ReactCurrentOwner, $e = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, xr, Rr, tr;
    tr = {};
    function Nr(s) {
      if (er.call(s, "ref")) {
        var w = Object.getOwnPropertyDescriptor(s, "ref").get;
        if (w && w.isReactWarning)
          return !1;
      }
      return s.ref !== void 0;
    }
    function $r(s) {
      if (er.call(s, "key")) {
        var w = Object.getOwnPropertyDescriptor(s, "key").get;
        if (w && w.isReactWarning)
          return !1;
      }
      return s.key !== void 0;
    }
    function Gr(s, w) {
      if (typeof s.ref == "string" && fe.current && w && fe.current.stateNode !== w) {
        var R = bn(fe.current.type);
        tr[R] || (b('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', bn(fe.current.type), s.ref), tr[R] = !0);
      }
    }
    function Hr(s, w) {
      {
        var R = function() {
          xr || (xr = !0, b("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", w));
        };
        R.isReactWarning = !0, Object.defineProperty(s, "key", {
          get: R,
          configurable: !0
        });
      }
    }
    function Yr(s, w) {
      {
        var R = function() {
          Rr || (Rr = !0, b("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", w));
        };
        R.isReactWarning = !0, Object.defineProperty(s, "ref", {
          get: R,
          configurable: !0
        });
      }
    }
    var qr = function(s, w, R, C, N, H, U) {
      var F = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: yn,
        // Built-in properties that belong on the element
        type: s,
        key: w,
        ref: R,
        props: U,
        // Record the component responsible for creating this element.
        _owner: H
      };
      return F._store = {}, Object.defineProperty(F._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(F, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: C
      }), Object.defineProperty(F, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: N
      }), Object.freeze && (Object.freeze(F.props), Object.freeze(F)), F;
    };
    function pi(s, w, R, C, N) {
      {
        var H, U = {}, F = null, gn = null;
        R !== void 0 && (pt(R), F = "" + R), $r(w) && (pt(w.key), F = "" + w.key), Nr(w) && (gn = w.ref, Gr(w, N));
        for (H in w)
          er.call(w, H) && !$e.hasOwnProperty(H) && (U[H] = w[H]);
        if (s && s.defaultProps) {
          var X = s.defaultProps;
          for (H in X)
            U[H] === void 0 && (U[H] = X[H]);
        }
        if (F || gn) {
          var k = typeof s == "function" ? s.displayName || s.name || "Unknown" : s;
          F && Hr(U, k), gn && Yr(U, k);
        }
        return qr(s, F, gn, N, C, fe.current, U);
      }
    }
    var Kr = an.ReactCurrentOwner, _t = an.ReactDebugCurrentFrame;
    function Te(s) {
      if (s) {
        var w = s._owner, R = de(s.type, s._source, w ? w.type : null);
        _t.setExtraStackFrame(R);
      } else
        _t.setExtraStackFrame(null);
    }
    var yr;
    yr = !1;
    function zr(s) {
      return typeof s == "object" && s !== null && s.$$typeof === yn;
    }
    function vt() {
      {
        if (Kr.current) {
          var s = bn(Kr.current.type);
          if (s)
            return `

Check the render method of \`` + s + "`.";
        }
        return "";
      }
    }
    function _i(s) {
      return "";
    }
    var dt = {};
    function wt(s) {
      {
        var w = vt();
        if (!w) {
          var R = typeof s == "string" ? s : s.displayName || s.name;
          R && (w = `

Check the top-level render call using <` + R + ">.");
        }
        return w;
      }
    }
    function xt(s, w) {
      {
        if (!s._store || s._store.validated || s.key != null)
          return;
        s._store.validated = !0;
        var R = wt(w);
        if (dt[R])
          return;
        dt[R] = !0;
        var C = "";
        s && s._owner && s._owner !== Kr.current && (C = " It was passed a child from " + bn(s._owner.type) + "."), Te(s), b('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', R, C), Te(null);
      }
    }
    function Rt(s, w) {
      {
        if (typeof s != "object")
          return;
        if (me(s))
          for (var R = 0; R < s.length; R++) {
            var C = s[R];
            zr(C) && xt(C, w);
          }
        else if (zr(s))
          s._store && (s._store.validated = !0);
        else if (s) {
          var N = rn(s);
          if (typeof N == "function" && N !== s.entries)
            for (var H = N.call(s), U; !(U = H.next()).done; )
              zr(U.value) && xt(U.value, w);
        }
      }
    }
    function vi(s) {
      {
        var w = s.type;
        if (w == null || typeof w == "string")
          return;
        var R;
        if (typeof w == "function")
          R = w.propTypes;
        else if (typeof w == "object" && (w.$$typeof === en || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        w.$$typeof === on))
          R = w.propTypes;
        else
          return;
        if (R) {
          var C = bn(w);
          rr(R, s.props, "prop", C, s);
        } else if (w.PropTypes !== void 0 && !yr) {
          yr = !0;
          var N = bn(w);
          b("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", N || "Unknown");
        }
        typeof w.getDefaultProps == "function" && !w.getDefaultProps.isReactClassApproved && b("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Jr(s) {
      {
        for (var w = Object.keys(s.props), R = 0; R < w.length; R++) {
          var C = w[R];
          if (C !== "children" && C !== "key") {
            Te(s), b("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", C), Te(null);
            break;
          }
        }
        s.ref !== null && (Te(s), b("Invalid attribute `ref` supplied to `React.Fragment`."), Te(null));
      }
    }
    var yt = {};
    function Er(s, w, R, C, N, H) {
      {
        var U = Qe(s);
        if (!U) {
          var F = "";
          (s === void 0 || typeof s == "object" && s !== null && Object.keys(s).length === 0) && (F += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var gn = _i();
          gn ? F += gn : F += vt();
          var X;
          s === null ? X = "null" : me(s) ? X = "array" : s !== void 0 && s.$$typeof === yn ? (X = "<" + (bn(s.type) || "Unknown") + " />", F = " Did you accidentally export a JSX literal instead of a component?") : X = typeof s, b("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", X, F);
        }
        var k = pi(s, w, R, N, H);
        if (k == null)
          return k;
        if (U) {
          var pn = w.children;
          if (pn !== void 0)
            if (C)
              if (me(pn)) {
                for (var Ge = 0; Ge < pn.length; Ge++)
                  Rt(pn[Ge], s);
                Object.freeze && Object.freeze(pn);
              } else
                b("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Rt(pn, s);
        }
        if (er.call(w, "key")) {
          var Vn = bn(s), wn = Object.keys(w).filter(function(Et) {
            return Et !== "key";
          }), Zr = wn.length > 0 ? "{key: someKey, " + wn.join(": ..., ") + ": ...}" : "{key: someKey}";
          if (!yt[Vn + Zr]) {
            var yi = wn.length > 0 ? "{" + wn.join(": ..., ") + ": ...}" : "{}";
            b(`A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`, Zr, Vn, yi, Vn), yt[Vn + Zr] = !0;
          }
        }
        return s === he ? Jr(k) : vi(k), k;
      }
    }
    function di(s, w, R) {
      return Er(s, w, R, !0);
    }
    function wi(s, w, R) {
      return Er(s, w, R, !1);
    }
    var xi = wi, Ri = di;
    ht.Fragment = he, ht.jsx = xi, ht.jsxs = Ri;
  }()), ht;
}
process.env.NODE_ENV === "production" ? Pu.exports = x_() : Pu.exports = R_();
var y_ = Pu.exports, ci = { exports: {} };
/**
 * @license
 * Lodash <https://lodash.com/>
 * Copyright OpenJS Foundation and other contributors <https://openjsf.org/>
 * Released under MIT license <https://lodash.com/license>
 * Based on Underscore.js 1.8.3 <http://underscorejs.org/LICENSE>
 * Copyright Jeremy Ashkenas, DocumentCloud and Investigative Reporters & Editors
 */
ci.exports;
(function(Q, yn) {
  (function() {
    var o, he = "4.17.21", ge = 200, We = "Unsupported core-js use. Try https://npms.io/search?q=ponyfill.", nn = "Expected a function", Kn = "Invalid `variable` option passed into `_.template`", en = "__lodash_hash_undefined__", pe = 500, fn = "__lodash_placeholder__", on = 1, zn = 2, Fn = 4, Wn = 1, Ue = 2, rn = 1, an = 2, b = 4, En = 8, _e = 16, Sn = 32, An = 64, dn = 128, Jn = 256, ye = 512, Qe = 30, ke = "...", gr = 800, bn = 16, Un = 1, Ee = 2, pr = 3, Zn = 1 / 0, L = 9007199254740991, On = 17976931348623157e292, cn = NaN, Z = 4294967295, ue = Z - 1, je = Z >>> 1, hi = [
      ["ary", dn],
      ["bind", rn],
      ["bindKey", an],
      ["curry", En],
      ["curryRight", _e],
      ["flip", ye],
      ["partial", Sn],
      ["partialRight", An],
      ["rearg", Jn]
    ], Me = "[object Arguments]", Be = "[object Array]", Mr = "[object AsyncFunction]", ve = "[object Boolean]", Ae = "[object Date]", _r = "[object DOMException]", vr = "[object Error]", nr = "[object Function]", gt = "[object GeneratorFunction]", Mn = "[object Map]", de = "[object Number]", er = "[object Null]", Xn = "[object Object]", Br = "[object Promise]", dr = "[object Proxy]", rr = "[object RegExp]", Bn = "[object Set]", me = "[object String]", wr = "[object Symbol]", gi = "[object Undefined]", Ne = "[object WeakMap]", pt = "[object WeakSet]", fe = "[object ArrayBuffer]", $e = "[object DataView]", xr = "[object Float32Array]", Rr = "[object Float64Array]", tr = "[object Int8Array]", Nr = "[object Int16Array]", $r = "[object Int32Array]", Gr = "[object Uint8Array]", Hr = "[object Uint8ClampedArray]", Yr = "[object Uint16Array]", qr = "[object Uint32Array]", pi = /\b__p \+= '';/g, Kr = /\b(__p \+=) '' \+/g, _t = /(__e\(.*?\)|\b__t\)) \+\n'';/g, Te = /&(?:amp|lt|gt|quot|#39);/g, yr = /[&<>"']/g, zr = RegExp(Te.source), vt = RegExp(yr.source), _i = /<%-([\s\S]+?)%>/g, dt = /<%([\s\S]+?)%>/g, wt = /<%=([\s\S]+?)%>/g, xt = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, Rt = /^\w*$/, vi = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, Jr = /[\\^$.*+?()[\]{}|]/g, yt = RegExp(Jr.source), Er = /^\s+/, di = /\s/, wi = /\{(?:\n\/\* \[wrapped with .+\] \*\/)?\n?/, xi = /\{\n\/\* \[wrapped with (.+)\] \*/, Ri = /,? & /, s = /[^\x00-\x2f\x3a-\x40\x5b-\x60\x7b-\x7f]+/g, w = /[()=,{}\[\]\/\s]/, R = /\\(\\)?/g, C = /\$\{([^\\}]*(?:\\.[^\\}]*)*)\}/g, N = /\w*$/, H = /^[-+]0x[0-9a-f]+$/i, U = /^0b[01]+$/i, F = /^\[object .+?Constructor\]$/, gn = /^0o[0-7]+$/i, X = /^(?:0|[1-9]\d*)$/, k = /[\xc0-\xd6\xd8-\xf6\xf8-\xff\u0100-\u017f]/g, pn = /($^)/, Ge = /['\n\r\u2028\u2029\\]/g, Vn = "\\ud800-\\udfff", wn = "\\u0300-\\u036f", Zr = "\\ufe20-\\ufe2f", yi = "\\u20d0-\\u20ff", Et = wn + Zr + yi, Du = "\\u2700-\\u27bf", Fu = "a-z\\xdf-\\xf6\\xf8-\\xff", la = "\\xac\\xb1\\xd7\\xf7", ca = "\\x00-\\x2f\\x3a-\\x40\\x5b-\\x60\\x7b-\\xbf", ha = "\\u2000-\\u206f", ga = " \\t\\x0b\\f\\xa0\\ufeff\\n\\r\\u2028\\u2029\\u1680\\u180e\\u2000\\u2001\\u2002\\u2003\\u2004\\u2005\\u2006\\u2007\\u2008\\u2009\\u200a\\u202f\\u205f\\u3000", Wu = "A-Z\\xc0-\\xd6\\xd8-\\xde", Uu = "\\ufe0e\\ufe0f", Mu = la + ca + ha + ga, Ei = "['’]", pa = "[" + Vn + "]", Bu = "[" + Mu + "]", At = "[" + Et + "]", Nu = "\\d+", _a = "[" + Du + "]", $u = "[" + Fu + "]", Gu = "[^" + Vn + Mu + Nu + Du + Fu + Wu + "]", Ai = "\\ud83c[\\udffb-\\udfff]", va = "(?:" + At + "|" + Ai + ")", Hu = "[^" + Vn + "]", mi = "(?:\\ud83c[\\udde6-\\uddff]){2}", Ti = "[\\ud800-\\udbff][\\udc00-\\udfff]", Ar = "[" + Wu + "]", Yu = "\\u200d", qu = "(?:" + $u + "|" + Gu + ")", da = "(?:" + Ar + "|" + Gu + ")", Ku = "(?:" + Ei + "(?:d|ll|m|re|s|t|ve))?", zu = "(?:" + Ei + "(?:D|LL|M|RE|S|T|VE))?", Ju = va + "?", Zu = "[" + Uu + "]?", wa = "(?:" + Yu + "(?:" + [Hu, mi, Ti].join("|") + ")" + Zu + Ju + ")*", xa = "\\d*(?:1st|2nd|3rd|(?![123])\\dth)(?=\\b|[A-Z_])", Ra = "\\d*(?:1ST|2ND|3RD|(?![123])\\dTH)(?=\\b|[a-z_])", Xu = Zu + Ju + wa, ya = "(?:" + [_a, mi, Ti].join("|") + ")" + Xu, Ea = "(?:" + [Hu + At + "?", At, mi, Ti, pa].join("|") + ")", Aa = RegExp(Ei, "g"), ma = RegExp(At, "g"), Ci = RegExp(Ai + "(?=" + Ai + ")|" + Ea + Xu, "g"), Ta = RegExp([
      Ar + "?" + $u + "+" + Ku + "(?=" + [Bu, Ar, "$"].join("|") + ")",
      da + "+" + zu + "(?=" + [Bu, Ar + qu, "$"].join("|") + ")",
      Ar + "?" + qu + "+" + Ku,
      Ar + "+" + zu,
      Ra,
      xa,
      Nu,
      ya
    ].join("|"), "g"), Ca = RegExp("[" + Yu + Vn + Et + Uu + "]"), Sa = /[a-z][A-Z]|[A-Z]{2}[a-z]|[0-9][a-zA-Z]|[a-zA-Z][0-9]|[^a-zA-Z0-9 ]/, ba = [
      "Array",
      "Buffer",
      "DataView",
      "Date",
      "Error",
      "Float32Array",
      "Float64Array",
      "Function",
      "Int8Array",
      "Int16Array",
      "Int32Array",
      "Map",
      "Math",
      "Object",
      "Promise",
      "RegExp",
      "Set",
      "String",
      "Symbol",
      "TypeError",
      "Uint8Array",
      "Uint8ClampedArray",
      "Uint16Array",
      "Uint32Array",
      "WeakMap",
      "_",
      "clearTimeout",
      "isFinite",
      "parseInt",
      "setTimeout"
    ], Oa = -1, z = {};
    z[xr] = z[Rr] = z[tr] = z[Nr] = z[$r] = z[Gr] = z[Hr] = z[Yr] = z[qr] = !0, z[Me] = z[Be] = z[fe] = z[ve] = z[$e] = z[Ae] = z[vr] = z[nr] = z[Mn] = z[de] = z[Xn] = z[rr] = z[Bn] = z[me] = z[Ne] = !1;
    var K = {};
    K[Me] = K[Be] = K[fe] = K[$e] = K[ve] = K[Ae] = K[xr] = K[Rr] = K[tr] = K[Nr] = K[$r] = K[Mn] = K[de] = K[Xn] = K[rr] = K[Bn] = K[me] = K[wr] = K[Gr] = K[Hr] = K[Yr] = K[qr] = !0, K[vr] = K[nr] = K[Ne] = !1;
    var La = {
      // Latin-1 Supplement block.
      À: "A",
      Á: "A",
      Â: "A",
      Ã: "A",
      Ä: "A",
      Å: "A",
      à: "a",
      á: "a",
      â: "a",
      ã: "a",
      ä: "a",
      å: "a",
      Ç: "C",
      ç: "c",
      Ð: "D",
      ð: "d",
      È: "E",
      É: "E",
      Ê: "E",
      Ë: "E",
      è: "e",
      é: "e",
      ê: "e",
      ë: "e",
      Ì: "I",
      Í: "I",
      Î: "I",
      Ï: "I",
      ì: "i",
      í: "i",
      î: "i",
      ï: "i",
      Ñ: "N",
      ñ: "n",
      Ò: "O",
      Ó: "O",
      Ô: "O",
      Õ: "O",
      Ö: "O",
      Ø: "O",
      ò: "o",
      ó: "o",
      ô: "o",
      õ: "o",
      ö: "o",
      ø: "o",
      Ù: "U",
      Ú: "U",
      Û: "U",
      Ü: "U",
      ù: "u",
      ú: "u",
      û: "u",
      ü: "u",
      Ý: "Y",
      ý: "y",
      ÿ: "y",
      Æ: "Ae",
      æ: "ae",
      Þ: "Th",
      þ: "th",
      ß: "ss",
      // Latin Extended-A block.
      Ā: "A",
      Ă: "A",
      Ą: "A",
      ā: "a",
      ă: "a",
      ą: "a",
      Ć: "C",
      Ĉ: "C",
      Ċ: "C",
      Č: "C",
      ć: "c",
      ĉ: "c",
      ċ: "c",
      č: "c",
      Ď: "D",
      Đ: "D",
      ď: "d",
      đ: "d",
      Ē: "E",
      Ĕ: "E",
      Ė: "E",
      Ę: "E",
      Ě: "E",
      ē: "e",
      ĕ: "e",
      ė: "e",
      ę: "e",
      ě: "e",
      Ĝ: "G",
      Ğ: "G",
      Ġ: "G",
      Ģ: "G",
      ĝ: "g",
      ğ: "g",
      ġ: "g",
      ģ: "g",
      Ĥ: "H",
      Ħ: "H",
      ĥ: "h",
      ħ: "h",
      Ĩ: "I",
      Ī: "I",
      Ĭ: "I",
      Į: "I",
      İ: "I",
      ĩ: "i",
      ī: "i",
      ĭ: "i",
      į: "i",
      ı: "i",
      Ĵ: "J",
      ĵ: "j",
      Ķ: "K",
      ķ: "k",
      ĸ: "k",
      Ĺ: "L",
      Ļ: "L",
      Ľ: "L",
      Ŀ: "L",
      Ł: "L",
      ĺ: "l",
      ļ: "l",
      ľ: "l",
      ŀ: "l",
      ł: "l",
      Ń: "N",
      Ņ: "N",
      Ň: "N",
      Ŋ: "N",
      ń: "n",
      ņ: "n",
      ň: "n",
      ŋ: "n",
      Ō: "O",
      Ŏ: "O",
      Ő: "O",
      ō: "o",
      ŏ: "o",
      ő: "o",
      Ŕ: "R",
      Ŗ: "R",
      Ř: "R",
      ŕ: "r",
      ŗ: "r",
      ř: "r",
      Ś: "S",
      Ŝ: "S",
      Ş: "S",
      Š: "S",
      ś: "s",
      ŝ: "s",
      ş: "s",
      š: "s",
      Ţ: "T",
      Ť: "T",
      Ŧ: "T",
      ţ: "t",
      ť: "t",
      ŧ: "t",
      Ũ: "U",
      Ū: "U",
      Ŭ: "U",
      Ů: "U",
      Ű: "U",
      Ų: "U",
      ũ: "u",
      ū: "u",
      ŭ: "u",
      ů: "u",
      ű: "u",
      ų: "u",
      Ŵ: "W",
      ŵ: "w",
      Ŷ: "Y",
      ŷ: "y",
      Ÿ: "Y",
      Ź: "Z",
      Ż: "Z",
      Ž: "Z",
      ź: "z",
      ż: "z",
      ž: "z",
      Ĳ: "IJ",
      ĳ: "ij",
      Œ: "Oe",
      œ: "oe",
      ŉ: "'n",
      ſ: "s"
    }, Ia = {
      "&": "&amp;",
      "<": "&lt;",
      ">": "&gt;",
      '"': "&quot;",
      "'": "&#39;"
    }, Pa = {
      "&amp;": "&",
      "&lt;": "<",
      "&gt;": ">",
      "&quot;": '"',
      "&#39;": "'"
    }, Da = {
      "\\": "\\",
      "'": "'",
      "\n": "n",
      "\r": "r",
      "\u2028": "u2028",
      "\u2029": "u2029"
    }, Fa = parseFloat, Wa = parseInt, Vu = typeof lt == "object" && lt && lt.Object === Object && lt, Ua = typeof self == "object" && self && self.Object === Object && self, _n = Vu || Ua || Function("return this")(), Si = yn && !yn.nodeType && yn, ir = Si && !0 && Q && !Q.nodeType && Q, Qu = ir && ir.exports === Si, bi = Qu && Vu.process, Qn = function() {
      try {
        var c = ir && ir.require && ir.require("util").types;
        return c || bi && bi.binding && bi.binding("util");
      } catch {
      }
    }(), ku = Qn && Qn.isArrayBuffer, ju = Qn && Qn.isDate, nf = Qn && Qn.isMap, ef = Qn && Qn.isRegExp, rf = Qn && Qn.isSet, tf = Qn && Qn.isTypedArray;
    function Nn(c, p, g) {
      switch (g.length) {
        case 0:
          return c.call(p);
        case 1:
          return c.call(p, g[0]);
        case 2:
          return c.call(p, g[0], g[1]);
        case 3:
          return c.call(p, g[0], g[1], g[2]);
      }
      return c.apply(p, g);
    }
    function Ma(c, p, g, y) {
      for (var S = -1, $ = c == null ? 0 : c.length; ++S < $; ) {
        var sn = c[S];
        p(y, sn, g(sn), c);
      }
      return y;
    }
    function kn(c, p) {
      for (var g = -1, y = c == null ? 0 : c.length; ++g < y && p(c[g], g, c) !== !1; )
        ;
      return c;
    }
    function Ba(c, p) {
      for (var g = c == null ? 0 : c.length; g-- && p(c[g], g, c) !== !1; )
        ;
      return c;
    }
    function uf(c, p) {
      for (var g = -1, y = c == null ? 0 : c.length; ++g < y; )
        if (!p(c[g], g, c))
          return !1;
      return !0;
    }
    function He(c, p) {
      for (var g = -1, y = c == null ? 0 : c.length, S = 0, $ = []; ++g < y; ) {
        var sn = c[g];
        p(sn, g, c) && ($[S++] = sn);
      }
      return $;
    }
    function mt(c, p) {
      var g = c == null ? 0 : c.length;
      return !!g && mr(c, p, 0) > -1;
    }
    function Oi(c, p, g) {
      for (var y = -1, S = c == null ? 0 : c.length; ++y < S; )
        if (g(p, c[y]))
          return !0;
      return !1;
    }
    function J(c, p) {
      for (var g = -1, y = c == null ? 0 : c.length, S = Array(y); ++g < y; )
        S[g] = p(c[g], g, c);
      return S;
    }
    function Ye(c, p) {
      for (var g = -1, y = p.length, S = c.length; ++g < y; )
        c[S + g] = p[g];
      return c;
    }
    function Li(c, p, g, y) {
      var S = -1, $ = c == null ? 0 : c.length;
      for (y && $ && (g = c[++S]); ++S < $; )
        g = p(g, c[S], S, c);
      return g;
    }
    function Na(c, p, g, y) {
      var S = c == null ? 0 : c.length;
      for (y && S && (g = c[--S]); S--; )
        g = p(g, c[S], S, c);
      return g;
    }
    function Ii(c, p) {
      for (var g = -1, y = c == null ? 0 : c.length; ++g < y; )
        if (p(c[g], g, c))
          return !0;
      return !1;
    }
    var $a = Pi("length");
    function Ga(c) {
      return c.split("");
    }
    function Ha(c) {
      return c.match(s) || [];
    }
    function ff(c, p, g) {
      var y;
      return g(c, function(S, $, sn) {
        if (p(S, $, sn))
          return y = $, !1;
      }), y;
    }
    function Tt(c, p, g, y) {
      for (var S = c.length, $ = g + (y ? 1 : -1); y ? $-- : ++$ < S; )
        if (p(c[$], $, c))
          return $;
      return -1;
    }
    function mr(c, p, g) {
      return p === p ? ns(c, p, g) : Tt(c, of, g);
    }
    function Ya(c, p, g, y) {
      for (var S = g - 1, $ = c.length; ++S < $; )
        if (y(c[S], p))
          return S;
      return -1;
    }
    function of(c) {
      return c !== c;
    }
    function af(c, p) {
      var g = c == null ? 0 : c.length;
      return g ? Fi(c, p) / g : cn;
    }
    function Pi(c) {
      return function(p) {
        return p == null ? o : p[c];
      };
    }
    function Di(c) {
      return function(p) {
        return c == null ? o : c[p];
      };
    }
    function sf(c, p, g, y, S) {
      return S(c, function($, sn, q) {
        g = y ? (y = !1, $) : p(g, $, sn, q);
      }), g;
    }
    function qa(c, p) {
      var g = c.length;
      for (c.sort(p); g--; )
        c[g] = c[g].value;
      return c;
    }
    function Fi(c, p) {
      for (var g, y = -1, S = c.length; ++y < S; ) {
        var $ = p(c[y]);
        $ !== o && (g = g === o ? $ : g + $);
      }
      return g;
    }
    function Wi(c, p) {
      for (var g = -1, y = Array(c); ++g < c; )
        y[g] = p(g);
      return y;
    }
    function Ka(c, p) {
      return J(p, function(g) {
        return [g, c[g]];
      });
    }
    function lf(c) {
      return c && c.slice(0, pf(c) + 1).replace(Er, "");
    }
    function $n(c) {
      return function(p) {
        return c(p);
      };
    }
    function Ui(c, p) {
      return J(p, function(g) {
        return c[g];
      });
    }
    function Xr(c, p) {
      return c.has(p);
    }
    function cf(c, p) {
      for (var g = -1, y = c.length; ++g < y && mr(p, c[g], 0) > -1; )
        ;
      return g;
    }
    function hf(c, p) {
      for (var g = c.length; g-- && mr(p, c[g], 0) > -1; )
        ;
      return g;
    }
    function za(c, p) {
      for (var g = c.length, y = 0; g--; )
        c[g] === p && ++y;
      return y;
    }
    var Ja = Di(La), Za = Di(Ia);
    function Xa(c) {
      return "\\" + Da[c];
    }
    function Va(c, p) {
      return c == null ? o : c[p];
    }
    function Tr(c) {
      return Ca.test(c);
    }
    function Qa(c) {
      return Sa.test(c);
    }
    function ka(c) {
      for (var p, g = []; !(p = c.next()).done; )
        g.push(p.value);
      return g;
    }
    function Mi(c) {
      var p = -1, g = Array(c.size);
      return c.forEach(function(y, S) {
        g[++p] = [S, y];
      }), g;
    }
    function gf(c, p) {
      return function(g) {
        return c(p(g));
      };
    }
    function qe(c, p) {
      for (var g = -1, y = c.length, S = 0, $ = []; ++g < y; ) {
        var sn = c[g];
        (sn === p || sn === fn) && (c[g] = fn, $[S++] = g);
      }
      return $;
    }
    function Ct(c) {
      var p = -1, g = Array(c.size);
      return c.forEach(function(y) {
        g[++p] = y;
      }), g;
    }
    function ja(c) {
      var p = -1, g = Array(c.size);
      return c.forEach(function(y) {
        g[++p] = [y, y];
      }), g;
    }
    function ns(c, p, g) {
      for (var y = g - 1, S = c.length; ++y < S; )
        if (c[y] === p)
          return y;
      return -1;
    }
    function es(c, p, g) {
      for (var y = g + 1; y--; )
        if (c[y] === p)
          return y;
      return y;
    }
    function Cr(c) {
      return Tr(c) ? ts(c) : $a(c);
    }
    function oe(c) {
      return Tr(c) ? is(c) : Ga(c);
    }
    function pf(c) {
      for (var p = c.length; p-- && di.test(c.charAt(p)); )
        ;
      return p;
    }
    var rs = Di(Pa);
    function ts(c) {
      for (var p = Ci.lastIndex = 0; Ci.test(c); )
        ++p;
      return p;
    }
    function is(c) {
      return c.match(Ci) || [];
    }
    function us(c) {
      return c.match(Ta) || [];
    }
    var fs = function c(p) {
      p = p == null ? _n : Sr.defaults(_n.Object(), p, Sr.pick(_n, ba));
      var g = p.Array, y = p.Date, S = p.Error, $ = p.Function, sn = p.Math, q = p.Object, Bi = p.RegExp, os = p.String, jn = p.TypeError, St = g.prototype, as = $.prototype, br = q.prototype, bt = p["__core-js_shared__"], Ot = as.toString, Y = br.hasOwnProperty, ss = 0, _f = function() {
        var n = /[^.]+$/.exec(bt && bt.keys && bt.keys.IE_PROTO || "");
        return n ? "Symbol(src)_1." + n : "";
      }(), Lt = br.toString, ls = Ot.call(q), cs = _n._, hs = Bi(
        "^" + Ot.call(Y).replace(Jr, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
      ), It = Qu ? p.Buffer : o, Ke = p.Symbol, Pt = p.Uint8Array, vf = It ? It.allocUnsafe : o, Dt = gf(q.getPrototypeOf, q), df = q.create, wf = br.propertyIsEnumerable, Ft = St.splice, xf = Ke ? Ke.isConcatSpreadable : o, Vr = Ke ? Ke.iterator : o, ur = Ke ? Ke.toStringTag : o, Wt = function() {
        try {
          var n = lr(q, "defineProperty");
          return n({}, "", {}), n;
        } catch {
        }
      }(), gs = p.clearTimeout !== _n.clearTimeout && p.clearTimeout, ps = y && y.now !== _n.Date.now && y.now, _s = p.setTimeout !== _n.setTimeout && p.setTimeout, Ut = sn.ceil, Mt = sn.floor, Ni = q.getOwnPropertySymbols, vs = It ? It.isBuffer : o, Rf = p.isFinite, ds = St.join, ws = gf(q.keys, q), ln = sn.max, xn = sn.min, xs = y.now, Rs = p.parseInt, yf = sn.random, ys = St.reverse, $i = lr(p, "DataView"), Qr = lr(p, "Map"), Gi = lr(p, "Promise"), Or = lr(p, "Set"), kr = lr(p, "WeakMap"), jr = lr(q, "create"), Bt = kr && new kr(), Lr = {}, Es = cr($i), As = cr(Qr), ms = cr(Gi), Ts = cr(Or), Cs = cr(kr), Nt = Ke ? Ke.prototype : o, nt = Nt ? Nt.valueOf : o, Ef = Nt ? Nt.toString : o;
      function u(n) {
        if (j(n) && !O(n) && !(n instanceof M)) {
          if (n instanceof ne)
            return n;
          if (Y.call(n, "__wrapped__"))
            return mo(n);
        }
        return new ne(n);
      }
      var Ir = /* @__PURE__ */ function() {
        function n() {
        }
        return function(e) {
          if (!V(e))
            return {};
          if (df)
            return df(e);
          n.prototype = e;
          var r = new n();
          return n.prototype = o, r;
        };
      }();
      function $t() {
      }
      function ne(n, e) {
        this.__wrapped__ = n, this.__actions__ = [], this.__chain__ = !!e, this.__index__ = 0, this.__values__ = o;
      }
      u.templateSettings = {
        /**
         * Used to detect `data` property values to be HTML-escaped.
         *
         * @memberOf _.templateSettings
         * @type {RegExp}
         */
        escape: _i,
        /**
         * Used to detect code to be evaluated.
         *
         * @memberOf _.templateSettings
         * @type {RegExp}
         */
        evaluate: dt,
        /**
         * Used to detect `data` property values to inject.
         *
         * @memberOf _.templateSettings
         * @type {RegExp}
         */
        interpolate: wt,
        /**
         * Used to reference the data object in the template text.
         *
         * @memberOf _.templateSettings
         * @type {string}
         */
        variable: "",
        /**
         * Used to import variables into the compiled template.
         *
         * @memberOf _.templateSettings
         * @type {Object}
         */
        imports: {
          /**
           * A reference to the `lodash` function.
           *
           * @memberOf _.templateSettings.imports
           * @type {Function}
           */
          _: u
        }
      }, u.prototype = $t.prototype, u.prototype.constructor = u, ne.prototype = Ir($t.prototype), ne.prototype.constructor = ne;
      function M(n) {
        this.__wrapped__ = n, this.__actions__ = [], this.__dir__ = 1, this.__filtered__ = !1, this.__iteratees__ = [], this.__takeCount__ = Z, this.__views__ = [];
      }
      function Ss() {
        var n = new M(this.__wrapped__);
        return n.__actions__ = Ln(this.__actions__), n.__dir__ = this.__dir__, n.__filtered__ = this.__filtered__, n.__iteratees__ = Ln(this.__iteratees__), n.__takeCount__ = this.__takeCount__, n.__views__ = Ln(this.__views__), n;
      }
      function bs() {
        if (this.__filtered__) {
          var n = new M(this);
          n.__dir__ = -1, n.__filtered__ = !0;
        } else
          n = this.clone(), n.__dir__ *= -1;
        return n;
      }
      function Os() {
        var n = this.__wrapped__.value(), e = this.__dir__, r = O(n), t = e < 0, i = r ? n.length : 0, f = Gl(0, i, this.__views__), a = f.start, l = f.end, h = l - a, _ = t ? l : a - 1, v = this.__iteratees__, d = v.length, x = 0, E = xn(h, this.__takeCount__);
        if (!r || !t && i == h && E == h)
          return zf(n, this.__actions__);
        var m = [];
        n:
          for (; h-- && x < E; ) {
            _ += e;
            for (var P = -1, T = n[_]; ++P < d; ) {
              var W = v[P], B = W.iteratee, Yn = W.type, Cn = B(T);
              if (Yn == Ee)
                T = Cn;
              else if (!Cn) {
                if (Yn == Un)
                  continue n;
                break n;
              }
            }
            m[x++] = T;
          }
        return m;
      }
      M.prototype = Ir($t.prototype), M.prototype.constructor = M;
      function fr(n) {
        var e = -1, r = n == null ? 0 : n.length;
        for (this.clear(); ++e < r; ) {
          var t = n[e];
          this.set(t[0], t[1]);
        }
      }
      function Ls() {
        this.__data__ = jr ? jr(null) : {}, this.size = 0;
      }
      function Is(n) {
        var e = this.has(n) && delete this.__data__[n];
        return this.size -= e ? 1 : 0, e;
      }
      function Ps(n) {
        var e = this.__data__;
        if (jr) {
          var r = e[n];
          return r === en ? o : r;
        }
        return Y.call(e, n) ? e[n] : o;
      }
      function Ds(n) {
        var e = this.__data__;
        return jr ? e[n] !== o : Y.call(e, n);
      }
      function Fs(n, e) {
        var r = this.__data__;
        return this.size += this.has(n) ? 0 : 1, r[n] = jr && e === o ? en : e, this;
      }
      fr.prototype.clear = Ls, fr.prototype.delete = Is, fr.prototype.get = Ps, fr.prototype.has = Ds, fr.prototype.set = Fs;
      function Ce(n) {
        var e = -1, r = n == null ? 0 : n.length;
        for (this.clear(); ++e < r; ) {
          var t = n[e];
          this.set(t[0], t[1]);
        }
      }
      function Ws() {
        this.__data__ = [], this.size = 0;
      }
      function Us(n) {
        var e = this.__data__, r = Gt(e, n);
        if (r < 0)
          return !1;
        var t = e.length - 1;
        return r == t ? e.pop() : Ft.call(e, r, 1), --this.size, !0;
      }
      function Ms(n) {
        var e = this.__data__, r = Gt(e, n);
        return r < 0 ? o : e[r][1];
      }
      function Bs(n) {
        return Gt(this.__data__, n) > -1;
      }
      function Ns(n, e) {
        var r = this.__data__, t = Gt(r, n);
        return t < 0 ? (++this.size, r.push([n, e])) : r[t][1] = e, this;
      }
      Ce.prototype.clear = Ws, Ce.prototype.delete = Us, Ce.prototype.get = Ms, Ce.prototype.has = Bs, Ce.prototype.set = Ns;
      function Se(n) {
        var e = -1, r = n == null ? 0 : n.length;
        for (this.clear(); ++e < r; ) {
          var t = n[e];
          this.set(t[0], t[1]);
        }
      }
      function $s() {
        this.size = 0, this.__data__ = {
          hash: new fr(),
          map: new (Qr || Ce)(),
          string: new fr()
        };
      }
      function Gs(n) {
        var e = jt(this, n).delete(n);
        return this.size -= e ? 1 : 0, e;
      }
      function Hs(n) {
        return jt(this, n).get(n);
      }
      function Ys(n) {
        return jt(this, n).has(n);
      }
      function qs(n, e) {
        var r = jt(this, n), t = r.size;
        return r.set(n, e), this.size += r.size == t ? 0 : 1, this;
      }
      Se.prototype.clear = $s, Se.prototype.delete = Gs, Se.prototype.get = Hs, Se.prototype.has = Ys, Se.prototype.set = qs;
      function or(n) {
        var e = -1, r = n == null ? 0 : n.length;
        for (this.__data__ = new Se(); ++e < r; )
          this.add(n[e]);
      }
      function Ks(n) {
        return this.__data__.set(n, en), this;
      }
      function zs(n) {
        return this.__data__.has(n);
      }
      or.prototype.add = or.prototype.push = Ks, or.prototype.has = zs;
      function ae(n) {
        var e = this.__data__ = new Ce(n);
        this.size = e.size;
      }
      function Js() {
        this.__data__ = new Ce(), this.size = 0;
      }
      function Zs(n) {
        var e = this.__data__, r = e.delete(n);
        return this.size = e.size, r;
      }
      function Xs(n) {
        return this.__data__.get(n);
      }
      function Vs(n) {
        return this.__data__.has(n);
      }
      function Qs(n, e) {
        var r = this.__data__;
        if (r instanceof Ce) {
          var t = r.__data__;
          if (!Qr || t.length < ge - 1)
            return t.push([n, e]), this.size = ++r.size, this;
          r = this.__data__ = new Se(t);
        }
        return r.set(n, e), this.size = r.size, this;
      }
      ae.prototype.clear = Js, ae.prototype.delete = Zs, ae.prototype.get = Xs, ae.prototype.has = Vs, ae.prototype.set = Qs;
      function Af(n, e) {
        var r = O(n), t = !r && hr(n), i = !r && !t && Ve(n), f = !r && !t && !i && Wr(n), a = r || t || i || f, l = a ? Wi(n.length, os) : [], h = l.length;
        for (var _ in n)
          (e || Y.call(n, _)) && !(a && // Safari 9 has enumerable `arguments.length` in strict mode.
          (_ == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
          i && (_ == "offset" || _ == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
          f && (_ == "buffer" || _ == "byteLength" || _ == "byteOffset") || // Skip index properties.
          Ie(_, h))) && l.push(_);
        return l;
      }
      function mf(n) {
        var e = n.length;
        return e ? n[ki(0, e - 1)] : o;
      }
      function ks(n, e) {
        return ni(Ln(n), ar(e, 0, n.length));
      }
      function js(n) {
        return ni(Ln(n));
      }
      function Hi(n, e, r) {
        (r !== o && !se(n[e], r) || r === o && !(e in n)) && be(n, e, r);
      }
      function et(n, e, r) {
        var t = n[e];
        (!(Y.call(n, e) && se(t, r)) || r === o && !(e in n)) && be(n, e, r);
      }
      function Gt(n, e) {
        for (var r = n.length; r--; )
          if (se(n[r][0], e))
            return r;
        return -1;
      }
      function nl(n, e, r, t) {
        return ze(n, function(i, f, a) {
          e(t, i, r(i), a);
        }), t;
      }
      function Tf(n, e) {
        return n && xe(e, hn(e), n);
      }
      function el(n, e) {
        return n && xe(e, Pn(e), n);
      }
      function be(n, e, r) {
        e == "__proto__" && Wt ? Wt(n, e, {
          configurable: !0,
          enumerable: !0,
          value: r,
          writable: !0
        }) : n[e] = r;
      }
      function Yi(n, e) {
        for (var r = -1, t = e.length, i = g(t), f = n == null; ++r < t; )
          i[r] = f ? o : Au(n, e[r]);
        return i;
      }
      function ar(n, e, r) {
        return n === n && (r !== o && (n = n <= r ? n : r), e !== o && (n = n >= e ? n : e)), n;
      }
      function ee(n, e, r, t, i, f) {
        var a, l = e & on, h = e & zn, _ = e & Fn;
        if (r && (a = i ? r(n, t, i, f) : r(n)), a !== o)
          return a;
        if (!V(n))
          return n;
        var v = O(n);
        if (v) {
          if (a = Yl(n), !l)
            return Ln(n, a);
        } else {
          var d = Rn(n), x = d == nr || d == gt;
          if (Ve(n))
            return Xf(n, l);
          if (d == Xn || d == Me || x && !i) {
            if (a = h || x ? {} : po(n), !l)
              return h ? Pl(n, el(a, n)) : Il(n, Tf(a, n));
          } else {
            if (!K[d])
              return i ? n : {};
            a = ql(n, d, l);
          }
        }
        f || (f = new ae());
        var E = f.get(n);
        if (E)
          return E;
        f.set(n, a), qo(n) ? n.forEach(function(T) {
          a.add(ee(T, e, r, T, n, f));
        }) : Ho(n) && n.forEach(function(T, W) {
          a.set(W, ee(T, e, r, W, n, f));
        });
        var m = _ ? h ? su : au : h ? Pn : hn, P = v ? o : m(n);
        return kn(P || n, function(T, W) {
          P && (W = T, T = n[W]), et(a, W, ee(T, e, r, W, n, f));
        }), a;
      }
      function rl(n) {
        var e = hn(n);
        return function(r) {
          return Cf(r, n, e);
        };
      }
      function Cf(n, e, r) {
        var t = r.length;
        if (n == null)
          return !t;
        for (n = q(n); t--; ) {
          var i = r[t], f = e[i], a = n[i];
          if (a === o && !(i in n) || !f(a))
            return !1;
        }
        return !0;
      }
      function Sf(n, e, r) {
        if (typeof n != "function")
          throw new jn(nn);
        return at(function() {
          n.apply(o, r);
        }, e);
      }
      function rt(n, e, r, t) {
        var i = -1, f = mt, a = !0, l = n.length, h = [], _ = e.length;
        if (!l)
          return h;
        r && (e = J(e, $n(r))), t ? (f = Oi, a = !1) : e.length >= ge && (f = Xr, a = !1, e = new or(e));
        n:
          for (; ++i < l; ) {
            var v = n[i], d = r == null ? v : r(v);
            if (v = t || v !== 0 ? v : 0, a && d === d) {
              for (var x = _; x--; )
                if (e[x] === d)
                  continue n;
              h.push(v);
            } else f(e, d, t) || h.push(v);
          }
        return h;
      }
      var ze = no(we), bf = no(Ki, !0);
      function tl(n, e) {
        var r = !0;
        return ze(n, function(t, i, f) {
          return r = !!e(t, i, f), r;
        }), r;
      }
      function Ht(n, e, r) {
        for (var t = -1, i = n.length; ++t < i; ) {
          var f = n[t], a = e(f);
          if (a != null && (l === o ? a === a && !Hn(a) : r(a, l)))
            var l = a, h = f;
        }
        return h;
      }
      function il(n, e, r, t) {
        var i = n.length;
        for (r = I(r), r < 0 && (r = -r > i ? 0 : i + r), t = t === o || t > i ? i : I(t), t < 0 && (t += i), t = r > t ? 0 : zo(t); r < t; )
          n[r++] = e;
        return n;
      }
      function Of(n, e) {
        var r = [];
        return ze(n, function(t, i, f) {
          e(t, i, f) && r.push(t);
        }), r;
      }
      function vn(n, e, r, t, i) {
        var f = -1, a = n.length;
        for (r || (r = zl), i || (i = []); ++f < a; ) {
          var l = n[f];
          e > 0 && r(l) ? e > 1 ? vn(l, e - 1, r, t, i) : Ye(i, l) : t || (i[i.length] = l);
        }
        return i;
      }
      var qi = eo(), Lf = eo(!0);
      function we(n, e) {
        return n && qi(n, e, hn);
      }
      function Ki(n, e) {
        return n && Lf(n, e, hn);
      }
      function Yt(n, e) {
        return He(e, function(r) {
          return Pe(n[r]);
        });
      }
      function sr(n, e) {
        e = Ze(e, n);
        for (var r = 0, t = e.length; n != null && r < t; )
          n = n[Re(e[r++])];
        return r && r == t ? n : o;
      }
      function If(n, e, r) {
        var t = e(n);
        return O(n) ? t : Ye(t, r(n));
      }
      function mn(n) {
        return n == null ? n === o ? gi : er : ur && ur in q(n) ? $l(n) : jl(n);
      }
      function zi(n, e) {
        return n > e;
      }
      function ul(n, e) {
        return n != null && Y.call(n, e);
      }
      function fl(n, e) {
        return n != null && e in q(n);
      }
      function ol(n, e, r) {
        return n >= xn(e, r) && n < ln(e, r);
      }
      function Ji(n, e, r) {
        for (var t = r ? Oi : mt, i = n[0].length, f = n.length, a = f, l = g(f), h = 1 / 0, _ = []; a--; ) {
          var v = n[a];
          a && e && (v = J(v, $n(e))), h = xn(v.length, h), l[a] = !r && (e || i >= 120 && v.length >= 120) ? new or(a && v) : o;
        }
        v = n[0];
        var d = -1, x = l[0];
        n:
          for (; ++d < i && _.length < h; ) {
            var E = v[d], m = e ? e(E) : E;
            if (E = r || E !== 0 ? E : 0, !(x ? Xr(x, m) : t(_, m, r))) {
              for (a = f; --a; ) {
                var P = l[a];
                if (!(P ? Xr(P, m) : t(n[a], m, r)))
                  continue n;
              }
              x && x.push(m), _.push(E);
            }
          }
        return _;
      }
      function al(n, e, r, t) {
        return we(n, function(i, f, a) {
          e(t, r(i), f, a);
        }), t;
      }
      function tt(n, e, r) {
        e = Ze(e, n), n = xo(n, e);
        var t = n == null ? n : n[Re(te(e))];
        return t == null ? o : Nn(t, n, r);
      }
      function Pf(n) {
        return j(n) && mn(n) == Me;
      }
      function sl(n) {
        return j(n) && mn(n) == fe;
      }
      function ll(n) {
        return j(n) && mn(n) == Ae;
      }
      function it(n, e, r, t, i) {
        return n === e ? !0 : n == null || e == null || !j(n) && !j(e) ? n !== n && e !== e : cl(n, e, r, t, it, i);
      }
      function cl(n, e, r, t, i, f) {
        var a = O(n), l = O(e), h = a ? Be : Rn(n), _ = l ? Be : Rn(e);
        h = h == Me ? Xn : h, _ = _ == Me ? Xn : _;
        var v = h == Xn, d = _ == Xn, x = h == _;
        if (x && Ve(n)) {
          if (!Ve(e))
            return !1;
          a = !0, v = !1;
        }
        if (x && !v)
          return f || (f = new ae()), a || Wr(n) ? co(n, e, r, t, i, f) : Bl(n, e, h, r, t, i, f);
        if (!(r & Wn)) {
          var E = v && Y.call(n, "__wrapped__"), m = d && Y.call(e, "__wrapped__");
          if (E || m) {
            var P = E ? n.value() : n, T = m ? e.value() : e;
            return f || (f = new ae()), i(P, T, r, t, f);
          }
        }
        return x ? (f || (f = new ae()), Nl(n, e, r, t, i, f)) : !1;
      }
      function hl(n) {
        return j(n) && Rn(n) == Mn;
      }
      function Zi(n, e, r, t) {
        var i = r.length, f = i, a = !t;
        if (n == null)
          return !f;
        for (n = q(n); i--; ) {
          var l = r[i];
          if (a && l[2] ? l[1] !== n[l[0]] : !(l[0] in n))
            return !1;
        }
        for (; ++i < f; ) {
          l = r[i];
          var h = l[0], _ = n[h], v = l[1];
          if (a && l[2]) {
            if (_ === o && !(h in n))
              return !1;
          } else {
            var d = new ae();
            if (t)
              var x = t(_, v, h, n, e, d);
            if (!(x === o ? it(v, _, Wn | Ue, t, d) : x))
              return !1;
          }
        }
        return !0;
      }
      function Df(n) {
        if (!V(n) || Zl(n))
          return !1;
        var e = Pe(n) ? hs : F;
        return e.test(cr(n));
      }
      function gl(n) {
        return j(n) && mn(n) == rr;
      }
      function pl(n) {
        return j(n) && Rn(n) == Bn;
      }
      function _l(n) {
        return j(n) && fi(n.length) && !!z[mn(n)];
      }
      function Ff(n) {
        return typeof n == "function" ? n : n == null ? Dn : typeof n == "object" ? O(n) ? Mf(n[0], n[1]) : Uf(n) : ta(n);
      }
      function Xi(n) {
        if (!ot(n))
          return ws(n);
        var e = [];
        for (var r in q(n))
          Y.call(n, r) && r != "constructor" && e.push(r);
        return e;
      }
      function vl(n) {
        if (!V(n))
          return kl(n);
        var e = ot(n), r = [];
        for (var t in n)
          t == "constructor" && (e || !Y.call(n, t)) || r.push(t);
        return r;
      }
      function Vi(n, e) {
        return n < e;
      }
      function Wf(n, e) {
        var r = -1, t = In(n) ? g(n.length) : [];
        return ze(n, function(i, f, a) {
          t[++r] = e(i, f, a);
        }), t;
      }
      function Uf(n) {
        var e = cu(n);
        return e.length == 1 && e[0][2] ? vo(e[0][0], e[0][1]) : function(r) {
          return r === n || Zi(r, n, e);
        };
      }
      function Mf(n, e) {
        return gu(n) && _o(e) ? vo(Re(n), e) : function(r) {
          var t = Au(r, n);
          return t === o && t === e ? mu(r, n) : it(e, t, Wn | Ue);
        };
      }
      function qt(n, e, r, t, i) {
        n !== e && qi(e, function(f, a) {
          if (i || (i = new ae()), V(f))
            dl(n, e, a, r, qt, t, i);
          else {
            var l = t ? t(_u(n, a), f, a + "", n, e, i) : o;
            l === o && (l = f), Hi(n, a, l);
          }
        }, Pn);
      }
      function dl(n, e, r, t, i, f, a) {
        var l = _u(n, r), h = _u(e, r), _ = a.get(h);
        if (_) {
          Hi(n, r, _);
          return;
        }
        var v = f ? f(l, h, r + "", n, e, a) : o, d = v === o;
        if (d) {
          var x = O(h), E = !x && Ve(h), m = !x && !E && Wr(h);
          v = h, x || E || m ? O(l) ? v = l : tn(l) ? v = Ln(l) : E ? (d = !1, v = Xf(h, !0)) : m ? (d = !1, v = Vf(h, !0)) : v = [] : st(h) || hr(h) ? (v = l, hr(l) ? v = Jo(l) : (!V(l) || Pe(l)) && (v = po(h))) : d = !1;
        }
        d && (a.set(h, v), i(v, h, t, f, a), a.delete(h)), Hi(n, r, v);
      }
      function Bf(n, e) {
        var r = n.length;
        if (r)
          return e += e < 0 ? r : 0, Ie(e, r) ? n[e] : o;
      }
      function Nf(n, e, r) {
        e.length ? e = J(e, function(f) {
          return O(f) ? function(a) {
            return sr(a, f.length === 1 ? f[0] : f);
          } : f;
        }) : e = [Dn];
        var t = -1;
        e = J(e, $n(A()));
        var i = Wf(n, function(f, a, l) {
          var h = J(e, function(_) {
            return _(f);
          });
          return { criteria: h, index: ++t, value: f };
        });
        return qa(i, function(f, a) {
          return Ll(f, a, r);
        });
      }
      function wl(n, e) {
        return $f(n, e, function(r, t) {
          return mu(n, t);
        });
      }
      function $f(n, e, r) {
        for (var t = -1, i = e.length, f = {}; ++t < i; ) {
          var a = e[t], l = sr(n, a);
          r(l, a) && ut(f, Ze(a, n), l);
        }
        return f;
      }
      function xl(n) {
        return function(e) {
          return sr(e, n);
        };
      }
      function Qi(n, e, r, t) {
        var i = t ? Ya : mr, f = -1, a = e.length, l = n;
        for (n === e && (e = Ln(e)), r && (l = J(n, $n(r))); ++f < a; )
          for (var h = 0, _ = e[f], v = r ? r(_) : _; (h = i(l, v, h, t)) > -1; )
            l !== n && Ft.call(l, h, 1), Ft.call(n, h, 1);
        return n;
      }
      function Gf(n, e) {
        for (var r = n ? e.length : 0, t = r - 1; r--; ) {
          var i = e[r];
          if (r == t || i !== f) {
            var f = i;
            Ie(i) ? Ft.call(n, i, 1) : eu(n, i);
          }
        }
        return n;
      }
      function ki(n, e) {
        return n + Mt(yf() * (e - n + 1));
      }
      function Rl(n, e, r, t) {
        for (var i = -1, f = ln(Ut((e - n) / (r || 1)), 0), a = g(f); f--; )
          a[t ? f : ++i] = n, n += r;
        return a;
      }
      function ji(n, e) {
        var r = "";
        if (!n || e < 1 || e > L)
          return r;
        do
          e % 2 && (r += n), e = Mt(e / 2), e && (n += n);
        while (e);
        return r;
      }
      function D(n, e) {
        return vu(wo(n, e, Dn), n + "");
      }
      function yl(n) {
        return mf(Ur(n));
      }
      function El(n, e) {
        var r = Ur(n);
        return ni(r, ar(e, 0, r.length));
      }
      function ut(n, e, r, t) {
        if (!V(n))
          return n;
        e = Ze(e, n);
        for (var i = -1, f = e.length, a = f - 1, l = n; l != null && ++i < f; ) {
          var h = Re(e[i]), _ = r;
          if (h === "__proto__" || h === "constructor" || h === "prototype")
            return n;
          if (i != a) {
            var v = l[h];
            _ = t ? t(v, h, l) : o, _ === o && (_ = V(v) ? v : Ie(e[i + 1]) ? [] : {});
          }
          et(l, h, _), l = l[h];
        }
        return n;
      }
      var Hf = Bt ? function(n, e) {
        return Bt.set(n, e), n;
      } : Dn, Al = Wt ? function(n, e) {
        return Wt(n, "toString", {
          configurable: !0,
          enumerable: !1,
          value: Cu(e),
          writable: !0
        });
      } : Dn;
      function ml(n) {
        return ni(Ur(n));
      }
      function re(n, e, r) {
        var t = -1, i = n.length;
        e < 0 && (e = -e > i ? 0 : i + e), r = r > i ? i : r, r < 0 && (r += i), i = e > r ? 0 : r - e >>> 0, e >>>= 0;
        for (var f = g(i); ++t < i; )
          f[t] = n[t + e];
        return f;
      }
      function Tl(n, e) {
        var r;
        return ze(n, function(t, i, f) {
          return r = e(t, i, f), !r;
        }), !!r;
      }
      function Kt(n, e, r) {
        var t = 0, i = n == null ? t : n.length;
        if (typeof e == "number" && e === e && i <= je) {
          for (; t < i; ) {
            var f = t + i >>> 1, a = n[f];
            a !== null && !Hn(a) && (r ? a <= e : a < e) ? t = f + 1 : i = f;
          }
          return i;
        }
        return nu(n, e, Dn, r);
      }
      function nu(n, e, r, t) {
        var i = 0, f = n == null ? 0 : n.length;
        if (f === 0)
          return 0;
        e = r(e);
        for (var a = e !== e, l = e === null, h = Hn(e), _ = e === o; i < f; ) {
          var v = Mt((i + f) / 2), d = r(n[v]), x = d !== o, E = d === null, m = d === d, P = Hn(d);
          if (a)
            var T = t || m;
          else _ ? T = m && (t || x) : l ? T = m && x && (t || !E) : h ? T = m && x && !E && (t || !P) : E || P ? T = !1 : T = t ? d <= e : d < e;
          T ? i = v + 1 : f = v;
        }
        return xn(f, ue);
      }
      function Yf(n, e) {
        for (var r = -1, t = n.length, i = 0, f = []; ++r < t; ) {
          var a = n[r], l = e ? e(a) : a;
          if (!r || !se(l, h)) {
            var h = l;
            f[i++] = a === 0 ? 0 : a;
          }
        }
        return f;
      }
      function qf(n) {
        return typeof n == "number" ? n : Hn(n) ? cn : +n;
      }
      function Gn(n) {
        if (typeof n == "string")
          return n;
        if (O(n))
          return J(n, Gn) + "";
        if (Hn(n))
          return Ef ? Ef.call(n) : "";
        var e = n + "";
        return e == "0" && 1 / n == -Zn ? "-0" : e;
      }
      function Je(n, e, r) {
        var t = -1, i = mt, f = n.length, a = !0, l = [], h = l;
        if (r)
          a = !1, i = Oi;
        else if (f >= ge) {
          var _ = e ? null : Ul(n);
          if (_)
            return Ct(_);
          a = !1, i = Xr, h = new or();
        } else
          h = e ? [] : l;
        n:
          for (; ++t < f; ) {
            var v = n[t], d = e ? e(v) : v;
            if (v = r || v !== 0 ? v : 0, a && d === d) {
              for (var x = h.length; x--; )
                if (h[x] === d)
                  continue n;
              e && h.push(d), l.push(v);
            } else i(h, d, r) || (h !== l && h.push(d), l.push(v));
          }
        return l;
      }
      function eu(n, e) {
        return e = Ze(e, n), n = xo(n, e), n == null || delete n[Re(te(e))];
      }
      function Kf(n, e, r, t) {
        return ut(n, e, r(sr(n, e)), t);
      }
      function zt(n, e, r, t) {
        for (var i = n.length, f = t ? i : -1; (t ? f-- : ++f < i) && e(n[f], f, n); )
          ;
        return r ? re(n, t ? 0 : f, t ? f + 1 : i) : re(n, t ? f + 1 : 0, t ? i : f);
      }
      function zf(n, e) {
        var r = n;
        return r instanceof M && (r = r.value()), Li(e, function(t, i) {
          return i.func.apply(i.thisArg, Ye([t], i.args));
        }, r);
      }
      function ru(n, e, r) {
        var t = n.length;
        if (t < 2)
          return t ? Je(n[0]) : [];
        for (var i = -1, f = g(t); ++i < t; )
          for (var a = n[i], l = -1; ++l < t; )
            l != i && (f[i] = rt(f[i] || a, n[l], e, r));
        return Je(vn(f, 1), e, r);
      }
      function Jf(n, e, r) {
        for (var t = -1, i = n.length, f = e.length, a = {}; ++t < i; ) {
          var l = t < f ? e[t] : o;
          r(a, n[t], l);
        }
        return a;
      }
      function tu(n) {
        return tn(n) ? n : [];
      }
      function iu(n) {
        return typeof n == "function" ? n : Dn;
      }
      function Ze(n, e) {
        return O(n) ? n : gu(n, e) ? [n] : Ao(G(n));
      }
      var Cl = D;
      function Xe(n, e, r) {
        var t = n.length;
        return r = r === o ? t : r, !e && r >= t ? n : re(n, e, r);
      }
      var Zf = gs || function(n) {
        return _n.clearTimeout(n);
      };
      function Xf(n, e) {
        if (e)
          return n.slice();
        var r = n.length, t = vf ? vf(r) : new n.constructor(r);
        return n.copy(t), t;
      }
      function uu(n) {
        var e = new n.constructor(n.byteLength);
        return new Pt(e).set(new Pt(n)), e;
      }
      function Sl(n, e) {
        var r = e ? uu(n.buffer) : n.buffer;
        return new n.constructor(r, n.byteOffset, n.byteLength);
      }
      function bl(n) {
        var e = new n.constructor(n.source, N.exec(n));
        return e.lastIndex = n.lastIndex, e;
      }
      function Ol(n) {
        return nt ? q(nt.call(n)) : {};
      }
      function Vf(n, e) {
        var r = e ? uu(n.buffer) : n.buffer;
        return new n.constructor(r, n.byteOffset, n.length);
      }
      function Qf(n, e) {
        if (n !== e) {
          var r = n !== o, t = n === null, i = n === n, f = Hn(n), a = e !== o, l = e === null, h = e === e, _ = Hn(e);
          if (!l && !_ && !f && n > e || f && a && h && !l && !_ || t && a && h || !r && h || !i)
            return 1;
          if (!t && !f && !_ && n < e || _ && r && i && !t && !f || l && r && i || !a && i || !h)
            return -1;
        }
        return 0;
      }
      function Ll(n, e, r) {
        for (var t = -1, i = n.criteria, f = e.criteria, a = i.length, l = r.length; ++t < a; ) {
          var h = Qf(i[t], f[t]);
          if (h) {
            if (t >= l)
              return h;
            var _ = r[t];
            return h * (_ == "desc" ? -1 : 1);
          }
        }
        return n.index - e.index;
      }
      function kf(n, e, r, t) {
        for (var i = -1, f = n.length, a = r.length, l = -1, h = e.length, _ = ln(f - a, 0), v = g(h + _), d = !t; ++l < h; )
          v[l] = e[l];
        for (; ++i < a; )
          (d || i < f) && (v[r[i]] = n[i]);
        for (; _--; )
          v[l++] = n[i++];
        return v;
      }
      function jf(n, e, r, t) {
        for (var i = -1, f = n.length, a = -1, l = r.length, h = -1, _ = e.length, v = ln(f - l, 0), d = g(v + _), x = !t; ++i < v; )
          d[i] = n[i];
        for (var E = i; ++h < _; )
          d[E + h] = e[h];
        for (; ++a < l; )
          (x || i < f) && (d[E + r[a]] = n[i++]);
        return d;
      }
      function Ln(n, e) {
        var r = -1, t = n.length;
        for (e || (e = g(t)); ++r < t; )
          e[r] = n[r];
        return e;
      }
      function xe(n, e, r, t) {
        var i = !r;
        r || (r = {});
        for (var f = -1, a = e.length; ++f < a; ) {
          var l = e[f], h = t ? t(r[l], n[l], l, r, n) : o;
          h === o && (h = n[l]), i ? be(r, l, h) : et(r, l, h);
        }
        return r;
      }
      function Il(n, e) {
        return xe(n, hu(n), e);
      }
      function Pl(n, e) {
        return xe(n, ho(n), e);
      }
      function Jt(n, e) {
        return function(r, t) {
          var i = O(r) ? Ma : nl, f = e ? e() : {};
          return i(r, n, A(t, 2), f);
        };
      }
      function Pr(n) {
        return D(function(e, r) {
          var t = -1, i = r.length, f = i > 1 ? r[i - 1] : o, a = i > 2 ? r[2] : o;
          for (f = n.length > 3 && typeof f == "function" ? (i--, f) : o, a && Tn(r[0], r[1], a) && (f = i < 3 ? o : f, i = 1), e = q(e); ++t < i; ) {
            var l = r[t];
            l && n(e, l, t, f);
          }
          return e;
        });
      }
      function no(n, e) {
        return function(r, t) {
          if (r == null)
            return r;
          if (!In(r))
            return n(r, t);
          for (var i = r.length, f = e ? i : -1, a = q(r); (e ? f-- : ++f < i) && t(a[f], f, a) !== !1; )
            ;
          return r;
        };
      }
      function eo(n) {
        return function(e, r, t) {
          for (var i = -1, f = q(e), a = t(e), l = a.length; l--; ) {
            var h = a[n ? l : ++i];
            if (r(f[h], h, f) === !1)
              break;
          }
          return e;
        };
      }
      function Dl(n, e, r) {
        var t = e & rn, i = ft(n);
        function f() {
          var a = this && this !== _n && this instanceof f ? i : n;
          return a.apply(t ? r : this, arguments);
        }
        return f;
      }
      function ro(n) {
        return function(e) {
          e = G(e);
          var r = Tr(e) ? oe(e) : o, t = r ? r[0] : e.charAt(0), i = r ? Xe(r, 1).join("") : e.slice(1);
          return t[n]() + i;
        };
      }
      function Dr(n) {
        return function(e) {
          return Li(ea(na(e).replace(Aa, "")), n, "");
        };
      }
      function ft(n) {
        return function() {
          var e = arguments;
          switch (e.length) {
            case 0:
              return new n();
            case 1:
              return new n(e[0]);
            case 2:
              return new n(e[0], e[1]);
            case 3:
              return new n(e[0], e[1], e[2]);
            case 4:
              return new n(e[0], e[1], e[2], e[3]);
            case 5:
              return new n(e[0], e[1], e[2], e[3], e[4]);
            case 6:
              return new n(e[0], e[1], e[2], e[3], e[4], e[5]);
            case 7:
              return new n(e[0], e[1], e[2], e[3], e[4], e[5], e[6]);
          }
          var r = Ir(n.prototype), t = n.apply(r, e);
          return V(t) ? t : r;
        };
      }
      function Fl(n, e, r) {
        var t = ft(n);
        function i() {
          for (var f = arguments.length, a = g(f), l = f, h = Fr(i); l--; )
            a[l] = arguments[l];
          var _ = f < 3 && a[0] !== h && a[f - 1] !== h ? [] : qe(a, h);
          if (f -= _.length, f < r)
            return oo(
              n,
              e,
              Zt,
              i.placeholder,
              o,
              a,
              _,
              o,
              o,
              r - f
            );
          var v = this && this !== _n && this instanceof i ? t : n;
          return Nn(v, this, a);
        }
        return i;
      }
      function to(n) {
        return function(e, r, t) {
          var i = q(e);
          if (!In(e)) {
            var f = A(r, 3);
            e = hn(e), r = function(l) {
              return f(i[l], l, i);
            };
          }
          var a = n(e, r, t);
          return a > -1 ? i[f ? e[a] : a] : o;
        };
      }
      function io(n) {
        return Le(function(e) {
          var r = e.length, t = r, i = ne.prototype.thru;
          for (n && e.reverse(); t--; ) {
            var f = e[t];
            if (typeof f != "function")
              throw new jn(nn);
            if (i && !a && kt(f) == "wrapper")
              var a = new ne([], !0);
          }
          for (t = a ? t : r; ++t < r; ) {
            f = e[t];
            var l = kt(f), h = l == "wrapper" ? lu(f) : o;
            h && pu(h[0]) && h[1] == (dn | En | Sn | Jn) && !h[4].length && h[9] == 1 ? a = a[kt(h[0])].apply(a, h[3]) : a = f.length == 1 && pu(f) ? a[l]() : a.thru(f);
          }
          return function() {
            var _ = arguments, v = _[0];
            if (a && _.length == 1 && O(v))
              return a.plant(v).value();
            for (var d = 0, x = r ? e[d].apply(this, _) : v; ++d < r; )
              x = e[d].call(this, x);
            return x;
          };
        });
      }
      function Zt(n, e, r, t, i, f, a, l, h, _) {
        var v = e & dn, d = e & rn, x = e & an, E = e & (En | _e), m = e & ye, P = x ? o : ft(n);
        function T() {
          for (var W = arguments.length, B = g(W), Yn = W; Yn--; )
            B[Yn] = arguments[Yn];
          if (E)
            var Cn = Fr(T), qn = za(B, Cn);
          if (t && (B = kf(B, t, i, E)), f && (B = jf(B, f, a, E)), W -= qn, E && W < _) {
            var un = qe(B, Cn);
            return oo(
              n,
              e,
              Zt,
              T.placeholder,
              r,
              B,
              un,
              l,
              h,
              _ - W
            );
          }
          var le = d ? r : this, Fe = x ? le[n] : n;
          return W = B.length, l ? B = nc(B, l) : m && W > 1 && B.reverse(), v && h < W && (B.length = h), this && this !== _n && this instanceof T && (Fe = P || ft(Fe)), Fe.apply(le, B);
        }
        return T;
      }
      function uo(n, e) {
        return function(r, t) {
          return al(r, n, e(t), {});
        };
      }
      function Xt(n, e) {
        return function(r, t) {
          var i;
          if (r === o && t === o)
            return e;
          if (r !== o && (i = r), t !== o) {
            if (i === o)
              return t;
            typeof r == "string" || typeof t == "string" ? (r = Gn(r), t = Gn(t)) : (r = qf(r), t = qf(t)), i = n(r, t);
          }
          return i;
        };
      }
      function fu(n) {
        return Le(function(e) {
          return e = J(e, $n(A())), D(function(r) {
            var t = this;
            return n(e, function(i) {
              return Nn(i, t, r);
            });
          });
        });
      }
      function Vt(n, e) {
        e = e === o ? " " : Gn(e);
        var r = e.length;
        if (r < 2)
          return r ? ji(e, n) : e;
        var t = ji(e, Ut(n / Cr(e)));
        return Tr(e) ? Xe(oe(t), 0, n).join("") : t.slice(0, n);
      }
      function Wl(n, e, r, t) {
        var i = e & rn, f = ft(n);
        function a() {
          for (var l = -1, h = arguments.length, _ = -1, v = t.length, d = g(v + h), x = this && this !== _n && this instanceof a ? f : n; ++_ < v; )
            d[_] = t[_];
          for (; h--; )
            d[_++] = arguments[++l];
          return Nn(x, i ? r : this, d);
        }
        return a;
      }
      function fo(n) {
        return function(e, r, t) {
          return t && typeof t != "number" && Tn(e, r, t) && (r = t = o), e = De(e), r === o ? (r = e, e = 0) : r = De(r), t = t === o ? e < r ? 1 : -1 : De(t), Rl(e, r, t, n);
        };
      }
      function Qt(n) {
        return function(e, r) {
          return typeof e == "string" && typeof r == "string" || (e = ie(e), r = ie(r)), n(e, r);
        };
      }
      function oo(n, e, r, t, i, f, a, l, h, _) {
        var v = e & En, d = v ? a : o, x = v ? o : a, E = v ? f : o, m = v ? o : f;
        e |= v ? Sn : An, e &= ~(v ? An : Sn), e & b || (e &= ~(rn | an));
        var P = [
          n,
          e,
          i,
          E,
          d,
          m,
          x,
          l,
          h,
          _
        ], T = r.apply(o, P);
        return pu(n) && Ro(T, P), T.placeholder = t, yo(T, n, e);
      }
      function ou(n) {
        var e = sn[n];
        return function(r, t) {
          if (r = ie(r), t = t == null ? 0 : xn(I(t), 292), t && Rf(r)) {
            var i = (G(r) + "e").split("e"), f = e(i[0] + "e" + (+i[1] + t));
            return i = (G(f) + "e").split("e"), +(i[0] + "e" + (+i[1] - t));
          }
          return e(r);
        };
      }
      var Ul = Or && 1 / Ct(new Or([, -0]))[1] == Zn ? function(n) {
        return new Or(n);
      } : Ou;
      function ao(n) {
        return function(e) {
          var r = Rn(e);
          return r == Mn ? Mi(e) : r == Bn ? ja(e) : Ka(e, n(e));
        };
      }
      function Oe(n, e, r, t, i, f, a, l) {
        var h = e & an;
        if (!h && typeof n != "function")
          throw new jn(nn);
        var _ = t ? t.length : 0;
        if (_ || (e &= ~(Sn | An), t = i = o), a = a === o ? a : ln(I(a), 0), l = l === o ? l : I(l), _ -= i ? i.length : 0, e & An) {
          var v = t, d = i;
          t = i = o;
        }
        var x = h ? o : lu(n), E = [
          n,
          e,
          r,
          t,
          i,
          v,
          d,
          f,
          a,
          l
        ];
        if (x && Ql(E, x), n = E[0], e = E[1], r = E[2], t = E[3], i = E[4], l = E[9] = E[9] === o ? h ? 0 : n.length : ln(E[9] - _, 0), !l && e & (En | _e) && (e &= ~(En | _e)), !e || e == rn)
          var m = Dl(n, e, r);
        else e == En || e == _e ? m = Fl(n, e, l) : (e == Sn || e == (rn | Sn)) && !i.length ? m = Wl(n, e, r, t) : m = Zt.apply(o, E);
        var P = x ? Hf : Ro;
        return yo(P(m, E), n, e);
      }
      function so(n, e, r, t) {
        return n === o || se(n, br[r]) && !Y.call(t, r) ? e : n;
      }
      function lo(n, e, r, t, i, f) {
        return V(n) && V(e) && (f.set(e, n), qt(n, e, o, lo, f), f.delete(e)), n;
      }
      function Ml(n) {
        return st(n) ? o : n;
      }
      function co(n, e, r, t, i, f) {
        var a = r & Wn, l = n.length, h = e.length;
        if (l != h && !(a && h > l))
          return !1;
        var _ = f.get(n), v = f.get(e);
        if (_ && v)
          return _ == e && v == n;
        var d = -1, x = !0, E = r & Ue ? new or() : o;
        for (f.set(n, e), f.set(e, n); ++d < l; ) {
          var m = n[d], P = e[d];
          if (t)
            var T = a ? t(P, m, d, e, n, f) : t(m, P, d, n, e, f);
          if (T !== o) {
            if (T)
              continue;
            x = !1;
            break;
          }
          if (E) {
            if (!Ii(e, function(W, B) {
              if (!Xr(E, B) && (m === W || i(m, W, r, t, f)))
                return E.push(B);
            })) {
              x = !1;
              break;
            }
          } else if (!(m === P || i(m, P, r, t, f))) {
            x = !1;
            break;
          }
        }
        return f.delete(n), f.delete(e), x;
      }
      function Bl(n, e, r, t, i, f, a) {
        switch (r) {
          case $e:
            if (n.byteLength != e.byteLength || n.byteOffset != e.byteOffset)
              return !1;
            n = n.buffer, e = e.buffer;
          case fe:
            return !(n.byteLength != e.byteLength || !f(new Pt(n), new Pt(e)));
          case ve:
          case Ae:
          case de:
            return se(+n, +e);
          case vr:
            return n.name == e.name && n.message == e.message;
          case rr:
          case me:
            return n == e + "";
          case Mn:
            var l = Mi;
          case Bn:
            var h = t & Wn;
            if (l || (l = Ct), n.size != e.size && !h)
              return !1;
            var _ = a.get(n);
            if (_)
              return _ == e;
            t |= Ue, a.set(n, e);
            var v = co(l(n), l(e), t, i, f, a);
            return a.delete(n), v;
          case wr:
            if (nt)
              return nt.call(n) == nt.call(e);
        }
        return !1;
      }
      function Nl(n, e, r, t, i, f) {
        var a = r & Wn, l = au(n), h = l.length, _ = au(e), v = _.length;
        if (h != v && !a)
          return !1;
        for (var d = h; d--; ) {
          var x = l[d];
          if (!(a ? x in e : Y.call(e, x)))
            return !1;
        }
        var E = f.get(n), m = f.get(e);
        if (E && m)
          return E == e && m == n;
        var P = !0;
        f.set(n, e), f.set(e, n);
        for (var T = a; ++d < h; ) {
          x = l[d];
          var W = n[x], B = e[x];
          if (t)
            var Yn = a ? t(B, W, x, e, n, f) : t(W, B, x, n, e, f);
          if (!(Yn === o ? W === B || i(W, B, r, t, f) : Yn)) {
            P = !1;
            break;
          }
          T || (T = x == "constructor");
        }
        if (P && !T) {
          var Cn = n.constructor, qn = e.constructor;
          Cn != qn && "constructor" in n && "constructor" in e && !(typeof Cn == "function" && Cn instanceof Cn && typeof qn == "function" && qn instanceof qn) && (P = !1);
        }
        return f.delete(n), f.delete(e), P;
      }
      function Le(n) {
        return vu(wo(n, o, So), n + "");
      }
      function au(n) {
        return If(n, hn, hu);
      }
      function su(n) {
        return If(n, Pn, ho);
      }
      var lu = Bt ? function(n) {
        return Bt.get(n);
      } : Ou;
      function kt(n) {
        for (var e = n.name + "", r = Lr[e], t = Y.call(Lr, e) ? r.length : 0; t--; ) {
          var i = r[t], f = i.func;
          if (f == null || f == n)
            return i.name;
        }
        return e;
      }
      function Fr(n) {
        var e = Y.call(u, "placeholder") ? u : n;
        return e.placeholder;
      }
      function A() {
        var n = u.iteratee || Su;
        return n = n === Su ? Ff : n, arguments.length ? n(arguments[0], arguments[1]) : n;
      }
      function jt(n, e) {
        var r = n.__data__;
        return Jl(e) ? r[typeof e == "string" ? "string" : "hash"] : r.map;
      }
      function cu(n) {
        for (var e = hn(n), r = e.length; r--; ) {
          var t = e[r], i = n[t];
          e[r] = [t, i, _o(i)];
        }
        return e;
      }
      function lr(n, e) {
        var r = Va(n, e);
        return Df(r) ? r : o;
      }
      function $l(n) {
        var e = Y.call(n, ur), r = n[ur];
        try {
          n[ur] = o;
          var t = !0;
        } catch {
        }
        var i = Lt.call(n);
        return t && (e ? n[ur] = r : delete n[ur]), i;
      }
      var hu = Ni ? function(n) {
        return n == null ? [] : (n = q(n), He(Ni(n), function(e) {
          return wf.call(n, e);
        }));
      } : Lu, ho = Ni ? function(n) {
        for (var e = []; n; )
          Ye(e, hu(n)), n = Dt(n);
        return e;
      } : Lu, Rn = mn;
      ($i && Rn(new $i(new ArrayBuffer(1))) != $e || Qr && Rn(new Qr()) != Mn || Gi && Rn(Gi.resolve()) != Br || Or && Rn(new Or()) != Bn || kr && Rn(new kr()) != Ne) && (Rn = function(n) {
        var e = mn(n), r = e == Xn ? n.constructor : o, t = r ? cr(r) : "";
        if (t)
          switch (t) {
            case Es:
              return $e;
            case As:
              return Mn;
            case ms:
              return Br;
            case Ts:
              return Bn;
            case Cs:
              return Ne;
          }
        return e;
      });
      function Gl(n, e, r) {
        for (var t = -1, i = r.length; ++t < i; ) {
          var f = r[t], a = f.size;
          switch (f.type) {
            case "drop":
              n += a;
              break;
            case "dropRight":
              e -= a;
              break;
            case "take":
              e = xn(e, n + a);
              break;
            case "takeRight":
              n = ln(n, e - a);
              break;
          }
        }
        return { start: n, end: e };
      }
      function Hl(n) {
        var e = n.match(xi);
        return e ? e[1].split(Ri) : [];
      }
      function go(n, e, r) {
        e = Ze(e, n);
        for (var t = -1, i = e.length, f = !1; ++t < i; ) {
          var a = Re(e[t]);
          if (!(f = n != null && r(n, a)))
            break;
          n = n[a];
        }
        return f || ++t != i ? f : (i = n == null ? 0 : n.length, !!i && fi(i) && Ie(a, i) && (O(n) || hr(n)));
      }
      function Yl(n) {
        var e = n.length, r = new n.constructor(e);
        return e && typeof n[0] == "string" && Y.call(n, "index") && (r.index = n.index, r.input = n.input), r;
      }
      function po(n) {
        return typeof n.constructor == "function" && !ot(n) ? Ir(Dt(n)) : {};
      }
      function ql(n, e, r) {
        var t = n.constructor;
        switch (e) {
          case fe:
            return uu(n);
          case ve:
          case Ae:
            return new t(+n);
          case $e:
            return Sl(n, r);
          case xr:
          case Rr:
          case tr:
          case Nr:
          case $r:
          case Gr:
          case Hr:
          case Yr:
          case qr:
            return Vf(n, r);
          case Mn:
            return new t();
          case de:
          case me:
            return new t(n);
          case rr:
            return bl(n);
          case Bn:
            return new t();
          case wr:
            return Ol(n);
        }
      }
      function Kl(n, e) {
        var r = e.length;
        if (!r)
          return n;
        var t = r - 1;
        return e[t] = (r > 1 ? "& " : "") + e[t], e = e.join(r > 2 ? ", " : " "), n.replace(wi, `{
/* [wrapped with ` + e + `] */
`);
      }
      function zl(n) {
        return O(n) || hr(n) || !!(xf && n && n[xf]);
      }
      function Ie(n, e) {
        var r = typeof n;
        return e = e ?? L, !!e && (r == "number" || r != "symbol" && X.test(n)) && n > -1 && n % 1 == 0 && n < e;
      }
      function Tn(n, e, r) {
        if (!V(r))
          return !1;
        var t = typeof e;
        return (t == "number" ? In(r) && Ie(e, r.length) : t == "string" && e in r) ? se(r[e], n) : !1;
      }
      function gu(n, e) {
        if (O(n))
          return !1;
        var r = typeof n;
        return r == "number" || r == "symbol" || r == "boolean" || n == null || Hn(n) ? !0 : Rt.test(n) || !xt.test(n) || e != null && n in q(e);
      }
      function Jl(n) {
        var e = typeof n;
        return e == "string" || e == "number" || e == "symbol" || e == "boolean" ? n !== "__proto__" : n === null;
      }
      function pu(n) {
        var e = kt(n), r = u[e];
        if (typeof r != "function" || !(e in M.prototype))
          return !1;
        if (n === r)
          return !0;
        var t = lu(r);
        return !!t && n === t[0];
      }
      function Zl(n) {
        return !!_f && _f in n;
      }
      var Xl = bt ? Pe : Iu;
      function ot(n) {
        var e = n && n.constructor, r = typeof e == "function" && e.prototype || br;
        return n === r;
      }
      function _o(n) {
        return n === n && !V(n);
      }
      function vo(n, e) {
        return function(r) {
          return r == null ? !1 : r[n] === e && (e !== o || n in q(r));
        };
      }
      function Vl(n) {
        var e = ii(n, function(t) {
          return r.size === pe && r.clear(), t;
        }), r = e.cache;
        return e;
      }
      function Ql(n, e) {
        var r = n[1], t = e[1], i = r | t, f = i < (rn | an | dn), a = t == dn && r == En || t == dn && r == Jn && n[7].length <= e[8] || t == (dn | Jn) && e[7].length <= e[8] && r == En;
        if (!(f || a))
          return n;
        t & rn && (n[2] = e[2], i |= r & rn ? 0 : b);
        var l = e[3];
        if (l) {
          var h = n[3];
          n[3] = h ? kf(h, l, e[4]) : l, n[4] = h ? qe(n[3], fn) : e[4];
        }
        return l = e[5], l && (h = n[5], n[5] = h ? jf(h, l, e[6]) : l, n[6] = h ? qe(n[5], fn) : e[6]), l = e[7], l && (n[7] = l), t & dn && (n[8] = n[8] == null ? e[8] : xn(n[8], e[8])), n[9] == null && (n[9] = e[9]), n[0] = e[0], n[1] = i, n;
      }
      function kl(n) {
        var e = [];
        if (n != null)
          for (var r in q(n))
            e.push(r);
        return e;
      }
      function jl(n) {
        return Lt.call(n);
      }
      function wo(n, e, r) {
        return e = ln(e === o ? n.length - 1 : e, 0), function() {
          for (var t = arguments, i = -1, f = ln(t.length - e, 0), a = g(f); ++i < f; )
            a[i] = t[e + i];
          i = -1;
          for (var l = g(e + 1); ++i < e; )
            l[i] = t[i];
          return l[e] = r(a), Nn(n, this, l);
        };
      }
      function xo(n, e) {
        return e.length < 2 ? n : sr(n, re(e, 0, -1));
      }
      function nc(n, e) {
        for (var r = n.length, t = xn(e.length, r), i = Ln(n); t--; ) {
          var f = e[t];
          n[t] = Ie(f, r) ? i[f] : o;
        }
        return n;
      }
      function _u(n, e) {
        if (!(e === "constructor" && typeof n[e] == "function") && e != "__proto__")
          return n[e];
      }
      var Ro = Eo(Hf), at = _s || function(n, e) {
        return _n.setTimeout(n, e);
      }, vu = Eo(Al);
      function yo(n, e, r) {
        var t = e + "";
        return vu(n, Kl(t, ec(Hl(t), r)));
      }
      function Eo(n) {
        var e = 0, r = 0;
        return function() {
          var t = xs(), i = bn - (t - r);
          if (r = t, i > 0) {
            if (++e >= gr)
              return arguments[0];
          } else
            e = 0;
          return n.apply(o, arguments);
        };
      }
      function ni(n, e) {
        var r = -1, t = n.length, i = t - 1;
        for (e = e === o ? t : e; ++r < e; ) {
          var f = ki(r, i), a = n[f];
          n[f] = n[r], n[r] = a;
        }
        return n.length = e, n;
      }
      var Ao = Vl(function(n) {
        var e = [];
        return n.charCodeAt(0) === 46 && e.push(""), n.replace(vi, function(r, t, i, f) {
          e.push(i ? f.replace(R, "$1") : t || r);
        }), e;
      });
      function Re(n) {
        if (typeof n == "string" || Hn(n))
          return n;
        var e = n + "";
        return e == "0" && 1 / n == -Zn ? "-0" : e;
      }
      function cr(n) {
        if (n != null) {
          try {
            return Ot.call(n);
          } catch {
          }
          try {
            return n + "";
          } catch {
          }
        }
        return "";
      }
      function ec(n, e) {
        return kn(hi, function(r) {
          var t = "_." + r[0];
          e & r[1] && !mt(n, t) && n.push(t);
        }), n.sort();
      }
      function mo(n) {
        if (n instanceof M)
          return n.clone();
        var e = new ne(n.__wrapped__, n.__chain__);
        return e.__actions__ = Ln(n.__actions__), e.__index__ = n.__index__, e.__values__ = n.__values__, e;
      }
      function rc(n, e, r) {
        (r ? Tn(n, e, r) : e === o) ? e = 1 : e = ln(I(e), 0);
        var t = n == null ? 0 : n.length;
        if (!t || e < 1)
          return [];
        for (var i = 0, f = 0, a = g(Ut(t / e)); i < t; )
          a[f++] = re(n, i, i += e);
        return a;
      }
      function tc(n) {
        for (var e = -1, r = n == null ? 0 : n.length, t = 0, i = []; ++e < r; ) {
          var f = n[e];
          f && (i[t++] = f);
        }
        return i;
      }
      function ic() {
        var n = arguments.length;
        if (!n)
          return [];
        for (var e = g(n - 1), r = arguments[0], t = n; t--; )
          e[t - 1] = arguments[t];
        return Ye(O(r) ? Ln(r) : [r], vn(e, 1));
      }
      var uc = D(function(n, e) {
        return tn(n) ? rt(n, vn(e, 1, tn, !0)) : [];
      }), fc = D(function(n, e) {
        var r = te(e);
        return tn(r) && (r = o), tn(n) ? rt(n, vn(e, 1, tn, !0), A(r, 2)) : [];
      }), oc = D(function(n, e) {
        var r = te(e);
        return tn(r) && (r = o), tn(n) ? rt(n, vn(e, 1, tn, !0), o, r) : [];
      });
      function ac(n, e, r) {
        var t = n == null ? 0 : n.length;
        return t ? (e = r || e === o ? 1 : I(e), re(n, e < 0 ? 0 : e, t)) : [];
      }
      function sc(n, e, r) {
        var t = n == null ? 0 : n.length;
        return t ? (e = r || e === o ? 1 : I(e), e = t - e, re(n, 0, e < 0 ? 0 : e)) : [];
      }
      function lc(n, e) {
        return n && n.length ? zt(n, A(e, 3), !0, !0) : [];
      }
      function cc(n, e) {
        return n && n.length ? zt(n, A(e, 3), !0) : [];
      }
      function hc(n, e, r, t) {
        var i = n == null ? 0 : n.length;
        return i ? (r && typeof r != "number" && Tn(n, e, r) && (r = 0, t = i), il(n, e, r, t)) : [];
      }
      function To(n, e, r) {
        var t = n == null ? 0 : n.length;
        if (!t)
          return -1;
        var i = r == null ? 0 : I(r);
        return i < 0 && (i = ln(t + i, 0)), Tt(n, A(e, 3), i);
      }
      function Co(n, e, r) {
        var t = n == null ? 0 : n.length;
        if (!t)
          return -1;
        var i = t - 1;
        return r !== o && (i = I(r), i = r < 0 ? ln(t + i, 0) : xn(i, t - 1)), Tt(n, A(e, 3), i, !0);
      }
      function So(n) {
        var e = n == null ? 0 : n.length;
        return e ? vn(n, 1) : [];
      }
      function gc(n) {
        var e = n == null ? 0 : n.length;
        return e ? vn(n, Zn) : [];
      }
      function pc(n, e) {
        var r = n == null ? 0 : n.length;
        return r ? (e = e === o ? 1 : I(e), vn(n, e)) : [];
      }
      function _c(n) {
        for (var e = -1, r = n == null ? 0 : n.length, t = {}; ++e < r; ) {
          var i = n[e];
          t[i[0]] = i[1];
        }
        return t;
      }
      function bo(n) {
        return n && n.length ? n[0] : o;
      }
      function vc(n, e, r) {
        var t = n == null ? 0 : n.length;
        if (!t)
          return -1;
        var i = r == null ? 0 : I(r);
        return i < 0 && (i = ln(t + i, 0)), mr(n, e, i);
      }
      function dc(n) {
        var e = n == null ? 0 : n.length;
        return e ? re(n, 0, -1) : [];
      }
      var wc = D(function(n) {
        var e = J(n, tu);
        return e.length && e[0] === n[0] ? Ji(e) : [];
      }), xc = D(function(n) {
        var e = te(n), r = J(n, tu);
        return e === te(r) ? e = o : r.pop(), r.length && r[0] === n[0] ? Ji(r, A(e, 2)) : [];
      }), Rc = D(function(n) {
        var e = te(n), r = J(n, tu);
        return e = typeof e == "function" ? e : o, e && r.pop(), r.length && r[0] === n[0] ? Ji(r, o, e) : [];
      });
      function yc(n, e) {
        return n == null ? "" : ds.call(n, e);
      }
      function te(n) {
        var e = n == null ? 0 : n.length;
        return e ? n[e - 1] : o;
      }
      function Ec(n, e, r) {
        var t = n == null ? 0 : n.length;
        if (!t)
          return -1;
        var i = t;
        return r !== o && (i = I(r), i = i < 0 ? ln(t + i, 0) : xn(i, t - 1)), e === e ? es(n, e, i) : Tt(n, of, i, !0);
      }
      function Ac(n, e) {
        return n && n.length ? Bf(n, I(e)) : o;
      }
      var mc = D(Oo);
      function Oo(n, e) {
        return n && n.length && e && e.length ? Qi(n, e) : n;
      }
      function Tc(n, e, r) {
        return n && n.length && e && e.length ? Qi(n, e, A(r, 2)) : n;
      }
      function Cc(n, e, r) {
        return n && n.length && e && e.length ? Qi(n, e, o, r) : n;
      }
      var Sc = Le(function(n, e) {
        var r = n == null ? 0 : n.length, t = Yi(n, e);
        return Gf(n, J(e, function(i) {
          return Ie(i, r) ? +i : i;
        }).sort(Qf)), t;
      });
      function bc(n, e) {
        var r = [];
        if (!(n && n.length))
          return r;
        var t = -1, i = [], f = n.length;
        for (e = A(e, 3); ++t < f; ) {
          var a = n[t];
          e(a, t, n) && (r.push(a), i.push(t));
        }
        return Gf(n, i), r;
      }
      function du(n) {
        return n == null ? n : ys.call(n);
      }
      function Oc(n, e, r) {
        var t = n == null ? 0 : n.length;
        return t ? (r && typeof r != "number" && Tn(n, e, r) ? (e = 0, r = t) : (e = e == null ? 0 : I(e), r = r === o ? t : I(r)), re(n, e, r)) : [];
      }
      function Lc(n, e) {
        return Kt(n, e);
      }
      function Ic(n, e, r) {
        return nu(n, e, A(r, 2));
      }
      function Pc(n, e) {
        var r = n == null ? 0 : n.length;
        if (r) {
          var t = Kt(n, e);
          if (t < r && se(n[t], e))
            return t;
        }
        return -1;
      }
      function Dc(n, e) {
        return Kt(n, e, !0);
      }
      function Fc(n, e, r) {
        return nu(n, e, A(r, 2), !0);
      }
      function Wc(n, e) {
        var r = n == null ? 0 : n.length;
        if (r) {
          var t = Kt(n, e, !0) - 1;
          if (se(n[t], e))
            return t;
        }
        return -1;
      }
      function Uc(n) {
        return n && n.length ? Yf(n) : [];
      }
      function Mc(n, e) {
        return n && n.length ? Yf(n, A(e, 2)) : [];
      }
      function Bc(n) {
        var e = n == null ? 0 : n.length;
        return e ? re(n, 1, e) : [];
      }
      function Nc(n, e, r) {
        return n && n.length ? (e = r || e === o ? 1 : I(e), re(n, 0, e < 0 ? 0 : e)) : [];
      }
      function $c(n, e, r) {
        var t = n == null ? 0 : n.length;
        return t ? (e = r || e === o ? 1 : I(e), e = t - e, re(n, e < 0 ? 0 : e, t)) : [];
      }
      function Gc(n, e) {
        return n && n.length ? zt(n, A(e, 3), !1, !0) : [];
      }
      function Hc(n, e) {
        return n && n.length ? zt(n, A(e, 3)) : [];
      }
      var Yc = D(function(n) {
        return Je(vn(n, 1, tn, !0));
      }), qc = D(function(n) {
        var e = te(n);
        return tn(e) && (e = o), Je(vn(n, 1, tn, !0), A(e, 2));
      }), Kc = D(function(n) {
        var e = te(n);
        return e = typeof e == "function" ? e : o, Je(vn(n, 1, tn, !0), o, e);
      });
      function zc(n) {
        return n && n.length ? Je(n) : [];
      }
      function Jc(n, e) {
        return n && n.length ? Je(n, A(e, 2)) : [];
      }
      function Zc(n, e) {
        return e = typeof e == "function" ? e : o, n && n.length ? Je(n, o, e) : [];
      }
      function wu(n) {
        if (!(n && n.length))
          return [];
        var e = 0;
        return n = He(n, function(r) {
          if (tn(r))
            return e = ln(r.length, e), !0;
        }), Wi(e, function(r) {
          return J(n, Pi(r));
        });
      }
      function Lo(n, e) {
        if (!(n && n.length))
          return [];
        var r = wu(n);
        return e == null ? r : J(r, function(t) {
          return Nn(e, o, t);
        });
      }
      var Xc = D(function(n, e) {
        return tn(n) ? rt(n, e) : [];
      }), Vc = D(function(n) {
        return ru(He(n, tn));
      }), Qc = D(function(n) {
        var e = te(n);
        return tn(e) && (e = o), ru(He(n, tn), A(e, 2));
      }), kc = D(function(n) {
        var e = te(n);
        return e = typeof e == "function" ? e : o, ru(He(n, tn), o, e);
      }), jc = D(wu);
      function nh(n, e) {
        return Jf(n || [], e || [], et);
      }
      function eh(n, e) {
        return Jf(n || [], e || [], ut);
      }
      var rh = D(function(n) {
        var e = n.length, r = e > 1 ? n[e - 1] : o;
        return r = typeof r == "function" ? (n.pop(), r) : o, Lo(n, r);
      });
      function Io(n) {
        var e = u(n);
        return e.__chain__ = !0, e;
      }
      function th(n, e) {
        return e(n), n;
      }
      function ei(n, e) {
        return e(n);
      }
      var ih = Le(function(n) {
        var e = n.length, r = e ? n[0] : 0, t = this.__wrapped__, i = function(f) {
          return Yi(f, n);
        };
        return e > 1 || this.__actions__.length || !(t instanceof M) || !Ie(r) ? this.thru(i) : (t = t.slice(r, +r + (e ? 1 : 0)), t.__actions__.push({
          func: ei,
          args: [i],
          thisArg: o
        }), new ne(t, this.__chain__).thru(function(f) {
          return e && !f.length && f.push(o), f;
        }));
      });
      function uh() {
        return Io(this);
      }
      function fh() {
        return new ne(this.value(), this.__chain__);
      }
      function oh() {
        this.__values__ === o && (this.__values__ = Ko(this.value()));
        var n = this.__index__ >= this.__values__.length, e = n ? o : this.__values__[this.__index__++];
        return { done: n, value: e };
      }
      function ah() {
        return this;
      }
      function sh(n) {
        for (var e, r = this; r instanceof $t; ) {
          var t = mo(r);
          t.__index__ = 0, t.__values__ = o, e ? i.__wrapped__ = t : e = t;
          var i = t;
          r = r.__wrapped__;
        }
        return i.__wrapped__ = n, e;
      }
      function lh() {
        var n = this.__wrapped__;
        if (n instanceof M) {
          var e = n;
          return this.__actions__.length && (e = new M(this)), e = e.reverse(), e.__actions__.push({
            func: ei,
            args: [du],
            thisArg: o
          }), new ne(e, this.__chain__);
        }
        return this.thru(du);
      }
      function ch() {
        return zf(this.__wrapped__, this.__actions__);
      }
      var hh = Jt(function(n, e, r) {
        Y.call(n, r) ? ++n[r] : be(n, r, 1);
      });
      function gh(n, e, r) {
        var t = O(n) ? uf : tl;
        return r && Tn(n, e, r) && (e = o), t(n, A(e, 3));
      }
      function ph(n, e) {
        var r = O(n) ? He : Of;
        return r(n, A(e, 3));
      }
      var _h = to(To), vh = to(Co);
      function dh(n, e) {
        return vn(ri(n, e), 1);
      }
      function wh(n, e) {
        return vn(ri(n, e), Zn);
      }
      function xh(n, e, r) {
        return r = r === o ? 1 : I(r), vn(ri(n, e), r);
      }
      function Po(n, e) {
        var r = O(n) ? kn : ze;
        return r(n, A(e, 3));
      }
      function Do(n, e) {
        var r = O(n) ? Ba : bf;
        return r(n, A(e, 3));
      }
      var Rh = Jt(function(n, e, r) {
        Y.call(n, r) ? n[r].push(e) : be(n, r, [e]);
      });
      function yh(n, e, r, t) {
        n = In(n) ? n : Ur(n), r = r && !t ? I(r) : 0;
        var i = n.length;
        return r < 0 && (r = ln(i + r, 0)), oi(n) ? r <= i && n.indexOf(e, r) > -1 : !!i && mr(n, e, r) > -1;
      }
      var Eh = D(function(n, e, r) {
        var t = -1, i = typeof e == "function", f = In(n) ? g(n.length) : [];
        return ze(n, function(a) {
          f[++t] = i ? Nn(e, a, r) : tt(a, e, r);
        }), f;
      }), Ah = Jt(function(n, e, r) {
        be(n, r, e);
      });
      function ri(n, e) {
        var r = O(n) ? J : Wf;
        return r(n, A(e, 3));
      }
      function mh(n, e, r, t) {
        return n == null ? [] : (O(e) || (e = e == null ? [] : [e]), r = t ? o : r, O(r) || (r = r == null ? [] : [r]), Nf(n, e, r));
      }
      var Th = Jt(function(n, e, r) {
        n[r ? 0 : 1].push(e);
      }, function() {
        return [[], []];
      });
      function Ch(n, e, r) {
        var t = O(n) ? Li : sf, i = arguments.length < 3;
        return t(n, A(e, 4), r, i, ze);
      }
      function Sh(n, e, r) {
        var t = O(n) ? Na : sf, i = arguments.length < 3;
        return t(n, A(e, 4), r, i, bf);
      }
      function bh(n, e) {
        var r = O(n) ? He : Of;
        return r(n, ui(A(e, 3)));
      }
      function Oh(n) {
        var e = O(n) ? mf : yl;
        return e(n);
      }
      function Lh(n, e, r) {
        (r ? Tn(n, e, r) : e === o) ? e = 1 : e = I(e);
        var t = O(n) ? ks : El;
        return t(n, e);
      }
      function Ih(n) {
        var e = O(n) ? js : ml;
        return e(n);
      }
      function Ph(n) {
        if (n == null)
          return 0;
        if (In(n))
          return oi(n) ? Cr(n) : n.length;
        var e = Rn(n);
        return e == Mn || e == Bn ? n.size : Xi(n).length;
      }
      function Dh(n, e, r) {
        var t = O(n) ? Ii : Tl;
        return r && Tn(n, e, r) && (e = o), t(n, A(e, 3));
      }
      var Fh = D(function(n, e) {
        if (n == null)
          return [];
        var r = e.length;
        return r > 1 && Tn(n, e[0], e[1]) ? e = [] : r > 2 && Tn(e[0], e[1], e[2]) && (e = [e[0]]), Nf(n, vn(e, 1), []);
      }), ti = ps || function() {
        return _n.Date.now();
      };
      function Wh(n, e) {
        if (typeof e != "function")
          throw new jn(nn);
        return n = I(n), function() {
          if (--n < 1)
            return e.apply(this, arguments);
        };
      }
      function Fo(n, e, r) {
        return e = r ? o : e, e = n && e == null ? n.length : e, Oe(n, dn, o, o, o, o, e);
      }
      function Wo(n, e) {
        var r;
        if (typeof e != "function")
          throw new jn(nn);
        return n = I(n), function() {
          return --n > 0 && (r = e.apply(this, arguments)), n <= 1 && (e = o), r;
        };
      }
      var xu = D(function(n, e, r) {
        var t = rn;
        if (r.length) {
          var i = qe(r, Fr(xu));
          t |= Sn;
        }
        return Oe(n, t, e, r, i);
      }), Uo = D(function(n, e, r) {
        var t = rn | an;
        if (r.length) {
          var i = qe(r, Fr(Uo));
          t |= Sn;
        }
        return Oe(e, t, n, r, i);
      });
      function Mo(n, e, r) {
        e = r ? o : e;
        var t = Oe(n, En, o, o, o, o, o, e);
        return t.placeholder = Mo.placeholder, t;
      }
      function Bo(n, e, r) {
        e = r ? o : e;
        var t = Oe(n, _e, o, o, o, o, o, e);
        return t.placeholder = Bo.placeholder, t;
      }
      function No(n, e, r) {
        var t, i, f, a, l, h, _ = 0, v = !1, d = !1, x = !0;
        if (typeof n != "function")
          throw new jn(nn);
        e = ie(e) || 0, V(r) && (v = !!r.leading, d = "maxWait" in r, f = d ? ln(ie(r.maxWait) || 0, e) : f, x = "trailing" in r ? !!r.trailing : x);
        function E(un) {
          var le = t, Fe = i;
          return t = i = o, _ = un, a = n.apply(Fe, le), a;
        }
        function m(un) {
          return _ = un, l = at(W, e), v ? E(un) : a;
        }
        function P(un) {
          var le = un - h, Fe = un - _, ia = e - le;
          return d ? xn(ia, f - Fe) : ia;
        }
        function T(un) {
          var le = un - h, Fe = un - _;
          return h === o || le >= e || le < 0 || d && Fe >= f;
        }
        function W() {
          var un = ti();
          if (T(un))
            return B(un);
          l = at(W, P(un));
        }
        function B(un) {
          return l = o, x && t ? E(un) : (t = i = o, a);
        }
        function Yn() {
          l !== o && Zf(l), _ = 0, t = h = i = l = o;
        }
        function Cn() {
          return l === o ? a : B(ti());
        }
        function qn() {
          var un = ti(), le = T(un);
          if (t = arguments, i = this, h = un, le) {
            if (l === o)
              return m(h);
            if (d)
              return Zf(l), l = at(W, e), E(h);
          }
          return l === o && (l = at(W, e)), a;
        }
        return qn.cancel = Yn, qn.flush = Cn, qn;
      }
      var Uh = D(function(n, e) {
        return Sf(n, 1, e);
      }), Mh = D(function(n, e, r) {
        return Sf(n, ie(e) || 0, r);
      });
      function Bh(n) {
        return Oe(n, ye);
      }
      function ii(n, e) {
        if (typeof n != "function" || e != null && typeof e != "function")
          throw new jn(nn);
        var r = function() {
          var t = arguments, i = e ? e.apply(this, t) : t[0], f = r.cache;
          if (f.has(i))
            return f.get(i);
          var a = n.apply(this, t);
          return r.cache = f.set(i, a) || f, a;
        };
        return r.cache = new (ii.Cache || Se)(), r;
      }
      ii.Cache = Se;
      function ui(n) {
        if (typeof n != "function")
          throw new jn(nn);
        return function() {
          var e = arguments;
          switch (e.length) {
            case 0:
              return !n.call(this);
            case 1:
              return !n.call(this, e[0]);
            case 2:
              return !n.call(this, e[0], e[1]);
            case 3:
              return !n.call(this, e[0], e[1], e[2]);
          }
          return !n.apply(this, e);
        };
      }
      function Nh(n) {
        return Wo(2, n);
      }
      var $h = Cl(function(n, e) {
        e = e.length == 1 && O(e[0]) ? J(e[0], $n(A())) : J(vn(e, 1), $n(A()));
        var r = e.length;
        return D(function(t) {
          for (var i = -1, f = xn(t.length, r); ++i < f; )
            t[i] = e[i].call(this, t[i]);
          return Nn(n, this, t);
        });
      }), Ru = D(function(n, e) {
        var r = qe(e, Fr(Ru));
        return Oe(n, Sn, o, e, r);
      }), $o = D(function(n, e) {
        var r = qe(e, Fr($o));
        return Oe(n, An, o, e, r);
      }), Gh = Le(function(n, e) {
        return Oe(n, Jn, o, o, o, e);
      });
      function Hh(n, e) {
        if (typeof n != "function")
          throw new jn(nn);
        return e = e === o ? e : I(e), D(n, e);
      }
      function Yh(n, e) {
        if (typeof n != "function")
          throw new jn(nn);
        return e = e == null ? 0 : ln(I(e), 0), D(function(r) {
          var t = r[e], i = Xe(r, 0, e);
          return t && Ye(i, t), Nn(n, this, i);
        });
      }
      function qh(n, e, r) {
        var t = !0, i = !0;
        if (typeof n != "function")
          throw new jn(nn);
        return V(r) && (t = "leading" in r ? !!r.leading : t, i = "trailing" in r ? !!r.trailing : i), No(n, e, {
          leading: t,
          maxWait: e,
          trailing: i
        });
      }
      function Kh(n) {
        return Fo(n, 1);
      }
      function zh(n, e) {
        return Ru(iu(e), n);
      }
      function Jh() {
        if (!arguments.length)
          return [];
        var n = arguments[0];
        return O(n) ? n : [n];
      }
      function Zh(n) {
        return ee(n, Fn);
      }
      function Xh(n, e) {
        return e = typeof e == "function" ? e : o, ee(n, Fn, e);
      }
      function Vh(n) {
        return ee(n, on | Fn);
      }
      function Qh(n, e) {
        return e = typeof e == "function" ? e : o, ee(n, on | Fn, e);
      }
      function kh(n, e) {
        return e == null || Cf(n, e, hn(e));
      }
      function se(n, e) {
        return n === e || n !== n && e !== e;
      }
      var jh = Qt(zi), ng = Qt(function(n, e) {
        return n >= e;
      }), hr = Pf(/* @__PURE__ */ function() {
        return arguments;
      }()) ? Pf : function(n) {
        return j(n) && Y.call(n, "callee") && !wf.call(n, "callee");
      }, O = g.isArray, eg = ku ? $n(ku) : sl;
      function In(n) {
        return n != null && fi(n.length) && !Pe(n);
      }
      function tn(n) {
        return j(n) && In(n);
      }
      function rg(n) {
        return n === !0 || n === !1 || j(n) && mn(n) == ve;
      }
      var Ve = vs || Iu, tg = ju ? $n(ju) : ll;
      function ig(n) {
        return j(n) && n.nodeType === 1 && !st(n);
      }
      function ug(n) {
        if (n == null)
          return !0;
        if (In(n) && (O(n) || typeof n == "string" || typeof n.splice == "function" || Ve(n) || Wr(n) || hr(n)))
          return !n.length;
        var e = Rn(n);
        if (e == Mn || e == Bn)
          return !n.size;
        if (ot(n))
          return !Xi(n).length;
        for (var r in n)
          if (Y.call(n, r))
            return !1;
        return !0;
      }
      function fg(n, e) {
        return it(n, e);
      }
      function og(n, e, r) {
        r = typeof r == "function" ? r : o;
        var t = r ? r(n, e) : o;
        return t === o ? it(n, e, o, r) : !!t;
      }
      function yu(n) {
        if (!j(n))
          return !1;
        var e = mn(n);
        return e == vr || e == _r || typeof n.message == "string" && typeof n.name == "string" && !st(n);
      }
      function ag(n) {
        return typeof n == "number" && Rf(n);
      }
      function Pe(n) {
        if (!V(n))
          return !1;
        var e = mn(n);
        return e == nr || e == gt || e == Mr || e == dr;
      }
      function Go(n) {
        return typeof n == "number" && n == I(n);
      }
      function fi(n) {
        return typeof n == "number" && n > -1 && n % 1 == 0 && n <= L;
      }
      function V(n) {
        var e = typeof n;
        return n != null && (e == "object" || e == "function");
      }
      function j(n) {
        return n != null && typeof n == "object";
      }
      var Ho = nf ? $n(nf) : hl;
      function sg(n, e) {
        return n === e || Zi(n, e, cu(e));
      }
      function lg(n, e, r) {
        return r = typeof r == "function" ? r : o, Zi(n, e, cu(e), r);
      }
      function cg(n) {
        return Yo(n) && n != +n;
      }
      function hg(n) {
        if (Xl(n))
          throw new S(We);
        return Df(n);
      }
      function gg(n) {
        return n === null;
      }
      function pg(n) {
        return n == null;
      }
      function Yo(n) {
        return typeof n == "number" || j(n) && mn(n) == de;
      }
      function st(n) {
        if (!j(n) || mn(n) != Xn)
          return !1;
        var e = Dt(n);
        if (e === null)
          return !0;
        var r = Y.call(e, "constructor") && e.constructor;
        return typeof r == "function" && r instanceof r && Ot.call(r) == ls;
      }
      var Eu = ef ? $n(ef) : gl;
      function _g(n) {
        return Go(n) && n >= -L && n <= L;
      }
      var qo = rf ? $n(rf) : pl;
      function oi(n) {
        return typeof n == "string" || !O(n) && j(n) && mn(n) == me;
      }
      function Hn(n) {
        return typeof n == "symbol" || j(n) && mn(n) == wr;
      }
      var Wr = tf ? $n(tf) : _l;
      function vg(n) {
        return n === o;
      }
      function dg(n) {
        return j(n) && Rn(n) == Ne;
      }
      function wg(n) {
        return j(n) && mn(n) == pt;
      }
      var xg = Qt(Vi), Rg = Qt(function(n, e) {
        return n <= e;
      });
      function Ko(n) {
        if (!n)
          return [];
        if (In(n))
          return oi(n) ? oe(n) : Ln(n);
        if (Vr && n[Vr])
          return ka(n[Vr]());
        var e = Rn(n), r = e == Mn ? Mi : e == Bn ? Ct : Ur;
        return r(n);
      }
      function De(n) {
        if (!n)
          return n === 0 ? n : 0;
        if (n = ie(n), n === Zn || n === -Zn) {
          var e = n < 0 ? -1 : 1;
          return e * On;
        }
        return n === n ? n : 0;
      }
      function I(n) {
        var e = De(n), r = e % 1;
        return e === e ? r ? e - r : e : 0;
      }
      function zo(n) {
        return n ? ar(I(n), 0, Z) : 0;
      }
      function ie(n) {
        if (typeof n == "number")
          return n;
        if (Hn(n))
          return cn;
        if (V(n)) {
          var e = typeof n.valueOf == "function" ? n.valueOf() : n;
          n = V(e) ? e + "" : e;
        }
        if (typeof n != "string")
          return n === 0 ? n : +n;
        n = lf(n);
        var r = U.test(n);
        return r || gn.test(n) ? Wa(n.slice(2), r ? 2 : 8) : H.test(n) ? cn : +n;
      }
      function Jo(n) {
        return xe(n, Pn(n));
      }
      function yg(n) {
        return n ? ar(I(n), -L, L) : n === 0 ? n : 0;
      }
      function G(n) {
        return n == null ? "" : Gn(n);
      }
      var Eg = Pr(function(n, e) {
        if (ot(e) || In(e)) {
          xe(e, hn(e), n);
          return;
        }
        for (var r in e)
          Y.call(e, r) && et(n, r, e[r]);
      }), Zo = Pr(function(n, e) {
        xe(e, Pn(e), n);
      }), ai = Pr(function(n, e, r, t) {
        xe(e, Pn(e), n, t);
      }), Ag = Pr(function(n, e, r, t) {
        xe(e, hn(e), n, t);
      }), mg = Le(Yi);
      function Tg(n, e) {
        var r = Ir(n);
        return e == null ? r : Tf(r, e);
      }
      var Cg = D(function(n, e) {
        n = q(n);
        var r = -1, t = e.length, i = t > 2 ? e[2] : o;
        for (i && Tn(e[0], e[1], i) && (t = 1); ++r < t; )
          for (var f = e[r], a = Pn(f), l = -1, h = a.length; ++l < h; ) {
            var _ = a[l], v = n[_];
            (v === o || se(v, br[_]) && !Y.call(n, _)) && (n[_] = f[_]);
          }
        return n;
      }), Sg = D(function(n) {
        return n.push(o, lo), Nn(Xo, o, n);
      });
      function bg(n, e) {
        return ff(n, A(e, 3), we);
      }
      function Og(n, e) {
        return ff(n, A(e, 3), Ki);
      }
      function Lg(n, e) {
        return n == null ? n : qi(n, A(e, 3), Pn);
      }
      function Ig(n, e) {
        return n == null ? n : Lf(n, A(e, 3), Pn);
      }
      function Pg(n, e) {
        return n && we(n, A(e, 3));
      }
      function Dg(n, e) {
        return n && Ki(n, A(e, 3));
      }
      function Fg(n) {
        return n == null ? [] : Yt(n, hn(n));
      }
      function Wg(n) {
        return n == null ? [] : Yt(n, Pn(n));
      }
      function Au(n, e, r) {
        var t = n == null ? o : sr(n, e);
        return t === o ? r : t;
      }
      function Ug(n, e) {
        return n != null && go(n, e, ul);
      }
      function mu(n, e) {
        return n != null && go(n, e, fl);
      }
      var Mg = uo(function(n, e, r) {
        e != null && typeof e.toString != "function" && (e = Lt.call(e)), n[e] = r;
      }, Cu(Dn)), Bg = uo(function(n, e, r) {
        e != null && typeof e.toString != "function" && (e = Lt.call(e)), Y.call(n, e) ? n[e].push(r) : n[e] = [r];
      }, A), Ng = D(tt);
      function hn(n) {
        return In(n) ? Af(n) : Xi(n);
      }
      function Pn(n) {
        return In(n) ? Af(n, !0) : vl(n);
      }
      function $g(n, e) {
        var r = {};
        return e = A(e, 3), we(n, function(t, i, f) {
          be(r, e(t, i, f), t);
        }), r;
      }
      function Gg(n, e) {
        var r = {};
        return e = A(e, 3), we(n, function(t, i, f) {
          be(r, i, e(t, i, f));
        }), r;
      }
      var Hg = Pr(function(n, e, r) {
        qt(n, e, r);
      }), Xo = Pr(function(n, e, r, t) {
        qt(n, e, r, t);
      }), Yg = Le(function(n, e) {
        var r = {};
        if (n == null)
          return r;
        var t = !1;
        e = J(e, function(f) {
          return f = Ze(f, n), t || (t = f.length > 1), f;
        }), xe(n, su(n), r), t && (r = ee(r, on | zn | Fn, Ml));
        for (var i = e.length; i--; )
          eu(r, e[i]);
        return r;
      });
      function qg(n, e) {
        return Vo(n, ui(A(e)));
      }
      var Kg = Le(function(n, e) {
        return n == null ? {} : wl(n, e);
      });
      function Vo(n, e) {
        if (n == null)
          return {};
        var r = J(su(n), function(t) {
          return [t];
        });
        return e = A(e), $f(n, r, function(t, i) {
          return e(t, i[0]);
        });
      }
      function zg(n, e, r) {
        e = Ze(e, n);
        var t = -1, i = e.length;
        for (i || (i = 1, n = o); ++t < i; ) {
          var f = n == null ? o : n[Re(e[t])];
          f === o && (t = i, f = r), n = Pe(f) ? f.call(n) : f;
        }
        return n;
      }
      function Jg(n, e, r) {
        return n == null ? n : ut(n, e, r);
      }
      function Zg(n, e, r, t) {
        return t = typeof t == "function" ? t : o, n == null ? n : ut(n, e, r, t);
      }
      var Qo = ao(hn), ko = ao(Pn);
      function Xg(n, e, r) {
        var t = O(n), i = t || Ve(n) || Wr(n);
        if (e = A(e, 4), r == null) {
          var f = n && n.constructor;
          i ? r = t ? new f() : [] : V(n) ? r = Pe(f) ? Ir(Dt(n)) : {} : r = {};
        }
        return (i ? kn : we)(n, function(a, l, h) {
          return e(r, a, l, h);
        }), r;
      }
      function Vg(n, e) {
        return n == null ? !0 : eu(n, e);
      }
      function Qg(n, e, r) {
        return n == null ? n : Kf(n, e, iu(r));
      }
      function kg(n, e, r, t) {
        return t = typeof t == "function" ? t : o, n == null ? n : Kf(n, e, iu(r), t);
      }
      function Ur(n) {
        return n == null ? [] : Ui(n, hn(n));
      }
      function jg(n) {
        return n == null ? [] : Ui(n, Pn(n));
      }
      function np(n, e, r) {
        return r === o && (r = e, e = o), r !== o && (r = ie(r), r = r === r ? r : 0), e !== o && (e = ie(e), e = e === e ? e : 0), ar(ie(n), e, r);
      }
      function ep(n, e, r) {
        return e = De(e), r === o ? (r = e, e = 0) : r = De(r), n = ie(n), ol(n, e, r);
      }
      function rp(n, e, r) {
        if (r && typeof r != "boolean" && Tn(n, e, r) && (e = r = o), r === o && (typeof e == "boolean" ? (r = e, e = o) : typeof n == "boolean" && (r = n, n = o)), n === o && e === o ? (n = 0, e = 1) : (n = De(n), e === o ? (e = n, n = 0) : e = De(e)), n > e) {
          var t = n;
          n = e, e = t;
        }
        if (r || n % 1 || e % 1) {
          var i = yf();
          return xn(n + i * (e - n + Fa("1e-" + ((i + "").length - 1))), e);
        }
        return ki(n, e);
      }
      var tp = Dr(function(n, e, r) {
        return e = e.toLowerCase(), n + (r ? jo(e) : e);
      });
      function jo(n) {
        return Tu(G(n).toLowerCase());
      }
      function na(n) {
        return n = G(n), n && n.replace(k, Ja).replace(ma, "");
      }
      function ip(n, e, r) {
        n = G(n), e = Gn(e);
        var t = n.length;
        r = r === o ? t : ar(I(r), 0, t);
        var i = r;
        return r -= e.length, r >= 0 && n.slice(r, i) == e;
      }
      function up(n) {
        return n = G(n), n && vt.test(n) ? n.replace(yr, Za) : n;
      }
      function fp(n) {
        return n = G(n), n && yt.test(n) ? n.replace(Jr, "\\$&") : n;
      }
      var op = Dr(function(n, e, r) {
        return n + (r ? "-" : "") + e.toLowerCase();
      }), ap = Dr(function(n, e, r) {
        return n + (r ? " " : "") + e.toLowerCase();
      }), sp = ro("toLowerCase");
      function lp(n, e, r) {
        n = G(n), e = I(e);
        var t = e ? Cr(n) : 0;
        if (!e || t >= e)
          return n;
        var i = (e - t) / 2;
        return Vt(Mt(i), r) + n + Vt(Ut(i), r);
      }
      function cp(n, e, r) {
        n = G(n), e = I(e);
        var t = e ? Cr(n) : 0;
        return e && t < e ? n + Vt(e - t, r) : n;
      }
      function hp(n, e, r) {
        n = G(n), e = I(e);
        var t = e ? Cr(n) : 0;
        return e && t < e ? Vt(e - t, r) + n : n;
      }
      function gp(n, e, r) {
        return r || e == null ? e = 0 : e && (e = +e), Rs(G(n).replace(Er, ""), e || 0);
      }
      function pp(n, e, r) {
        return (r ? Tn(n, e, r) : e === o) ? e = 1 : e = I(e), ji(G(n), e);
      }
      function _p() {
        var n = arguments, e = G(n[0]);
        return n.length < 3 ? e : e.replace(n[1], n[2]);
      }
      var vp = Dr(function(n, e, r) {
        return n + (r ? "_" : "") + e.toLowerCase();
      });
      function dp(n, e, r) {
        return r && typeof r != "number" && Tn(n, e, r) && (e = r = o), r = r === o ? Z : r >>> 0, r ? (n = G(n), n && (typeof e == "string" || e != null && !Eu(e)) && (e = Gn(e), !e && Tr(n)) ? Xe(oe(n), 0, r) : n.split(e, r)) : [];
      }
      var wp = Dr(function(n, e, r) {
        return n + (r ? " " : "") + Tu(e);
      });
      function xp(n, e, r) {
        return n = G(n), r = r == null ? 0 : ar(I(r), 0, n.length), e = Gn(e), n.slice(r, r + e.length) == e;
      }
      function Rp(n, e, r) {
        var t = u.templateSettings;
        r && Tn(n, e, r) && (e = o), n = G(n), e = ai({}, e, t, so);
        var i = ai({}, e.imports, t.imports, so), f = hn(i), a = Ui(i, f), l, h, _ = 0, v = e.interpolate || pn, d = "__p += '", x = Bi(
          (e.escape || pn).source + "|" + v.source + "|" + (v === wt ? C : pn).source + "|" + (e.evaluate || pn).source + "|$",
          "g"
        ), E = "//# sourceURL=" + (Y.call(e, "sourceURL") ? (e.sourceURL + "").replace(/\s/g, " ") : "lodash.templateSources[" + ++Oa + "]") + `
`;
        n.replace(x, function(T, W, B, Yn, Cn, qn) {
          return B || (B = Yn), d += n.slice(_, qn).replace(Ge, Xa), W && (l = !0, d += `' +
__e(` + W + `) +
'`), Cn && (h = !0, d += `';
` + Cn + `;
__p += '`), B && (d += `' +
((__t = (` + B + `)) == null ? '' : __t) +
'`), _ = qn + T.length, T;
        }), d += `';
`;
        var m = Y.call(e, "variable") && e.variable;
        if (!m)
          d = `with (obj) {
` + d + `
}
`;
        else if (w.test(m))
          throw new S(Kn);
        d = (h ? d.replace(pi, "") : d).replace(Kr, "$1").replace(_t, "$1;"), d = "function(" + (m || "obj") + `) {
` + (m ? "" : `obj || (obj = {});
`) + "var __t, __p = ''" + (l ? ", __e = _.escape" : "") + (h ? `, __j = Array.prototype.join;
function print() { __p += __j.call(arguments, '') }
` : `;
`) + d + `return __p
}`;
        var P = ra(function() {
          return $(f, E + "return " + d).apply(o, a);
        });
        if (P.source = d, yu(P))
          throw P;
        return P;
      }
      function yp(n) {
        return G(n).toLowerCase();
      }
      function Ep(n) {
        return G(n).toUpperCase();
      }
      function Ap(n, e, r) {
        if (n = G(n), n && (r || e === o))
          return lf(n);
        if (!n || !(e = Gn(e)))
          return n;
        var t = oe(n), i = oe(e), f = cf(t, i), a = hf(t, i) + 1;
        return Xe(t, f, a).join("");
      }
      function mp(n, e, r) {
        if (n = G(n), n && (r || e === o))
          return n.slice(0, pf(n) + 1);
        if (!n || !(e = Gn(e)))
          return n;
        var t = oe(n), i = hf(t, oe(e)) + 1;
        return Xe(t, 0, i).join("");
      }
      function Tp(n, e, r) {
        if (n = G(n), n && (r || e === o))
          return n.replace(Er, "");
        if (!n || !(e = Gn(e)))
          return n;
        var t = oe(n), i = cf(t, oe(e));
        return Xe(t, i).join("");
      }
      function Cp(n, e) {
        var r = Qe, t = ke;
        if (V(e)) {
          var i = "separator" in e ? e.separator : i;
          r = "length" in e ? I(e.length) : r, t = "omission" in e ? Gn(e.omission) : t;
        }
        n = G(n);
        var f = n.length;
        if (Tr(n)) {
          var a = oe(n);
          f = a.length;
        }
        if (r >= f)
          return n;
        var l = r - Cr(t);
        if (l < 1)
          return t;
        var h = a ? Xe(a, 0, l).join("") : n.slice(0, l);
        if (i === o)
          return h + t;
        if (a && (l += h.length - l), Eu(i)) {
          if (n.slice(l).search(i)) {
            var _, v = h;
            for (i.global || (i = Bi(i.source, G(N.exec(i)) + "g")), i.lastIndex = 0; _ = i.exec(v); )
              var d = _.index;
            h = h.slice(0, d === o ? l : d);
          }
        } else if (n.indexOf(Gn(i), l) != l) {
          var x = h.lastIndexOf(i);
          x > -1 && (h = h.slice(0, x));
        }
        return h + t;
      }
      function Sp(n) {
        return n = G(n), n && zr.test(n) ? n.replace(Te, rs) : n;
      }
      var bp = Dr(function(n, e, r) {
        return n + (r ? " " : "") + e.toUpperCase();
      }), Tu = ro("toUpperCase");
      function ea(n, e, r) {
        return n = G(n), e = r ? o : e, e === o ? Qa(n) ? us(n) : Ha(n) : n.match(e) || [];
      }
      var ra = D(function(n, e) {
        try {
          return Nn(n, o, e);
        } catch (r) {
          return yu(r) ? r : new S(r);
        }
      }), Op = Le(function(n, e) {
        return kn(e, function(r) {
          r = Re(r), be(n, r, xu(n[r], n));
        }), n;
      });
      function Lp(n) {
        var e = n == null ? 0 : n.length, r = A();
        return n = e ? J(n, function(t) {
          if (typeof t[1] != "function")
            throw new jn(nn);
          return [r(t[0]), t[1]];
        }) : [], D(function(t) {
          for (var i = -1; ++i < e; ) {
            var f = n[i];
            if (Nn(f[0], this, t))
              return Nn(f[1], this, t);
          }
        });
      }
      function Ip(n) {
        return rl(ee(n, on));
      }
      function Cu(n) {
        return function() {
          return n;
        };
      }
      function Pp(n, e) {
        return n == null || n !== n ? e : n;
      }
      var Dp = io(), Fp = io(!0);
      function Dn(n) {
        return n;
      }
      function Su(n) {
        return Ff(typeof n == "function" ? n : ee(n, on));
      }
      function Wp(n) {
        return Uf(ee(n, on));
      }
      function Up(n, e) {
        return Mf(n, ee(e, on));
      }
      var Mp = D(function(n, e) {
        return function(r) {
          return tt(r, n, e);
        };
      }), Bp = D(function(n, e) {
        return function(r) {
          return tt(n, r, e);
        };
      });
      function bu(n, e, r) {
        var t = hn(e), i = Yt(e, t);
        r == null && !(V(e) && (i.length || !t.length)) && (r = e, e = n, n = this, i = Yt(e, hn(e)));
        var f = !(V(r) && "chain" in r) || !!r.chain, a = Pe(n);
        return kn(i, function(l) {
          var h = e[l];
          n[l] = h, a && (n.prototype[l] = function() {
            var _ = this.__chain__;
            if (f || _) {
              var v = n(this.__wrapped__), d = v.__actions__ = Ln(this.__actions__);
              return d.push({ func: h, args: arguments, thisArg: n }), v.__chain__ = _, v;
            }
            return h.apply(n, Ye([this.value()], arguments));
          });
        }), n;
      }
      function Np() {
        return _n._ === this && (_n._ = cs), this;
      }
      function Ou() {
      }
      function $p(n) {
        return n = I(n), D(function(e) {
          return Bf(e, n);
        });
      }
      var Gp = fu(J), Hp = fu(uf), Yp = fu(Ii);
      function ta(n) {
        return gu(n) ? Pi(Re(n)) : xl(n);
      }
      function qp(n) {
        return function(e) {
          return n == null ? o : sr(n, e);
        };
      }
      var Kp = fo(), zp = fo(!0);
      function Lu() {
        return [];
      }
      function Iu() {
        return !1;
      }
      function Jp() {
        return {};
      }
      function Zp() {
        return "";
      }
      function Xp() {
        return !0;
      }
      function Vp(n, e) {
        if (n = I(n), n < 1 || n > L)
          return [];
        var r = Z, t = xn(n, Z);
        e = A(e), n -= Z;
        for (var i = Wi(t, e); ++r < n; )
          e(r);
        return i;
      }
      function Qp(n) {
        return O(n) ? J(n, Re) : Hn(n) ? [n] : Ln(Ao(G(n)));
      }
      function kp(n) {
        var e = ++ss;
        return G(n) + e;
      }
      var jp = Xt(function(n, e) {
        return n + e;
      }, 0), n_ = ou("ceil"), e_ = Xt(function(n, e) {
        return n / e;
      }, 1), r_ = ou("floor");
      function t_(n) {
        return n && n.length ? Ht(n, Dn, zi) : o;
      }
      function i_(n, e) {
        return n && n.length ? Ht(n, A(e, 2), zi) : o;
      }
      function u_(n) {
        return af(n, Dn);
      }
      function f_(n, e) {
        return af(n, A(e, 2));
      }
      function o_(n) {
        return n && n.length ? Ht(n, Dn, Vi) : o;
      }
      function a_(n, e) {
        return n && n.length ? Ht(n, A(e, 2), Vi) : o;
      }
      var s_ = Xt(function(n, e) {
        return n * e;
      }, 1), l_ = ou("round"), c_ = Xt(function(n, e) {
        return n - e;
      }, 0);
      function h_(n) {
        return n && n.length ? Fi(n, Dn) : 0;
      }
      function g_(n, e) {
        return n && n.length ? Fi(n, A(e, 2)) : 0;
      }
      return u.after = Wh, u.ary = Fo, u.assign = Eg, u.assignIn = Zo, u.assignInWith = ai, u.assignWith = Ag, u.at = mg, u.before = Wo, u.bind = xu, u.bindAll = Op, u.bindKey = Uo, u.castArray = Jh, u.chain = Io, u.chunk = rc, u.compact = tc, u.concat = ic, u.cond = Lp, u.conforms = Ip, u.constant = Cu, u.countBy = hh, u.create = Tg, u.curry = Mo, u.curryRight = Bo, u.debounce = No, u.defaults = Cg, u.defaultsDeep = Sg, u.defer = Uh, u.delay = Mh, u.difference = uc, u.differenceBy = fc, u.differenceWith = oc, u.drop = ac, u.dropRight = sc, u.dropRightWhile = lc, u.dropWhile = cc, u.fill = hc, u.filter = ph, u.flatMap = dh, u.flatMapDeep = wh, u.flatMapDepth = xh, u.flatten = So, u.flattenDeep = gc, u.flattenDepth = pc, u.flip = Bh, u.flow = Dp, u.flowRight = Fp, u.fromPairs = _c, u.functions = Fg, u.functionsIn = Wg, u.groupBy = Rh, u.initial = dc, u.intersection = wc, u.intersectionBy = xc, u.intersectionWith = Rc, u.invert = Mg, u.invertBy = Bg, u.invokeMap = Eh, u.iteratee = Su, u.keyBy = Ah, u.keys = hn, u.keysIn = Pn, u.map = ri, u.mapKeys = $g, u.mapValues = Gg, u.matches = Wp, u.matchesProperty = Up, u.memoize = ii, u.merge = Hg, u.mergeWith = Xo, u.method = Mp, u.methodOf = Bp, u.mixin = bu, u.negate = ui, u.nthArg = $p, u.omit = Yg, u.omitBy = qg, u.once = Nh, u.orderBy = mh, u.over = Gp, u.overArgs = $h, u.overEvery = Hp, u.overSome = Yp, u.partial = Ru, u.partialRight = $o, u.partition = Th, u.pick = Kg, u.pickBy = Vo, u.property = ta, u.propertyOf = qp, u.pull = mc, u.pullAll = Oo, u.pullAllBy = Tc, u.pullAllWith = Cc, u.pullAt = Sc, u.range = Kp, u.rangeRight = zp, u.rearg = Gh, u.reject = bh, u.remove = bc, u.rest = Hh, u.reverse = du, u.sampleSize = Lh, u.set = Jg, u.setWith = Zg, u.shuffle = Ih, u.slice = Oc, u.sortBy = Fh, u.sortedUniq = Uc, u.sortedUniqBy = Mc, u.split = dp, u.spread = Yh, u.tail = Bc, u.take = Nc, u.takeRight = $c, u.takeRightWhile = Gc, u.takeWhile = Hc, u.tap = th, u.throttle = qh, u.thru = ei, u.toArray = Ko, u.toPairs = Qo, u.toPairsIn = ko, u.toPath = Qp, u.toPlainObject = Jo, u.transform = Xg, u.unary = Kh, u.union = Yc, u.unionBy = qc, u.unionWith = Kc, u.uniq = zc, u.uniqBy = Jc, u.uniqWith = Zc, u.unset = Vg, u.unzip = wu, u.unzipWith = Lo, u.update = Qg, u.updateWith = kg, u.values = Ur, u.valuesIn = jg, u.without = Xc, u.words = ea, u.wrap = zh, u.xor = Vc, u.xorBy = Qc, u.xorWith = kc, u.zip = jc, u.zipObject = nh, u.zipObjectDeep = eh, u.zipWith = rh, u.entries = Qo, u.entriesIn = ko, u.extend = Zo, u.extendWith = ai, bu(u, u), u.add = jp, u.attempt = ra, u.camelCase = tp, u.capitalize = jo, u.ceil = n_, u.clamp = np, u.clone = Zh, u.cloneDeep = Vh, u.cloneDeepWith = Qh, u.cloneWith = Xh, u.conformsTo = kh, u.deburr = na, u.defaultTo = Pp, u.divide = e_, u.endsWith = ip, u.eq = se, u.escape = up, u.escapeRegExp = fp, u.every = gh, u.find = _h, u.findIndex = To, u.findKey = bg, u.findLast = vh, u.findLastIndex = Co, u.findLastKey = Og, u.floor = r_, u.forEach = Po, u.forEachRight = Do, u.forIn = Lg, u.forInRight = Ig, u.forOwn = Pg, u.forOwnRight = Dg, u.get = Au, u.gt = jh, u.gte = ng, u.has = Ug, u.hasIn = mu, u.head = bo, u.identity = Dn, u.includes = yh, u.indexOf = vc, u.inRange = ep, u.invoke = Ng, u.isArguments = hr, u.isArray = O, u.isArrayBuffer = eg, u.isArrayLike = In, u.isArrayLikeObject = tn, u.isBoolean = rg, u.isBuffer = Ve, u.isDate = tg, u.isElement = ig, u.isEmpty = ug, u.isEqual = fg, u.isEqualWith = og, u.isError = yu, u.isFinite = ag, u.isFunction = Pe, u.isInteger = Go, u.isLength = fi, u.isMap = Ho, u.isMatch = sg, u.isMatchWith = lg, u.isNaN = cg, u.isNative = hg, u.isNil = pg, u.isNull = gg, u.isNumber = Yo, u.isObject = V, u.isObjectLike = j, u.isPlainObject = st, u.isRegExp = Eu, u.isSafeInteger = _g, u.isSet = qo, u.isString = oi, u.isSymbol = Hn, u.isTypedArray = Wr, u.isUndefined = vg, u.isWeakMap = dg, u.isWeakSet = wg, u.join = yc, u.kebabCase = op, u.last = te, u.lastIndexOf = Ec, u.lowerCase = ap, u.lowerFirst = sp, u.lt = xg, u.lte = Rg, u.max = t_, u.maxBy = i_, u.mean = u_, u.meanBy = f_, u.min = o_, u.minBy = a_, u.stubArray = Lu, u.stubFalse = Iu, u.stubObject = Jp, u.stubString = Zp, u.stubTrue = Xp, u.multiply = s_, u.nth = Ac, u.noConflict = Np, u.noop = Ou, u.now = ti, u.pad = lp, u.padEnd = cp, u.padStart = hp, u.parseInt = gp, u.random = rp, u.reduce = Ch, u.reduceRight = Sh, u.repeat = pp, u.replace = _p, u.result = zg, u.round = l_, u.runInContext = c, u.sample = Oh, u.size = Ph, u.snakeCase = vp, u.some = Dh, u.sortedIndex = Lc, u.sortedIndexBy = Ic, u.sortedIndexOf = Pc, u.sortedLastIndex = Dc, u.sortedLastIndexBy = Fc, u.sortedLastIndexOf = Wc, u.startCase = wp, u.startsWith = xp, u.subtract = c_, u.sum = h_, u.sumBy = g_, u.template = Rp, u.times = Vp, u.toFinite = De, u.toInteger = I, u.toLength = zo, u.toLower = yp, u.toNumber = ie, u.toSafeInteger = yg, u.toString = G, u.toUpper = Ep, u.trim = Ap, u.trimEnd = mp, u.trimStart = Tp, u.truncate = Cp, u.unescape = Sp, u.uniqueId = kp, u.upperCase = bp, u.upperFirst = Tu, u.each = Po, u.eachRight = Do, u.first = bo, bu(u, function() {
        var n = {};
        return we(u, function(e, r) {
          Y.call(u.prototype, r) || (n[r] = e);
        }), n;
      }(), { chain: !1 }), u.VERSION = he, kn(["bind", "bindKey", "curry", "curryRight", "partial", "partialRight"], function(n) {
        u[n].placeholder = u;
      }), kn(["drop", "take"], function(n, e) {
        M.prototype[n] = function(r) {
          r = r === o ? 1 : ln(I(r), 0);
          var t = this.__filtered__ && !e ? new M(this) : this.clone();
          return t.__filtered__ ? t.__takeCount__ = xn(r, t.__takeCount__) : t.__views__.push({
            size: xn(r, Z),
            type: n + (t.__dir__ < 0 ? "Right" : "")
          }), t;
        }, M.prototype[n + "Right"] = function(r) {
          return this.reverse()[n](r).reverse();
        };
      }), kn(["filter", "map", "takeWhile"], function(n, e) {
        var r = e + 1, t = r == Un || r == pr;
        M.prototype[n] = function(i) {
          var f = this.clone();
          return f.__iteratees__.push({
            iteratee: A(i, 3),
            type: r
          }), f.__filtered__ = f.__filtered__ || t, f;
        };
      }), kn(["head", "last"], function(n, e) {
        var r = "take" + (e ? "Right" : "");
        M.prototype[n] = function() {
          return this[r](1).value()[0];
        };
      }), kn(["initial", "tail"], function(n, e) {
        var r = "drop" + (e ? "" : "Right");
        M.prototype[n] = function() {
          return this.__filtered__ ? new M(this) : this[r](1);
        };
      }), M.prototype.compact = function() {
        return this.filter(Dn);
      }, M.prototype.find = function(n) {
        return this.filter(n).head();
      }, M.prototype.findLast = function(n) {
        return this.reverse().find(n);
      }, M.prototype.invokeMap = D(function(n, e) {
        return typeof n == "function" ? new M(this) : this.map(function(r) {
          return tt(r, n, e);
        });
      }), M.prototype.reject = function(n) {
        return this.filter(ui(A(n)));
      }, M.prototype.slice = function(n, e) {
        n = I(n);
        var r = this;
        return r.__filtered__ && (n > 0 || e < 0) ? new M(r) : (n < 0 ? r = r.takeRight(-n) : n && (r = r.drop(n)), e !== o && (e = I(e), r = e < 0 ? r.dropRight(-e) : r.take(e - n)), r);
      }, M.prototype.takeRightWhile = function(n) {
        return this.reverse().takeWhile(n).reverse();
      }, M.prototype.toArray = function() {
        return this.take(Z);
      }, we(M.prototype, function(n, e) {
        var r = /^(?:filter|find|map|reject)|While$/.test(e), t = /^(?:head|last)$/.test(e), i = u[t ? "take" + (e == "last" ? "Right" : "") : e], f = t || /^find/.test(e);
        i && (u.prototype[e] = function() {
          var a = this.__wrapped__, l = t ? [1] : arguments, h = a instanceof M, _ = l[0], v = h || O(a), d = function(W) {
            var B = i.apply(u, Ye([W], l));
            return t && x ? B[0] : B;
          };
          v && r && typeof _ == "function" && _.length != 1 && (h = v = !1);
          var x = this.__chain__, E = !!this.__actions__.length, m = f && !x, P = h && !E;
          if (!f && v) {
            a = P ? a : new M(this);
            var T = n.apply(a, l);
            return T.__actions__.push({ func: ei, args: [d], thisArg: o }), new ne(T, x);
          }
          return m && P ? n.apply(this, l) : (T = this.thru(d), m ? t ? T.value()[0] : T.value() : T);
        });
      }), kn(["pop", "push", "shift", "sort", "splice", "unshift"], function(n) {
        var e = St[n], r = /^(?:push|sort|unshift)$/.test(n) ? "tap" : "thru", t = /^(?:pop|shift)$/.test(n);
        u.prototype[n] = function() {
          var i = arguments;
          if (t && !this.__chain__) {
            var f = this.value();
            return e.apply(O(f) ? f : [], i);
          }
          return this[r](function(a) {
            return e.apply(O(a) ? a : [], i);
          });
        };
      }), we(M.prototype, function(n, e) {
        var r = u[e];
        if (r) {
          var t = r.name + "";
          Y.call(Lr, t) || (Lr[t] = []), Lr[t].push({ name: e, func: r });
        }
      }), Lr[Zt(o, an).name] = [{
        name: "wrapper",
        func: o
      }], M.prototype.clone = Ss, M.prototype.reverse = bs, M.prototype.value = Os, u.prototype.at = ih, u.prototype.chain = uh, u.prototype.commit = fh, u.prototype.next = oh, u.prototype.plant = sh, u.prototype.reverse = lh, u.prototype.toJSON = u.prototype.valueOf = u.prototype.value = ch, u.prototype.first = u.prototype.head, Vr && (u.prototype[Vr] = ah), u;
    }, Sr = fs();
    ir ? ((ir.exports = Sr)._ = Sr, Si._ = Sr) : _n._ = Sr;
  }).call(lt);
})(ci, ci.exports);
var E_ = ci.exports;
const oa = (Q) => ({
  [ce.JOINED]: Q.onParticipantJoined,
  [ce.LEFT]: Q.onParticipantLeft,
  [ce.LIST_UPDATED]: Q.onParticipantListUpdated,
  [ce.LOCAL_JOINED]: Q.onParticipantLocalJoined,
  [ce.LOCAL_LEFT]: Q.onParticipantLocalLeft,
  [ce.LOCAL_UPDATED]: Q.onParticipantLocalUpdated
}), sa = p_({
  hasProvider: !1,
  room: null
});
function T_(Q) {
  const { hasProvider: yn, ...o } = __(sa);
  return li(() => {
    yn || console.error("[SuperViz] You need to wrap your component with SuperVizRoomProvider");
  }, [yn]), Q ? {
    ...o,
    component: o.activeComponents[Q]
  } : o;
}
function C_(Q) {
  const {
    roomId: yn,
    group: o,
    debug: he,
    children: ge,
    environment: We,
    participant: nn,
    developerKey: Kn,
    customColors: en,
    onParticipantJoined: pe,
    onParticipantLeft: fn,
    onParticipantListUpdated: on,
    onParticipantLocalJoined: zn,
    onParticipantLocalLeft: Fn,
    onParticipantLocalUpdated: Wn,
    stopAutoStart: Ue
  } = Q, [rn, an] = si(!1), [b, En] = si(null), [_e, Sn] = si(null), [An, dn] = si({}), Jn = v_(!1);
  li(() => {
    if (!(!Qe || rn || Ue))
      return Jn.current || Qe(), () => {
        !ke || !rn || ke();
      };
  }, []), li(() => {
    const L = oa(Q);
    !b || E_.isEqual(_e, L) || ye(b, Q);
  }, [
    pe,
    fn,
    on,
    zn,
    Fn,
    Wn,
    b
  ]), li(() => {
    rn && (b == null || b.unsubscribe(ce.LOCAL_UPDATED), b == null || b.subscribe(ce.LOCAL_UPDATED, Un));
  }, [b, An]);
  function ye(L, On) {
    const cn = oa(On);
    Object.entries(cn).forEach(([Z, ue]) => {
      const je = On[Z];
      (!ue && je || ue !== je) && L.unsubscribe(Z), ue && L.subscribe(Z, ue);
    }), Sn(cn);
  }
  async function Qe() {
    if (rn || b) {
      console.log("[SuperViz] Room already started");
      return;
    }
    try {
      Jn.current = !0;
      const L = await w_(Kn, {
        participant: nn,
        roomId: yn,
        group: o,
        environment: We,
        debug: he,
        customColors: en
      });
      ye(L, Q), En(L), an(!0), L.subscribe(ce.LOCAL_UPDATED, Un), L.subscribe(ce.LOCAL_JOINED, Ee), L.subscribe(ce.LOCAL_LEFT, pr);
    } catch (L) {
      Jn.current = !1, console.error("[SuperViz] Error starting room", L);
    }
    Jn.current = !1;
  }
  function ke() {
    if (!b) {
      console.error(
        "[SuperViz] Room not started yet, you need to start the room before stopping it"
      );
      return;
    }
    b.destroy(), dn({}), En(null), an(!1);
  }
  function gr(L) {
    b && (An[L.name] || b == null || b.addComponent(L), dn((On) => ({ ...On, [L.name]: L })));
  }
  function bn(L) {
    b && (An[L.name] && (b == null || b.removeComponent(L)), dn((On) => {
      const cn = { ...On };
      return delete cn[L.name], cn;
    }));
  }
  function Un(L) {
    if (Wn && Wn(L), !(L != null && L.activeComponents) || !(L != null && L.activeComponents.length)) return;
    const On = [];
    Object.keys(An).forEach((cn) => {
      var Z;
      (Z = L.activeComponents) != null && Z.includes(cn) || On.push(cn);
    }), dn((cn) => {
      const Z = { ...cn };
      return On.forEach((ue) => {
        delete Z[ue];
      }), Z;
    });
  }
  function Ee() {
    an(!0);
  }
  function pr() {
    an(!1);
  }
  const Zn = d_(() => ({
    hasProvider: !0,
    hasJoinedRoom: rn,
    room: b,
    addComponent: gr,
    removeComponent: bn,
    activeComponents: An,
    startRoom: Qe,
    stopRoom: ke
  }), [b, Kn, nn, yn, o, An, rn]);
  return /* @__PURE__ */ y_.jsx(sa.Provider, { value: Zn, children: ge });
}
export {
  C_ as S,
  sa as a,
  y_ as j,
  T_ as u
};
