// @superviz/react-sdk/dist/superviz-sdk-react.es36.js
import { u as O, j as c } from "./room-BEOW-846.js";
import { useState as d, useEffect as s } from "react";
import { m as r, j } from "./index-CtPfqlcy.js";
import { FieldEvents as i } from "./superviz-sdk-react.es37.js";
function U({
  onMount: t,
  onUnmount: u,
  disableOutline: E,
  disableRealtimeSync: l,
  fields: n,
  children: a,
  onInteraction: m,
  onContentChange: f
}) {
  const { room: b, component: e, addComponent: p } = O("formElements"), [N, T] = d(null);
  return s(() => {
    e && (e.unsubscribe(r.MOUNT), t && e.subscribe(r.MOUNT, t));
  }, [e, t]), s(() => {
    e && (e.unsubscribe(r.UNMOUNT), u && e.subscribe(r.UNMOUNT, u));
  }, [e, u]), s(() => {
    e && (e.unsubscribe(i.CONTENT_CHANGE), f && e.subscribe(i.CONTENT_CHANGE, f));
  }, [e, f]), s(() => {
    e && (e.unsubscribe(i.INTERACTION), m && e.subscribe(i.INTERACTION, m));
  }, [e, m]), s(() => {
    if (!b || N) return;
    const o = new j({
      fields: n,
      disableOutline: E,
      disableRealtimeSync: l
    });
    p(o), T(Date.now());
  }, [b]), s(() => {
    !e && N && T(null);
  }, [e]), a ?? /* @__PURE__ */ c.jsx(c.Fragment, {});
}
export {
  U as FormElements
};
