// @superviz/react-sdk/dist/superviz-sdk-react.es33.js
import { u as V, j as _ } from "./room-BEOW-846.js";
import { useState as W, useRef as s, useEffect as u, useCallback as Y } from "react";
import { c as w, N as t, p as z } from "./index-CtPfqlcy.js";
const P = (n) => ({
  [t.DESTROY]: n.onDestroy,
  [z.FRAME_DIMENSIONS_UPDATE]: n.onFrameDimensionsChange,
  [t.MEETING_WAITING_FOR_HOST]: n.onWaitingForHostChange,
  [t.MEETING_CONNECTION_STATUS_CHANGE]: n.onConnectionStatusChange,
  [t.MEETING_STATE_UPDATE]: n.onMeetingStateChange,
  [t.MEETING_SAME_PARTICIPANT_ERROR]: n.onSameAccountError,
  [t.MEETING_DEVICES_CHANGE]: n.onDevicesStateChange,
  [t.MEETING_HOST_CHANGE]: n.onHostChange,
  [t.MEETING_HOST_AVAILABLE]: n.onHostAvailable,
  [t.MEETING_NO_HOST_AVAILABLE]: n.onNoHostAvailable,
  [t.MEETING_START]: n.onMeetingStart,
  [t.MEETING_PARTICIPANT_JOINED]: n.onParticipantJoin,
  [t.MY_PARTICIPANT_JOINED]: n.onLocalParticipantJoin,
  [t.MEETING_PARTICIPANT_LEFT]: n.onParticipantLeave,
  [t.MY_PARTICIPANT_LEFT]: n.onLocalParticipantLeave,
  [t.MEETING_PARTICIPANT_LIST_UPDATE]: n.onParticipantListChange,
  [t.MEETING_PARTICIPANT_AMOUNT_UPDATE]: n.onParticipantAmountChange,
  [t.MEETING_KICK_PARTICIPANTS]: n.onKickAllParticipants,
  [t.MEETING_KICK_PARTICIPANT]: n.onKickLocalParticipant,
  mount: n.onMount,
  unmount: n.onUnmount
});
function Q(n) {
  const {
    onMeetingStart: c,
    onDestroy: M,
    onFrameDimensionsChange: p,
    onWaitingForHostChange: S,
    onConnectionStatusChange: g,
    onMeetingStateChange: m,
    onSameAccountError: L,
    onDevicesStateChange: G,
    onHostChange: h,
    onHostAvailable: R,
    onNoHostAvailable: O,
    onParticipantJoin: D,
    onLocalParticipantJoin: d,
    onParticipantLeave: f,
    onLocalParticipantLeave: H,
    onParticipantListChange: v,
    onParticipantAmountChange: F,
    onKickAllParticipants: K,
    onKickLocalParticipant: U,
    onMount: j,
    onUnmount: J
  } = n, { room: I, component: A, addComponent: y, removeComponent: x } = V("videoConference"), [C, a] = W(null), i = s({}), o = s(null);
  u(() => {
    if (!I || C || o.current) return;
    o.current = new w(n);
    const r = () => {
      x(o.current), o.current = null, a(null);
    };
    o.current.subscribe("destroy", r), N(n), y(o.current), a(Date.now());
  }, [I]), u(() => {
    const r = P(n);
    !o.current || !Object.values(i.current).length && Object.values(r).every((E) => !E) || N(n);
  }, [
    c,
    M,
    p,
    S,
    g,
    m,
    L,
    G,
    h,
    R,
    O,
    D,
    d,
    f,
    H,
    v,
    F,
    K,
    U,
    j,
    J
  ]), u(() => {
    !A && C && a(null);
  }, [A]);
  const N = Y((r) => {
    const E = P(r);
    Object.entries(E).forEach(([e, T]) => {
      i.current[e] && (o.current.unsubscribe(e, i.current[e]), i.current[e] = void 0), T && (o.current.subscribe(e, T), i.current[e] = T);
    });
  }, []);
  return /* @__PURE__ */ _.jsx(_.Fragment, {});
}
export {
  Q as VideoConference
};
