// @superviz/react-sdk/dist/superviz-sdk-react.es16.js
import { useRef as h, useState as C, useEffect as o, useCallback as r, useMemo as S } from "react";
import { u as d } from "./room-BEOW-846.js";
function v() {
  const { component: t, room: n, ...g } = d("videoConference"), e = h(null), [c, u] = C(!1);
  o(() => {
    t && (e.current = t, u(!0));
  }, [t]), o(() => {
    !n && e.current && (e.current = null, u(!1));
  }, [n]);
  const s = r(() => {
    e.current && e.current.toggleMeetingSetup();
  }, [e.current]), i = r(() => {
    e.current && e.current.toggleMicrophone();
  }, [e.current]), l = r(() => {
    e.current && e.current.toggleCam();
  }, [e.current]), a = r(() => {
    e.current && e.current.toggleScreenShare();
  }, [e.current]), f = r(() => {
    e.current && e.current.toggleChat();
  }, [e.current]), p = r(() => {
    e.current && e.current.toggleRecording();
  }, [e.current]), m = r(() => {
    e.current && e.current.hangUp();
  }, [e.current]);
  return S(() => ({
    isReady: c,
    toggleMeetingSetup: s,
    toggleMicrophone: i,
    toggleCam: l,
    toggleScreenShare: a,
    toggleChat: f,
    toggleRecording: p,
    hangUp: m
  }), [n, g.activeComponents, t, c]);
}
export {
  v as useVideo
};
