// @superviz/react-sdk/dist/superviz-sdk-react.es15.js
import { useRef as p, useState as w, useEffect as L, useCallback as a, useMemo as g } from "react";
import { u as v } from "./room-BEOW-846.js";
function R() {
  const { component: n, room: u, ...S } = v("yjsProvider"), e = p(null), [c, o] = w(!1);
  L(() => {
    n && (e.current = n, o(!0)), !u && c && (e.current = null, o(!1));
  }, [n]);
  const l = a(() => {
    var t;
    return (t = e.current) != null && t.awareness ? e.current.awareness.getLocalState() : null;
  }, [e.current]), i = a(
    (t) => {
      var s, r;
      (r = (s = e.current) == null ? void 0 : s.awareness) == null || r.setLocalState(t);
    },
    [e.current]
  ), m = a(
    (t, s) => {
      var r, f;
      (f = (r = e.current) == null ? void 0 : r.awareness) == null || f.setLocalStateField(t, s);
    },
    [l, i]
  ), d = a(() => {
    var t;
    return (t = e.current) != null && t.awareness ? e.current.awareness.getStates() : /* @__PURE__ */ new Map();
  }, [e.current]);
  return g(() => ({
    isReady: c,
    getLocalState: l,
    getStates: d,
    setLocalState: i,
    setLocalStateField: m,
    provider: e.current
  }), [u, S.activeComponents, n, c]);
}
export {
  R as useYjsProvider
};
