// @superviz/react-sdk/dist/superviz-sdk-react.es31.js
import { u as h, j as d } from "./room-BEOW-846.js";
import { useState as p, useEffect as r } from "react";
import { d as M } from "./index-CtPfqlcy.js";
const S = /* @__PURE__ */ new Map([
  ["onConnect", "connect"],
  ["onDisconnect", "disconnect"],
  ["onSynced", "synced"],
  ["onSync", "sync"],
  ["onDestroy", "destroy"],
  ["onMessage", "message"],
  ["onOutgoingMessage", "outgoingMessage"],
  ["onStateChange", "state"]
]);
function D({
  doc: g,
  awareness: j,
  onMount: x,
  onUnmount: E,
  children: v,
  ...o
}) {
  const { room: i, component: n, addComponent: y } = h("yjsProvider"), [a, c] = p(null), [s, C] = p({});
  return r(() => {
    var f;
    if (!n) return;
    const t = Object.entries(o), b = Object.entries(s);
    let u = !1;
    for (let e = 0; e < t.length; e++)
      if (t[e][1] !== ((f = b[e]) == null ? void 0 : f[1])) {
        u = !0;
        break;
      }
    u && (t.forEach(([e, m]) => {
      const l = S.get(e);
      s[e] && n.off(l, s[e]), m && n.on(l, m);
    }), C(o));
  }, [o]), r(() => {
    if (!i || a) return;
    const t = new M(g, { awareness: j });
    y(t), c(Date.now());
  }, [i]), r(() => {
    !n && a && c(null);
  }, [n]), v ?? /* @__PURE__ */ d.jsx(d.Fragment, {});
}
export {
  D as YjsProvider
};
