// @superviz/react-sdk/dist/tooltip-JQUVL2QH-Nwd00W2w.cjs
"use strict";Object.defineProperty(exports,Symbol.toStringTag,{value:"Module"});const P=require("./chunk-K5TMEGOR-CjzheANK.cjs");require("./chunk-3MEZQQ2J-Bai2OMk0.cjs");const b=require("./chunk-6U6BYUHV-h-2X1uK0.cjs"),z=require("./chunk-MVFOLHBC-C85clPX2.cjs"),p=require("./chunk-WG46WHJF-D0x9jAhS.cjs"),h=require("./index-N3n6WSly.cjs");h.y();h.y();var T=p.Ct`
  .superviz-who-is-online__tooltip {
    --host-height: 0px;
    --host-width: 0px;
    --vertical-offset: 12px;

    background-color: rgb(var(--sv-gray-600));
    padding: 10px;
    display: flex;
    flex-direction: column;
    gap: 8px;
    align-items: center;
    position: fixed;
    opacity: 0;
    border-radius: 2px;
    cursor: default;
    visibility: hidden;
    transition: opacity 0.2s ease-in-out display 0s;
    z-index: 100;
    overflow-x: clip;
  }

  .superviz-who-is-online__tooltip-arrow {
    width: 13px;
    height: 13px;
    position: absolute;
    background-color: rgb(var(--sv-gray-600));
    transform: rotate(45deg);
    border-top-left-radius: 1px;
    border-bottom-right-radius: 1px;
  }

  .show-tooltip {
    opacity: 1;
    visibility: visible;
  }

  .tooltip-name,
  .tooltip-action {
    margin: 0;
    font-family: roboto;
    white-space: nowrap;
    text-align: center;
    line-height: 1.2;
  }

  .tooltip-name {
    color: white;
    font-size: 14px;
  }

  .tooltip-action {
    color: rgb(var(--sv-gray-400));
    font-size: 12px;
  }

  .tooltip-top {
    bottom: calc(var(--host-height) + var(--vertical-offset));
  }

  .tooltip-bottom {
    bottom: auto;
  }

  .tooltip-center {
    right: auto;
  }

  .tooltip-bottom .superviz-who-is-online__tooltip-arrow {
    top: -6.5px;
  }

  .tooltip-top .superviz-who-is-online__tooltip-arrow {
    bottom: -6.5px;
  }

  .tooltip-center .superviz-who-is-online__tooltip-arrow {
    left: 0;
    margin-left: 50%;
    translate: -50% 0;
  }

  .tooltip-left .superviz-who-is-online__tooltip-arrow {
    translate: 50% 0;
    border-radius: 0;
    right: 0;
  }

  .tooltip-right .superviz-who-is-online__tooltip-arrow {
    translate: -50% 0;
    border-radius: 0;
    left: 0;
  }

  .shift-left {
    --vertical-offset: 2px;
  }
`;h.y();var y=z.D(p.C),_=[y.styles,T];exports.Tooltip=class extends y{constructor(){super(),this.positionFixedTooltip=()=>{var t,i,e,o,s;if(!this.canAnimate||!this.parentElement){this.hide();return}this.adjustTooltipPosition();let{bottom:n,left:r,width:l,right:d,top:x}=this.parentElement.getBoundingClientRect(),{width:g}=this.tooltip.getBoundingClientRect(),w="auto",f="auto",c="auto",m="auto",u="",v=window.innerWidth>780,a=v?14:12;this.tooltipHorizontalPosition.includes("right")&&(c=`${r+l/2}px`),this.tooltipHorizontalPosition.includes("left")&&(m=`${window.innerWidth-(d-l/2)}px`),this.tooltipHorizontalPosition.includes("center")&&(c=`${r}px`,u=`translateX(-${(g-l)/2}px)`),this.tooltipVerticalPosition.includes("bottom")&&(w=`${n+a}px`,a=-10),this.tooltipVerticalPosition.includes("top")&&(f=`${window.innerHeight-(x-a)}px`,a=10),this.shiftTooltipLeft&&(u=`translate(-${(g-(v?18:10)-42)/2}px, ${a}px)`),(t=this.tooltip)==null||t.style.setProperty("top",w),(i=this.tooltip)==null||i.style.setProperty("left",c),(e=this.tooltip)==null||e.style.setProperty("right",m),(o=this.tooltip)==null||o.style.setProperty("bottom",f),(s=this.tooltip)==null||s.style.setProperty("transform",u),this.animationFrame=window.requestAnimationFrame(this.positionFixedTooltip)},this.hide=()=>{this.canAnimate=!1,this.showTooltip=!1,window.cancelAnimationFrame(this.animationFrame)},this.show=()=>{this.canAnimate=!0,this.showTooltip=!0,this.positionFixedTooltip()},this.adjustTooltipVerticalPosition=()=>{let{bottom:t,top:i,width:e}=this.tooltip.getBoundingClientRect(),o=window.innerHeight;if(this.tooltipVerticalPosition.includes("top")&&i<0){this.tooltipVerticalPosition=this.tooltipVerticalPosition.replace("top","bottom");return}this.tooltipVerticalPosition.includes("bottom")&&t>o&&(this.tooltipVerticalPosition=this.tooltipVerticalPosition.replace("bottom","top"))},this.adjustTooltipHorizontalPosition=()=>{let{left:t,right:i,width:e}=this.tooltip.getBoundingClientRect(),o=window.innerWidth,{left:s,width:n,right:r}=this.parentElement.getBoundingClientRect(),l=s+n/2-e/2>0,d=e/2+r-n/2<o;if(l&&d){this.tooltipHorizontalPosition=this.tooltipHorizontalPosition.replace(/left|right/,"center");return}if(t<0){this.tooltipHorizontalPosition=this.tooltipHorizontalPosition.replace("center","right");return}i>o&&(this.tooltipHorizontalPosition=this.tooltipHorizontalPosition.replace("center","left"))},this.adjustTooltipPosition=()=>{if(!this.parentElement){this.hide();return}this.tooltip||(this.tooltip=this.shadowRoot.querySelector(".superviz-who-is-online__tooltip")),this.adjustTooltipVerticalPosition(),this.adjustTooltipHorizontalPosition()},this.tooltipVerticalPosition="tooltip-bottom",this.tooltipHorizontalPosition="tooltip-center",this.showTooltip=!1,this.parentSizes={height:0,width:0}}firstUpdated(t){let{parentElement:i}=this;i==null||i.addEventListener("mouseenter",this.show),i==null||i.addEventListener("mouseleave",this.hide),this.adjustTooltipPosition(),b.n.call(this,this.parentComponent)}disconnectedCallback(){super.disconnectedCallback();let{parentElement:t}=this;t==null||t.removeEventListener("mouseenter",this.show),t==null||t.removeEventListener("mouseleave",this.hide)}updated(t){if(t.has("showTooltip")&&this.showTooltip){let{parentElement:i}=this;if(!i)return;let{height:e,width:o}=i.getBoundingClientRect();(this.parentSizes.height!==e||this.parentSizes.width!==o)&&(this.parentSizes={height:e,width:o})}}getClass(t){return t?`${this.classesPrefix}__${t}`:this.classesPrefix}renderTooltip(){var t,i,e,o,s;let n=this.tooltipVerticalPosition,r=this.tooltipHorizontalPosition,l={"superviz-who-is-online__tooltip":!0,[this.getClass("")]:!0,[n]:!0,[r]:!0,"tooltip-extras":this.tooltipOnLeft,"show-tooltip":this.showTooltip,"shift-left":this.shiftTooltipLeft};return p.Vt`<div
      class=${P.u(l)}
      style="--host-height: ${(t=this.parentSizes)==null?void 0:t.height}px; --host-width: ${(i=this.parentSizes)==null?void 0:i.width}px;"
    >
      <p class="tooltip-name ${this.getClass("title")}">${(e=this.tooltipData)==null?void 0:e.name}</p>
      ${(o=this.tooltipData)!=null&&o.info?p.Vt`<p class="tooltip-action ${this.getClass("action")}">${(s=this.tooltipData)==null?void 0:s.info}</p>`:""}
      <div class="superviz-who-is-online__tooltip-arrow ${this.getClass("")}"></div>
    </div>`}render(){return p.Vt`${this.renderTooltip()}`}};exports.Tooltip.styles=_,exports.Tooltip.properties={tooltipData:{type:Object},tooltipOnLeft:{type:Boolean},showTooltip:{type:Boolean},tooltip:{type:Object},tooltipVerticalPosition:{type:String},tooltipHorizontalPosition:{type:String},parentSizes:{type:Object},shiftTooltipLeft:{type:Boolean},classesPrefix:{type:String},parentComponent:{type:String}},exports.Tooltip=h.F([p.ts("superviz-tooltip")],exports.Tooltip);
