// @superviz/react-sdk/dist/who-is-online-4MQU4252-D17VbSGQ.js
import { j as P } from "./chunk-DX57ODX6-DAIpi7BW.js";
import { y as p, e as v, i as C } from "./index-CtPfqlcy.js";
import { u as c } from "./chunk-K5TMEGOR-DaSvC8Mu.js";
import "./chunk-3MEZQQ2J-CHe_tQrN.js";
import { n as x } from "./chunk-6U6BYUHV-TZ4JwLT4.js";
import { D as f } from "./chunk-MVFOLHBC-BeDRo2V6.js";
import { C as b, V as l, a as y, t as $ } from "./chunk-WG46WHJF-JdNs68CV.js";
p();
p();
p();
p();
var D = b`
  .who-is-online__participant-list {
    display: flex;
    align-items: center;
    gap: 4px;
    position: relative;
  }

  .who-is-online {
    display: flex;
    flex-direction: column;
    position: fixed;
    z-index: 99;
  }

  .who-is-online__presence-control-message__text {
    margin: 0;
  }

  .who-is-online__participant {
    border-radius: 50%;
    box-sizing: border-box;
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    user-select: none;
    cursor: pointer;
    transition: opacity 0.3s ease-in-out;
    border: 2px solid #878291;
    border-radius: 50%;
    max-width: 40px;
    flex: 1 0 40px;
  }

  .followed {
    border-style: dashed !important;
    animation: rotate 15s linear infinite;
  }

  .followed .who-is-online__participant__avatar {
    animation: nullifyRotate 15s linear infinite;
  }

  .private {
    opacity: 0.3;
  }

  @keyframes rotate {
    100% {
      transform: rotate(1turn);
    }
  }

  @keyframes nullifyRotate {
    100% {
      transform: rotate(-1turn);
    }
  }

  .who-is-online__participant.disable-dropdown {
    cursor: default;
  }

  .who-is-online__participant__avatar {
    width: 32px;
    height: 32px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: 'Open Sans';
    font-size: 14px;
    line-height: 14px;
    line-height: 14px;
    font-weight: bold;
    color: #26242a;
    object-fit: contain;
  }

  .superviz-who-is-online__excess {
    width: 32px;
    height: 32px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: Roboto;
    font-size: 14px;
    line-height: 16px;
    text-align: center;
    font-weight: bold;
    cursor: pointer;
    color: #aea9b8;
  }

  .excess_participants:hover,
  .excess_participants--open {
    background-color: #aea9b8 !important;
  }

  .excess_participants:hover > div,
  .excess_participants--open > div {
    color: #fff !important;
  }

  @media (max-width: 780px) {
    .who-is-online__participant {
      width: 32px;
      height: 32px;
    }

    .who-is-online__participant {
      flex: 1 0 32px;
      max-width: 32px;
    }

    .who-is-online__participant__avatar {
      width: 24px;
      height: 24px;
    }

    .who-is-online__participant-list {
      gap: 8px;
    }

    .superviz-who-is-online__excess {
      width: 24px;
      height: 24px;
      font-size: 12px;
      line-height: 12px;
    }
  }
`;
p();
var E = b`
  .dropdown {
    position: relative;
  }

  .who-is-online__extra-participant {
    display: flex;
    user-select: none;
    align-items: center;
    justify-items: start;
    gap: 4px;
    width: 100%;
    padding: 10px;
    border-radius: 2px;
    position: relative;
    cursor: pointer;
  }

  .who-is-online__extra-participant.disable-dropdown {
    cursor: default;
  }

  .who-is-online__extra-participant:hover,
  .who-is-online__extra-participant--selected {
    background-color: rgb(var(--sv-gray-200));
  }

  .who-is-online__participant {
    border-radius: 50%;
    box-sizing: border-box;
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 40px;
    height: 40px;
    border: 2px solid #878291;
    border-radius: 50%;
    max-width: 40px;
    flex: 1 0 40px;
  }

  .who-is-online__participant__avatar {
    width: 32px;
    height: 32px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: 'Open Sans';
    font-size: 14px;
    line-height: 14px;
    font-weight: bold;
    color: #26242a;
    object-fit: contain;
  }

  .dropdown-list {
    position: relative;
    display: flex;
    flex-direction: column;
    z-index: 100;
  }

  .dropdown-list > div {
    padding: 4px;
    width: 216px;
    box-sizing: border-box;
  }

  .who-is-online__extras-dropdown {
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
    opacity: 0;
    display: none;
    background: #fff;
    padding: 0;
    z-index: 1;
    transition: 0.2s;
    border-radius: 3px;
    max-height: 240px;
    overflow: auto;
  }

  .who-is-online__extras-dropdown superviz-dropdown:hover {
    z-index: 999;
    position: relative;
  }

  .who-is-online__extras__arrow-icon {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .menu--bottom {
    top: 4px;
    min-width: 103px;
    position: absolute;
    right: 0;
  }

  .menu--top {
    bottom: 44px;
    min-width: 103px;
    position: absolute;
    right: 0px;
  }

  .menu-open {
    display: block;
    opacity: 1;
  }

  .superviz-who-is-online-dropdown__tooltip {
    background-color: rgb(var(--sv-gray-600));
    padding: 8px;
    display: flex;
    flex-direction: column;
    gap: 8px;
    align-items: center;
    position: absolute;
    top: 52px;
    left: 50%;
    transform: translateX(-50%);
    border-radius: 2px;
    opacity: 0;
    cursor: default;
    display: none;
    transition: opacity 0.2s ease-in-out display 0s;
  }

  .superviz-who-is-online-dropdown__tooltip-arrow {
    width: 13px;
    height: 13px;
    position: absolute;
    top: 0px;
    left: 50%;
    transform: rotate(45deg) translateX(-50%);
    background-color: rgb(var(--sv-gray-600));
    border-top-left-radius: 3px;
  }

  .dropdown-content:hover + .superviz-who-is-online-dropdown__tooltip {
    opacity: 1;
    display: block;
  }

  .tooltip-content {
    margin: 0;
    font-family: roboto;
    white-space: nowrap;
    text-align: center;
    color: white;
    font-size: 14px;
  }

  .who-is-online__extras__username {
    font-size: 14px;
    line-height: 20px;
    font-family: 'Open sans';
    color: rgb(var(--sv-gray-600));
  }

  .icon {
    flex: 1;
    justify-content: flex-end;
    display: flex;
  }

  .hide-icon {
    display: none;
  }

  @media (max-width: 780px) {
    .sv-icon,
    .who-is-online__participant {
      width: 32px;
      height: 32px;
    }

    .who-is-online__participant {
      flex: 1 0 32px;
      max-width: 32px;
    }

    .who-is-online__participant__avatar {
      width: 24px;
      height: 24px;
    }

    .dropdown-list > div {
      width: 192px;
    }

    .menu--top {
      bottom: 36px;
    }

    .superviz-who-is-online-dropdown__tooltip {
      top: 44px;
    }
  }
`;
p();
var M = b`
  .who-is-online__controls-messages {
    position: absolute;
    right: 0;
    display: flex;
    flex-direction: column;
    gap: 5px;
    margin-top: 4px;
  }

  .who-is-online__presence-control-message {
    box-sizing: border-box;
    font-size: 12px;
    padding: 8px 10px;
    font-family: 'Roboto';
    border-radius: 6px;
    background-color: #fff;
    color: rgb(var(--sv-gray-700));
    border: 2px solid #e0e0e0;
    white-space: nowrap;
  }

  .who-is-online__pcm__text {
    margin: 0;
  }

  .who-is-online__presence-control-message span {
    margin-left: 3px;
    text-decoration: underline;
    cursor: pointer;
  }

  .left-side {
    left: 0;
  }

  .right-side {
    right: 0;
  }

  .left-side .who-is-online__presence-control-message {
    align-self: flex-start;
  }

  .right-side .who-is-online__presence-control-message {
    align-self: flex-end;
  }

  .bottom-side {
    bottom: auto;
    top: calc(100% + 5px);
  }

  .top-side {
    top: auto;
    bottom: calc(100% + 5px);
  }
`, S = f(y), O = [S.styles, D], w = class extends S {
  constructor() {
    super(), this.onClickOutDropdown = ({ detail: n }) => {
      this.open = n.open;
    }, this.toggleShowTooltip = () => {
      this.showTooltip = !this.showTooltip;
    }, this.dropdownOptionsHandler = ({ detail: { label: n, participantId: r, source: a } }) => {
      switch (n) {
        case "go to":
          this.handleGoTo(r);
          break;
        case "follow":
          this.handleLocalFollow(r, a);
          break;
        case "unfollow":
          this.handleLocalUnfollow();
          break;
        case "private mode":
          this.handlePrivate(r);
          break;
        case "leave private mode":
          this.handleCancelPrivate(r);
          break;
        case "everyone follows me":
          this.handleFollow(r, a);
          break;
        case "stop followers":
          this.handleStopFollow();
          break;
        case "gather all":
          this.handleGatherAll(r);
          break;
      }
    }, this.position = "top: 20px; right: 40px;", this.showTooltip = !0, this.open = !1;
    let { localParticipant: i } = this.useStore("global-store"), { participants: e, following: o, extras: t, disablePresenceControls: s } = this.useStore("who-is-online-store");
    e.subscribe((n) => {
      this.participants = n;
    }), o.subscribe(), s.subscribe(), t.subscribe((n) => {
      this.amountOfExtras = n.length;
    }), i.subscribe((n) => {
      var r, a;
      let d = (r = n.activeComponents) == null ? void 0 : r.some((h) => h.toLowerCase().includes("presence"));
      this.localParticipantData = { id: n.id, joinedPresence: (a = n.activeComponents) == null ? void 0 : a.some((h) => h.toLowerCase().includes("presence")) }, this.disableDropdown = !d;
    });
  }
  firstUpdated(i) {
    super.firstUpdated(i), x.call(this, "who-is-online");
  }
  toggleOpen() {
    this.open = !this.open;
  }
  dropdownPosition(i) {
    if (this.participants.length === 1) return "bottom-center";
    if (i === 0) return "bottom-right";
    let e = this.participants.length > 4, o = i + 1 === this.participants.length;
    return e || !o ? "bottom-center" : "bottom-left";
  }
  renderExtras() {
    var i;
    if (!this.amountOfExtras) return;
    let e = { "who-is-online__participant": !0, excess_participants: !0, "excess_participants--open": this.open };
    return l`
      <superviz-who-is-online-dropdown
        @selected=${this.dropdownOptionsHandler}
        @clickout=${this.onClickOutDropdown}
        ?disableDropdown=${this.disableDropdown}
        ?showSeeMoreTooltip=${this.showTooltip}
        @toggle=${this.toggleOpen}
        @toggle-dropdown-state=${this.toggleShowTooltip}
        ?localParticipantJoinedPresence=${(i = this.localParticipantData) == null ? void 0 : i.joinedPresence}
        classesPrefix="who-is-online__controls"
        parentComponent="who-is-online"
        tooltipPrefix="who-is-online"
      >
        <div class=${c(e)} slot="dropdown">
          <div class="superviz-who-is-online__excess who-is-online__extras">
            +${this.amountOfExtras}
          </div>
        </div>
      </superviz-who-is-online-dropdown>
    `;
  }
  getAvatar({ color: i, imageUrl: e, firstLetter: o, letterColor: t }) {
    return e ? l` <img
        class="who-is-online__participant__avatar"
        style="background-color: ${i}"
        src=${e}
      />` : l`<div
      class="who-is-online__participant__avatar"
      style="background-color: ${i}; color: ${t}"
    >
      ${o}
    </div>`;
  }
  cancelPrivate() {
    this.isPrivate = void 0, this.emitEvent("realtime.private-mode", { id: this.localParticipantData.id });
  }
  renderParticipants() {
    return this.participants.length ? l`
      ${P(this.participants, (i) => i.id, (i, e) => {
      var o;
      let { avatar: t, id: s, name: n, tooltip: r, controls: a, disableDropdown: d, isLocalParticipant: h } = i, _ = this.dropdownPosition(e), m = ((o = this.following) == null ? void 0 : o.id) === s, F = { "who-is-online__participant": !0, "disable-dropdown": d, followed: m || this.everyoneFollowsMe && h, private: h && this.isPrivate };
      return l`
            <superviz-dropdown
              options=${JSON.stringify(a)}
              returnData=${JSON.stringify({ participantId: s, source: "participants" })}
              position=${_}
              name=${n}
              tooltipData=${JSON.stringify(r)}
              classesPrefix="who-is-online__controls"
              parentComponent="who-is-online"
              tooltipPrefix="who-is-online"
              ?disabled=${d}
              ?canShowTooltip=${this.showTooltip}
              @selected=${this.dropdownOptionsHandler}
              @toggle-dropdown-state=${this.toggleShowTooltip}
            >
              <div
                slot="dropdown"
                class=${c(F)}
                style="border-color: ${t.color}"
              >
                ${this.getAvatar(t)}
              </div>
            </superviz-dropdown>
          `;
    })}
    ` : l``;
  }
  handleGoTo(i) {
    this.emitEvent("realtime.go-to-participant", { id: i });
  }
  handleLocalFollow(i, e) {
    let { following: o } = this.useStore("who-is-online-store"), t = this.useStore("who-is-online-store")[e].value, { id: s, name: n, avatar: { color: r } } = t.find(({ id: a }) => a === i);
    this.everyoneFollowsMe && this.handleStopFollow(), o.publish({ name: n, id: s, color: r }), this.emitEvent("realtime.local-follow-participant", { id: s, source: e });
  }
  handleLocalUnfollow() {
    let { following: i } = this.useStore("who-is-online-store");
    i.publish(void 0), this.emitEvent("realtime.local-follow-participant", { id: void 0 });
  }
  handlePrivate(i) {
    this.everyoneFollowsMe && this.handleStopFollow(), this.emitEvent("realtime.private-mode", { id: i, isPrivate: !0 }), this.isPrivate = !0;
  }
  handleCancelPrivate(i) {
    this.emitEvent("realtime.private-mode", { id: i, isPrivate: !1 }), this.isPrivate = !1;
  }
  handleFollow(i, e) {
    this.isPrivate && this.cancelPrivate();
    let o = this.useStore("who-is-online-store")[e].value, { id: t, name: s, avatar: { color: n } } = o.find(({ id: a }) => a === i);
    this.everyoneFollowsMe = !0;
    let { following: r } = this.useStore("who-is-online-store");
    r.publish(void 0), this.emitEvent("realtime.follow-participant", { id: t, name: s, color: n });
  }
  handleStopFollow() {
    this.everyoneFollowsMe = !1, this.emitEvent("realtime.follow-participant", void 0);
  }
  handleGatherAll(i) {
    this.isPrivate && this.cancelPrivate(), this.emitEvent("realtime.gather", { id: i });
  }
  updated(i) {
    super.updated(i), this.updateComplete.then(() => {
      let e = this.shadowRoot.querySelector(".who-is-online");
      if (!e) return;
      let o = this.position.includes("left") ? "flex-start" : "flex-end", t = `${this.position} align-items: ${o};`;
      e.setAttribute("style", t);
    });
  }
  render() {
    return l`<div class="who-is-online who-is-online">
      <div class="who-is-online__participant-list">
        ${this.renderParticipants()} ${this.renderExtras()}
      </div>
      <superviz-who-is-online-messages
        ?everyoneFollowsMe=${this.everyoneFollowsMe}
        ?isPrivate=${this.isPrivate}
        @stop-following=${this.handleLocalUnfollow}
        @cancel-private=${this.cancelPrivate}
        @stop-everyone-follows-me=${this.handleStopFollow}
      ></superviz-who-is-online-messages>
    </div> `;
  }
};
w.styles = O, w.properties = { position: { type: String }, open: { type: Boolean }, localParticipantColor: { type: String }, isPrivate: { type: Boolean }, everyoneFollowsMe: { type: Boolean }, showTooltip: { type: Boolean } }, w = v([$("superviz-who-is-online")], w);
p();
p();
p();
var z = f(y), T = [z.styles, M], g = class extends z {
  constructor() {
    super(), this.repositionMessages = () => {
      let { following: o, everyoneFollowsMe: t, isPrivate: s } = this;
      if (o || t || s) {
        this.repositionInVerticalDirection(), this.repositionInHorizontalDirection(), this.animationFrame = window.requestAnimationFrame(this.repositionMessages);
        return;
      }
      window.cancelAnimationFrame(this.animationFrame);
    };
    let { localParticipant: i } = this.useStore("global-store");
    i.subscribe((o) => {
      var t, s;
      this.participantColor = (s = (t = o.slot) == null ? void 0 : t.color) != null ? s : C.gray;
    });
    let { following: e } = this.useStore("who-is-online-store");
    e.subscribe();
  }
  firstUpdated(i) {
    super.firstUpdated(i), x.call(this, "who-is-online");
  }
  updated(i) {
    super.updated(i), ["following", "everyoneFollowsMe", "isPrivate"].some((e) => i.has(e)) && this.repositionMessages();
  }
  repositionInVerticalDirection() {
    let i = window.innerHeight / 2, { top: e } = this.parentElement.getBoundingClientRect();
    if (e < i) {
      this.verticalSide = "bottom-side";
      return;
    }
    this.verticalSide = "top-side";
  }
  repositionInHorizontalDirection() {
    let i = window.innerWidth / 2, { left: e } = this.parentElement.getBoundingClientRect();
    if (e < i) {
      this.horizontalSide = "left-side";
      return;
    }
    this.horizontalSide = "right-side";
  }
  stopFollowing() {
    this.emitEvent("stop-following", {});
  }
  cancelPrivate() {
    this.emitEvent("cancel-private", {});
  }
  stopEveryoneFollowsMe() {
    this.emitEvent("stop-everyone-follows-me", {});
  }
  followingMessage() {
    if (!this.following) return "";
    let { name: i, color: e } = this.following;
    return l`<div
      class="who-is-online__following-message who-is-online__presence-control-message who-is-online__pcm"
      style="border-color: ${e}"
    >
      <p class="who-is-online__presence-control-message__text who-is-online__pcm__text">
        Following: ${i}
        <span
          class="who-is-online__presence-control-message__cancel-action-button who-is-online__pcm__cancel-action-button"
          @click=${this.stopFollowing}
          >Stop</span
        >
      </p>
    </div>`;
  }
  everyoneFollowsMeMessage() {
    return this.everyoneFollowsMe ? l`<div
      class="who-is-online__follow-me-message who-is-online__presence-control-message who-is-online__pcm"
      style="border-color: ${this.participantColor}"
    >
      <p class="who-is-online__presence-control-message__text who-is-online__pcm__text">
        Everyone is following you
        <span
          class="who-is-online__presence-control-message__cancel-action-button who-is-online__pcm__cancel-action-button"
          @click=${this.stopEveryoneFollowsMe}
          >Stop</span
        >
      </p>
    </div>` : "";
  }
  privateMessage() {
    return this.isPrivate ? l`<div
      class="who-is-online__private-mode-message who-is-online__presence-control-message who-is-online__pcm"
      style="border-color: ${this.participantColor}"
    >
      <p class="who-is-online__presence-control-message__text who-is-online__pcm__text">
        You are in Private Mode
        <span
          class="who-is-online__presence-control-message__cancel-action-button who-is-online__pcm__cancel-action-button"
          @click=${this.cancelPrivate}
          >Cancel</span
        >
      </p>
    </div>` : "";
  }
  render() {
    let i = { "who-is-online__controls-messages": !0, [this.verticalSide]: !0, [this.horizontalSide]: !0 };
    return l` <div class=${c(i)}>
      ${this.followingMessage()} ${this.everyoneFollowsMeMessage()} ${this.privateMessage()}
    </div>`;
  }
};
g.styles = T, g.properties = { everyoneFollowsMe: { type: Boolean }, isPrivate: { type: Boolean }, verticalSide: { type: String }, horizontalSide: { type: String } }, g = v([$("superviz-who-is-online-messages")], g);
p();
var k = f(y), L = [k.styles, E], u = class extends k {
  constructor() {
    super(), this.onClickOutDropdown = (e) => {
      if (e.stopPropagation(), !this.open) return;
      let o = e.composedPath(), t = this.shadowRoot.querySelector(".dropdown-content"), s = this.shadowRoot.querySelector(".dropdown-list"), n = this.shadowRoot.querySelector('slot[name="dropdown"]').assignedElements()[0], r = o.includes(t), a = o.includes(s), d = o.includes(n);
      r || a || d || this.close();
    }, this.close = () => {
      this.open = !1, this.selected = "", this.emitEvent("clickout", { bubbles: !1, composed: !1 });
    }, this.selectParticipant = (e, o) => () => {
      o || (this.selected = e);
    }, this.toggleShowTooltip = ({ detail: { open: e } }) => {
      this.showParticipantTooltip = !this.showParticipantTooltip, e || (this.selected = "");
    }, this.tooltip = () => this.showSeeMoreTooltip ? l`<superviz-tooltip
      tooltipData=${JSON.stringify({ name: "See more" })}
      classesPrefix="who-is-online__tooltip"
      parentComponent="who-is-online"
    ></superviz-tooltip>` : "", this.repositionDropdown = () => {
      if (!this.open || !this.parentElement) {
        window.cancelAnimationFrame(this.animationFrame);
        return;
      }
      this.repositionInVerticalDirection(), this.repositionInHorizontalDirection(), this.animationFrame = window.requestAnimationFrame(this.repositionDropdown);
    }, this.repositionInVerticalDirection = () => {
      let { bottom: e, top: o, height: t } = this.parentElement.getBoundingClientRect(), s = window.innerHeight / 2;
      if (o + t / 2 > s) {
        this.dropdownList.style.setProperty("bottom", `${window.innerHeight - o + 8}px`), this.dropdownList.style.setProperty("top", "");
        return;
      }
      this.dropdownList.style.setProperty("top", `${e + 8}px`), this.dropdownList.style.setProperty("bottom", "");
    }, this.repositionInHorizontalDirection = () => {
      let { right: e, left: o } = this.parentElement.getBoundingClientRect();
      this.dropdownList.style.setProperty("right", `${window.innerWidth - e}px`), this.dropdownList.style.setProperty("left", `${o}px`);
    }, this.selected = "", this.showParticipantTooltip = !0;
    let { extras: i } = this.useStore("who-is-online-store");
    i.subscribe((e) => {
      this.participants = e;
    });
  }
  firstUpdated(i) {
    this.shadowRoot.querySelector(".who-is-online__extras-dropdown").scrollTop = 0, x.call(this, "who-is-online"), this.dropdownList = this.shadowRoot.querySelector(".dropdown-list"), this.dropdownList.style.position = "fixed";
  }
  updated(i) {
    if (i.has("open")) {
      if (this.emitEvent("toggle-dropdown-state", { open: this.open, font: "toggle" }), this.open) {
        document.addEventListener("click", this.onClickOutDropdown);
        return;
      }
      document.removeEventListener("click", this.onClickOutDropdown);
    }
  }
  getAvatar({ color: i, imageUrl: e, firstLetter: o, letterColor: t }) {
    return e ? l` <img
        class="who-is-online__participant__avatar"
        style="background-color: ${i}"
        src=${e}
      />` : l`<div
      class="who-is-online__participant__avatar"
      style="background-color: ${i}; color: ${t}"
    >
      ${o}
    </div>`;
  }
  renderParticipants() {
    if (!this.participants) return;
    let i = this.participants.length - 1;
    return P(this.participants, (e) => e.id, (e, o) => {
      let { disableDropdown: t, id: s, avatar: n, controls: r, tooltip: a, name: d } = e, h = { "who-is-online__extra-participant": !0, "who-is-online__extra-participant--selected": this.selected === s, "disable-dropdown": t }, _ = { "who-is-online__extras__arrow-icon": !0, "hide-icon": t }, m = o === i;
      return l`
        <superviz-dropdown
          options=${JSON.stringify(r)}
          returnData=${JSON.stringify({ participantId: s, source: "extras" })}
          tooltipData=${JSON.stringify(a)}
          position="bottom-right"
          classesPrefix="who-is-online__controls"
          parentComponent="who-is-online"
          tooltipPrefix="who-is-online"
          ?disabled=${t}
          ?canShowTooltip=${this.showParticipantTooltip}
          ?shiftTooltipLeft=${!0}
          ?lastParticipant=${m}
          @toggle-dropdown-state=${this.toggleShowTooltip}
          @selected=${this.close}
        >
        <div 
          class=${c(h)} 
          @click=${this.selectParticipant(s, t)} slot="dropdown">
            <div class="who-is-online__participant" style="border-color: 
            ${n.color}">
              ${this.getAvatar(n)}
            </div>
            <span class="who-is-online__extras__username">${d}</span>
            <superviz-icon 
              class=${c(_)} 
              name="right" 
              color="var(--sv-gray-600)"
              size="sm"
            >
          </superviz-icon>
          </div>
        </div>
      </superviz-dropdown>
      `;
    });
  }
  toggle() {
    this.open = !this.open, this.open && (this.selected = "", this.repositionDropdown());
  }
  get menuClasses() {
    return { "who-is-online__extras-dropdown": !0, "menu--bottom": this.position === "bottom-side", "menu--top": this.position === "top-side", "menu-open": this.open };
  }
  render() {
    return l`
      <div class="dropdown">
        <div class="dropdown-content" @click=${this.toggle}>
          <slot name="dropdown"></slot>
        </div>
        ${this.tooltip()}
      </div>
      <div class="dropdown-list">
        <div class=${c(this.menuClasses)}>${this.renderParticipants()}</div>
      </div>
    `;
  }
};
u.styles = L, u.properties = { open: { type: Boolean }, align: { type: String }, position: { type: String }, selected: { type: String }, disableDropdown: { type: Boolean }, showSeeMoreTooltip: { type: Boolean }, showParticipantTooltip: { type: Boolean }, localParticipantJoinedPresence: { type: Boolean }, dropdownList: { type: Object } }, u = v([$("superviz-who-is-online-dropdown")], u);
export {
  w as WhoIsOnline,
  u as WhoIsOnlineDropdown,
  g as WhoIsOnlineMessages
};
